---
title: "RNAseq Analysis 2025"
author: Leo Zeef (Updates to the workflow source code are here https://github.com/leozeef2/DESeq2)
theme: united
output: 
 html_notebook:
   code_folding: hide
 toc: true
 toc_depth: 2
 toc_float: true
 collapsed: false
 smooth_scroll: false
---

## Analysis Sections {.tabset .tabset-pills}
### Preliminary

![Sample Numbers](samples1.png)
R libraries used
```{r}
# Example data that will run with this workflow can be downloaded from http://bartzabel.ls.manchester.ac.uk//Leo/lWDNQXgJIc
library(plotly)
library(apeglm)
library(DESeq2)
library(edgeR)
library(ggplot2)
library(RColorBrewer)
library(gplots)
library(ggrepel)
library(dplyr)
library(ComplexHeatmap)
library(clusterProfiler)
library(enrichR)
library(VennDiagram)
library(UpSetR)
library(gridExtra)
library(circlize) #colours
library(cluster)
library(factoextra)
library(NbClust)
library(biomaRt)
library(tidyverse)
library(reshape2)
library(reactome.db)
library(ReactomePA)
library(org.Hs.eg.db)#human
#library(org.Mm.eg.db)#mouse
#library(org.At.tair.db)#arabidopsis

writeLines(capture.output(sessionInfo()), "sessionInfo.txt")
capture.output(sessionInfo())

###########
#multiplot
###########
# Multiple plot function
#
# ggplot objects can be passed in ..., or to plotlist (as a list of ggplot objects)
# - cols:   Number of columns in layout
# - layout: A matrix specifying the layout. If present, 'cols' is ignored.

# If the layout is something like matrix(c(1,2,3,3), nrow=2, byrow=TRUE),
# then plot 1 will go in the upper left, 2 will go in the upper right, and
# 3 will go all the way across the bottom.

multiplot <- function(..., plotlist=NULL, file, cols=1, layout=NULL) {
  library(grid)
  
  # Make a list from the ... arguments and plotlist
  plots <- c(list(...), plotlist)
  
  numPlots = length(plots)
  
  # If layout is NULL, then use 'cols' to determine layout
  if (is.null(layout)) {
    # Make the panel
    # ncol: Number of columns of plots
    # nrow: Number of rows needed, calculated from # of cols
    layout <- matrix(seq(1, cols * ceiling(numPlots/cols)),
                     ncol = cols, nrow = ceiling(numPlots/cols))
  }
  
  if (numPlots==1) {
    print(plots[[1]])
    
  } else {
    # Set up the page
    grid.newpage()
    pushViewport(viewport(layout = grid.layout(nrow(layout), ncol(layout))))
    
    # Make each plot, in the correct location
    for (i in 1:numPlots) {
      # Get the i,j matrix positions of the regions that contain this subplot
      matchidx <- as.data.frame(which(layout == i, arr.ind = TRUE))
      
      print(plots[[i]], vp = viewport(layout.pos.row = matchidx$row,
                                      layout.pos.col = matchidx$col))
    }
  }
}
#function my code edit of plotPCA
#########
plotPCALeo<-function (x, intgroup = "Treatment", ntop = 500, returnData = FALSE, PCx=1, PCy=2)
{
  #rv <- rowVars(assay(x))
  rv = apply((assay(x)), 1, var)
  select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
                                                     length(rv)))]
  pca <- prcomp(t(assay(x)[select, ]))
  percentVar <- pca$sdev^2/sum(pca$sdev^2)
  if (!all(intgroup %in% names(colData(x)))) {
    stop("the argument 'intgroup' should specify columns of colData(dds)")
  }
  intgroup.df <- as.data.frame(colData(x)[, intgroup, drop = FALSE])
  group <- factor(apply(intgroup.df, 1, paste, collapse = " : "))
  d <- data.frame(PCX = pca$x[, PCx], PCY = pca$x[, PCy], group = group, 
                  intgroup.df, names = colnames(x))
  if (returnData) {
    attr(d, "percentVar") <- percentVar[PCx:PCy]
    return(d)
  }
  ggplot(data = d, aes_string(x = "PCX", y = "PCY", color = "group")) + 
    #ggplot(data = d, aes_string(x = "PCX", y = "PCY", color=Tgfb1, shape=Treatment)) + 
    geom_point(size = 3) + xlab(paste0("PC",PCx,": ", round(percentVar[1] * 
                                                              100), "% variance")) + ylab(paste0("PC",PCy,": ", round(percentVar[2] * 
                                                                                                                        100), "% variance"))
}
```


```{r}
#Colours
col_funF = colorRamp2(c(-1.5,-0.2, 0,0.2, 1.5), c("blue","cyan", "grey90","orange", "red"))#heatmap colours
col_fun = colorRamp2(c(-2, 0, 2), c("green", "black", "red"))
colorsV <- c("cornflowerblue", "yellow3", "brown1")#Venn colours
colorsV2 <- c("mediumorchid1",  "chartreuse3")#Venn colours
colorsV3 <- c("cornflowerblue", "yellow3", "brown1")#Venn colours
colorsV4<-c("cornflowerblue", "green3","purple","red")#Venn colours
colorsV5<-c("cornflowerblue", "orange2", "green3","purple","red")#Venn colours
#https://htmlcolorcodes.com/color-picker/
#https://html-color-codes.info/colors-from-image/
#col_fun(seq(-3, 3))
```

If starting from STAR output files then no edits required. If starting from a gene counts file then the chunk below needs to be run (and the 3 chunks after that giving qc information need to be skipped as they won't run successfully). 
```{r}
# If starting from a gene counts file then uncomment the line below and read in your counts file
#countsTable<-read.delim("mergedCounts_rev.txt", header = TRUE, sep = "\t",check.names=FALSE,row.names=1)
```


```{r}
#Read in counts from STAR output files ReadsPerGene.out.tab
ff <- list.files(  pattern = "*ReadsPerGene.out.tab$", full.names = TRUE )#assume files in working directory
counts.files <- lapply( ff, read.table, skip = 4 )
#countsTable <- as.data.frame( sapply( counts.files, function(x) x[ , 2 ] ) )#unstranded
countsTable <- as.data.frame( sapply( counts.files, function(x) x[ , 4 ] ) )#reverse
ff <- gsub( "ReadsPerGene.out.tab", "", ff )#tidy up sample names
ff <- gsub( "./", "", ff )#tidy up sample names
colnames(countsTable) <- ff
row.names(countsTable) <- counts.files[[1]]$V1
#head(countsTable)
columnSums<-as.data.frame(colSums(countsTable))
colnames(columnSums) <- c("ReadCount")
#columnSums
pX1<-ggplot(data=columnSums, aes(x=row.names(columnSums),y=ReadCount,fill=ReadCount)) +
  geom_bar(position="dodge", stat="identity", color="blue")+  ggtitle("Number of Reads Assigned to Genes")+
  coord_flip()+theme(axis.title.y = element_blank())+theme(axis.title.x = element_blank())+
  scale_x_discrete(limits=rev)
pX1
```




Mapping data from STAR matching log.final.out files
```{r,fig.height = 12,fig.width = 12}
library(data.table)  
files <- list.files(pattern = "Log.final.out")

read_plus <- function(flnm) {
    fread(flnm, sep="\t",fill=TRUE, header=FALSE) %>% 
        mutate(Sample_Name = gsub("Log.final.out","",flnm))
}

temp <- lapply(files, read_plus)
data <- rbindlist( temp )
#data

#remove unwanted rows
toMatch <- c("UNIQUE READS:","Uniquely mapped reads number", "Number of reads mapped to multiple loci", "Number of reads unmapped: too short","Number of reads unmapped: other")
data <- data[grep(paste(toMatch,collapse="|"), data$V1), ]
data$V2<-as.numeric(as.character(data$V2))

#lookup colSums from columnSums dataframe for assigned reads calculation
data$Assigned<-with(columnSums, ReadCount[match(data$Sample_Name, rownames(columnSums))])

#rename categories
data = mutate(data, Mapping_Assignments=
                ifelse(grepl("UNIQUE READS:", data$V1), "Assigned_to_Genes",
                   ifelse(grepl("Uniquely mapped reads number", data$V1), "Unassigned_Non_exonic", 
                          ifelse(grepl("Number of reads mapped to multiple loci", data$V1), "Unassigned_Multi_Mapping",
                                 ifelse(grepl("Number of reads unmapped: too short", data$V1), "Unassigned_Unmapped",
                                 "OtherUnmapped")))))


                             
#calculate unmapped reads and assigned reads
data = mutate(data, Read_Count=
                ifelse(grepl("UNIQUE READS:", data$V1), data$Assigned,
                   ifelse(grepl("Uniquely mapped reads number", data$V1), (data$V2-data$Assigned), 
                          ifelse(grepl("Number of reads mapped to multiple loci", data$V1), data$V2,
                                 ifelse(grepl("Number of reads unmapped: too short", data$V1), c(data$V2+lead(data$V2)),
                                 "NA")))))
#data
data <- subset(data, Mapping_Assignments!="OtherUnmapped")
data$Read_Count<-as.numeric(as.character(data$Read_Count))

data$Mapping_Assignments<-as.factor(as.character(data$Mapping_Assignments))
data$Mapping_Assignments <- relevel(data$Mapping_Assignments, 'Unassigned_Multi_Mapping')
data$Mapping_Assignments <- relevel(data$Mapping_Assignments, 'Unassigned_Non_exonic')
data$Mapping_Assignments <- relevel(data$Mapping_Assignments, 'Assigned_to_Genes')

#cleanup
data <- subset(data, select = -c(Assigned))
#data

qcplot<-ggplot(data, aes(x=Sample_Name, y=Read_Count, group=Mapping_Assignments, fill=Mapping_Assignments)) + 
    geom_bar(position = position_stack(reverse = TRUE), stat="identity")+
  scale_fill_manual(values=c("dodgerblue1","green","red","orange"))+
  scale_x_discrete(limits = rev(levels(as.factor(data$Sample_Name))))+
  coord_flip()+
    theme(axis.title = element_text(colour="black", size = 18)) + 
  theme(axis.text = element_text(colour="black", size = 14)) + 
  #theme(legend.title = element_text(colour="black", size = 14)) +
   theme(legend.title = element_blank()) +
  theme(legend.text = element_text(colour="black", size = 12)) +
  theme(legend.position="bottom")

qcplot2<-ggplot(data, aes(x=Sample_Name, y=Read_Count, group=Mapping_Assignments, fill=Mapping_Assignments)) + 
    geom_bar(position = position_fill(reverse = TRUE), stat="identity")+
  scale_fill_manual(values=c("dodgerblue1","green","red","orange"))+
  scale_x_discrete(limits = rev(levels(as.factor(data$Sample_Name))))+
  coord_flip()+
    theme(axis.title = element_text(colour="black", size = 18)) + 
  theme(axis.text = element_text(colour="black", size = 14)) + 
  theme(legend.position="bottom")+
  theme(legend.title = element_blank()) +
  theme(legend.text = element_text(colour="black", size = 12)) +
    labs( y = "Read_Count_Fraction")

multiplot(qcplot, qcplot2, cols=1)


```

```{r}
data_print<-reshape(data, idvar = "Sample_Name", timevar = "Mapping_Assignments", direction = "wide")
data_print<-data_print[,c(1,4,7,10,13)]
write.csv((data_print),"counts_qc.csv")
data_print
```

```{r}
#countsTable <-countsTable[,c(1:9, 15:63)]
Sample_group<-factor(c(rep("Cntl",4),rep("AAA1",4),rep("BAA2",4),rep("CAA3",4)),
             levels=c("Cntl","AAA1","BAA2","CAA3"))#note names
Genotype<-factor(c(rep("WT",8),rep("KO",8)),
             levels=c("WT","KO"))#note names
Drug<-factor(c(rep("Untreated",4),rep("AZT",4),rep("Untreated",4),rep("AZT",4)),
             levels=c("Untreated","AZT"))#note names
Batch<-factor(c("A","B","C","D","A","B","C","D","A","B","C","D","A","B","C","D"), levels=c("A","B","C","D"))
Sex<-factor(c("M","M","F","F","M","M","F","F","M","M","F","F","M","M","F","F"), levels=c("F","M"))

myNames<-colnames(countsTable)[1:16]#note length
colDataNames<-data.frame(row.names=myNames, Sample_group=Sample_group,Genotype=Genotype,Drug=Drug,Batch=Batch,Sex=Sex)
colDataNames
write.csv(colDataNames,"colDataNames.csv")
write.csv(colDataNames,"design.csv")
des<-formula(~Sample_group)
ddsHTSeq<-DESeqDataSetFromMatrix(countsTable, colData=colDataNames, design=des, ignoreRank = FALSE)

```


Run DESeq2 analysis
```{r}
dds<-DESeq(ddsHTSeq,betaPrior=FALSE)

```

```{r}
#Export normalised data
normCounts<-as.data.frame(counts(dds,normalized=TRUE))
colnames(normCounts) <- paste0(colnames(normCounts),'_normalised')
write.csv(normCounts,"DESeq_outputNormalised.csv")
baseMeans <- sapply( levels(dds$Sample_group), function(lvl) rowMeans( counts(dds,normalized=TRUE)[,dds$Sample_group == lvl,drop = FALSE] ) )
colnames(baseMeans) <- paste0(colnames(baseMeans),'_mean')
write.csv(baseMeans,"DESeq_Means_Sample_group.csv")
```


```{r}
#logData <- rlogTransformation(dds, blind=TRUE)
logData <- varianceStabilizingTransformation(dds, blind=TRUE)#quicker for big dataset

```

### PCA
```{r,fig.height = 7,fig.width = 17}

#vsd <- varianceStabilizingTransformation(dds, blind=TRUE)

PCAdata<-plotPCA(logData, intgroup=c("Sample_group"),ntop = 500,returnData=TRUE)

percentVar <- round(100 * attr(PCAdata, "percentVar"))#,digits=2)
myplot1<-qplot(PC1, PC2, color=Sample_group, data=PCAdata) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ggtitle("PCA") +
  scale_size_manual(values=c(3,5,7,9))+
  #scale_color_manual(values=c("#F8766D",,#7CAE00","#00BFC4","#C77CFF"))+
  geom_point(size=6) +
  guides(colour = guide_legend(override.aes = list(size=6))) +
  guides(shape = guide_legend(override.aes = list(size=6))) +
  theme(plot.title = element_text(colour="black", size = 16)) + 
  theme(axis.title = element_text(colour="black", size = 14)) + 
  theme(legend.title = element_text(colour="black", size = 16)) + 
  theme(legend.text = element_text(colour="black", size = 16)) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
#myplot1


#3v4 
PCAdata<-plotPCALeo(logData, intgroup=c("Sample_group"),ntop = 500,returnData=TRUE, PCx=3, PCy=4)
percentVar <- round(100 * attr(PCAdata, "percentVar"))#,digits=2)
myplot2<-qplot(PCX, PCY, color=Sample_group, data=PCAdata) +
  xlab(paste0("PC3: ",percentVar[1],"% variance")) +
  ggtitle("PCA") +
  scale_size_manual(values=c(3,5,7,9))+
  geom_point(size=6) +
  guides(colour = guide_legend(override.aes = list(size=6))) +
  guides(shape = guide_legend(override.aes = list(size=6))) +
  theme(plot.title = element_text(colour="black", size = 16)) + 
  theme(axis.title = element_text(colour="black", size = 14)) + 
  theme(legend.title = element_text(colour="black", size = 16)) + 
  theme(legend.text = element_text(colour="black", size = 16)) +
  ylab(paste0("PC4: ",percentVar[2],"% variance"))
#myplot2

multiplot(myplot1, myplot2, cols=2)

```

```{r}
ggplot_build(myplot1)$data
```

labelled
```{r,fig.height = 7,fig.width = 17}
#named
myNames2 <-c("01","02","03","04","05","06","07","08","09","10","11","12","13","14","15","16")
#myNames2 <-c("01","02","03","04","05","06","07","08","09","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36")
myplot1b<-myplot1+geom_text_repel(aes(label=myNames2),nudge_x = 1,nudge_y = 0.5,force = 1, max.overlaps = Inf)
myplot2b<-myplot2+geom_text_repel(aes(label=myNames2),nudge_x = 0.5,nudge_y = 0.5,force = 1, max.overlaps = Inf)

multiplot(myplot1b, myplot2b, cols=2)
```

```{r,fig.height = 7,fig.width = 17}

#vsd <- varianceStabilizingTransformation(dds, blind=TRUE)
PCAdata<-plotPCA(logData, intgroup=c("Sample_group"),ntop = 500,returnData=TRUE)

percentVar <- round(100 * attr(PCAdata, "percentVar"))#,digits=2)
myplot1<-qplot(PC1, PC2, color=Genotype, shape=Drug, data=PCAdata) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ggtitle("PCA") +
  scale_size_manual(values=c(3,5,7,9))+
  geom_point(size=6) +
  guides(colour = guide_legend(override.aes = list(size=6))) +
  guides(shape = guide_legend(override.aes = list(size=6))) +
  theme(plot.title = element_text(colour="black", size = 16)) + 
  theme(axis.title = element_text(colour="black", size = 14)) + 
  theme(legend.title = element_text(colour="black", size = 16)) + 
  theme(legend.text = element_text(colour="black", size = 16)) +
  theme_bw()+
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
#myplot1



#3v4 
PCAdata<-plotPCALeo(logData, intgroup=c("Sample_group"),ntop = 500,returnData=TRUE, PCx=3, PCy=4)
percentVar <- round(100 * attr(PCAdata, "percentVar"))#,digits=2)
myplot2<-qplot(PCX, PCY, color=Genotype, shape=Drug, data=PCAdata) +
  xlab(paste0("PC3: ",percentVar[1],"% variance")) +
  ggtitle("PCA") +
  scale_size_manual(values=c(3,5,7,9))+
  geom_point(size=6) +
  guides(colour = guide_legend(override.aes = list(size=6))) +
  guides(shape = guide_legend(override.aes = list(size=6))) +
  theme(plot.title = element_text(colour="black", size = 16)) + 
  theme(axis.title = element_text(colour="black", size = 14)) + 
  theme(legend.title = element_text(colour="black", size = 16)) + 
  theme(legend.text = element_text(colour="black", size = 16)) +
  theme_bw()+
  ylab(paste0("PC4: ",percentVar[2],"% variance"))
#myplot2

multiplot(myplot1, myplot2, cols=2)

```

labelled
```{r,fig.height = 7,fig.width = 17}
#named
#myNames2 <-c("01","02","03","04","05","06","07","08","09","10","11","12")

myplot1b<-myplot1+geom_text_repel(aes(label=myNames2),nudge_x = 1,nudge_y = 0.5,force = 1, max.overlaps = Inf,size=4)
myplot2b<-myplot2+geom_text_repel(aes(label=myNames2),nudge_x = 0.5,nudge_y = 0.5,force = 1, max.overlaps = Inf,size=4)

multiplot(myplot1b, myplot2b, cols=2)
```

```{r}
pca_data <-plotPCA(logData, intgroup=c("Sample_group"),ntop = 500,returnData=TRUE) #PC1&PC2
pca_data$Genotype<-Genotype
pca_data3_3<-plotPCALeo(logData, intgroup=c("Sample_group"),ntop = 500,returnData=TRUE, PCx=3, PCy=4) #PC1&PC2
pca_data$PC3<-pca_data3_3$PCX
pca_data$PC4<-pca_data3_3$PCY
pca_data3_3<-plotPCALeo(logData, intgroup=c("Sample_group"),ntop = 500,returnData=TRUE, PCx=5, PCy=6) #PC1&PC2
pca_data$PC5<-pca_data3_3$PCX
pca_data$PC6<-pca_data3_3$PCY
(pca_data)
```

```{r}
#Genotype<-c(6,6,6,6,6,6,3,3,3,3,3,3,6,6,6,6,6,6,3,3,3,3,3,3)
#GenotypePl<-c("circle","triangle","circle","triangle","circle","triangle","circle","triangle","circle","triangle","circle","triangle","circle","triangle","circle","triangle","circle","triangle","circle","triangle","circle","triangle","circle","triangle")
```

```{r,fig.height = 17,fig.width = 17}
plot_ly()%>%
  add_trace(data = pca_data, x = ~PC1, y = ~PC2, z = ~PC3, 
            mode = 'markers', 
            color = Drug,
            symbol=Genotype,
           # symbol=GenotypePl,
            #size = ~Batch, sizes=c(60,200),
           text = ~Sample_group, hoverinfo = 'text')%>%
        layout(title = '3D PCA Plot')
```


```{r}
DESeq2::plotDispEsts(dds, main="Dispersion Estimates")#dispersion plot
```
Sample Outlier Detection. 
```{r,fig.height = 8,fig.width = 17}
par(mar=c(8,5,2,2))
boxplot(log10(assays(dds)[["cooks"]]), range=0, las=2,main="Boxplot of Cook's distances for each Sample_group (for each sample) NB. outlier samples will have high values.")
```


```{r}
#Biomart annotation
BMannot<-read.table(text=row.names(countsTable), sep=".", header=FALSE, col.names = paste0("Ensembl", 1:2), stringsAsFactors=FALSE)#strip off the Ensembl version number
BMannot$Original_ID<-rownames(countsTable)
head(BMannot)

```


```{r}
#library( "biomaRt" )
#mouse
#ensembl = useMart( "ensembl", dataset = "mmusculus_gene_ensembl" )#mouse
#ensembl= useEnsembl( "ensembl", dataset = "mmusculus_gene_ensembl")
#listAttributes(ensembl)
#Genemap <- getBM( attributes = c("ensembl_gene_id","mgi_symbol", "description", "chromosome_name","start_position","end_position","strand","gene_biotype","entrezgene_id"),
#filters = "ensembl_gene_id",values = BMannot$Ensembl1,mart = ensembl )

#Or for human
ensembl = useMart( "ensembl", dataset = "hsapiens_gene_ensembl" )
Genemap <- getBM( attributes = c("ensembl_gene_id","hgnc_symbol", "description", "chromosome_name","start_position","end_position","strand","gene_biotype","entrezgene_id"),
filters = "ensembl_gene_id",values = BMannot$Ensembl1,mart = ensembl )

#alternatives to use if biomart site is flakey
#ensembl= useEnsembl( "ensembl", dataset = "mmusculus_gene_ensembl")
#ensembl= useEnsembl( "ensembl", dataset = "hsapiens_gene_ensembl")
#ensembl = useMart("ensembl", dataset = "hsapiens_gene_ensembl", host="useast.ensembl.org", ensemblRedirect = FALSE)

#other organisms
#ensembl <- useEnsembl(biomart = "ensembl")
#listDatasets(mart = ensembl)
```

```{r}
idx <- match( BMannot$Ensembl1, Genemap$ensembl_gene_id )
#BMannot$Gene_Symbol<- Genemap$mgi_symbol[ idx ]#mouse
BMannot$Gene_Symbol<- Genemap$hgnc_symbol[ idx ]#human
BMannot$description<- Genemap$description[ idx ]
BMannot$Ensembl<- Genemap$ensembl_gene_id[ idx ]
BMannot$chr<- Genemap$chromosome_name[ idx ]
BMannot$start<- Genemap$start_position[ idx ]
BMannot$end<- Genemap$end_position[ idx ]
BMannot$strand<- Genemap$strand[ idx ]
BMannot$gene_biotype<- Genemap$gene_biotype[ idx ]
BMannot$entrez<- Genemap$entrezgene_id[ idx ]
BMannot$Gene_Symbol <- ifelse(BMannot$Gene_Symbol=="", BMannot$Ensembl, BMannot$Gene_Symbol)
head(BMannot)
write.table(BMannot,"biomart.txt",  sep = "\t")
```


```{r}
#add optional supplied gene sets as annotation
bonus<-read.delim("list1.txt", header = TRUE, sep = "\t",check.names=FALSE)
#ncol(bonus)
bonus

```


```{r}
BMannot2<-BMannot
BMannot2$Gene_Symbol[BMannot2$Gene_Symbol==""]<-"null"  #if no gene symbol call it "null"
BMannot2[is.na(BMannot2)] = "_"
idxb<-match( BMannot2$Gene_Symbol, bonus$Chosen)
summary(idxb)
BMannot2$Chosen<- bonus$Chosen[ idxb ]
BMannot2$Chosen[!is.na(BMannot2$Chosen)] = "Chosen"
BMannot2[is.na(BMannot2)] = "Remaining_genes"
head(BMannot2)
```




```{r}
#Make some data holders for multiple contrasts
#Data frame of all DE results
baseMeans <- sapply( levels(dds$Sample_group), function(lvl) rowMeans( counts(dds,normalized=TRUE)[,dds$Sample_group == lvl,drop = FALSE] ) )
colnames(baseMeans) <- paste0(colnames(baseMeans),'_mean')
normCounts<-as.data.frame(counts(dds,normalized=TRUE))
colnames(normCounts) <- paste0(colnames(normCounts),'_normalised')
countsTableT<-countsTable
colnames(countsTableT) <- paste0(colnames(countsTableT),'_rawCounts')
resAll <- cbind(BMannot2, countsTableT, normCounts, baseMeans)
baseMeans <- sapply( levels(dds$Sample_group), function(lvl) rowMeans( counts(dds,normalized=TRUE)[,dds$Sample_group == lvl,drop = FALSE] ) )
head(resAll)

```

```{r}
ipa1000<-resAll[,c(3,5)]
head(ipa1000)
```

```{r}
#Data frame of all DE counts
DEcounts <- c('pvalue<0.05','p0.05_fc<0','p0.05_fc>0','padj<0.1','padj0.1_fc<0','padj0.1_fc>0')
DEcounter <- data.frame(DEcounts)

```

```{r}
library("variancePartition")
library("variancePartition")
rv <- rowVars(assay(logData)) ## calculate the variance for each gene
isexpr <- order(rv, decreasing=TRUE)[seq_len(min(500, length(rv)))] ## select the ntop genes by variance
quantLog <- assay(logData)[isexpr,]
form <- ~ (1 | Sample_group)+ (1 | Genotype)+ (1 | Drug) + (1 | Sex) + (1 | Batch)
colDataNames2<-data.frame(row.names=myNames, Sample_group=Sample_group,Genotype=Genotype,Drug=Drug,Sex=Sex,Batch=Batch)
varPart <- fitExtractVarPartModel(quantLog, form, colDataNames2)
```
```{r,fig.height = 7,fig.width = 10}
vp <- sortCols(varPart)
plotPercentBars(vp[1:50, ],main = "Bar plot of variance fractions for the first 50 genes")
```

```{r}
plotVarPart(vp,main = "Violin plot of contribution of each variable to total variance")
```

```{r}
#heatmap annotation of individual sample (from https://www.biostars.org/p/317349/)
# NB edits required to colours of levels of factor for each experiment & if levels are dropped
dataHM<-assay(logData)

#single factor annotation
y<-data.frame(t(dataHM),Sample_group)
metadataPlot<-y[,c(ncol(y)-1,ncol(y))]
ann <- data.frame(metadataPlot$Sample_group)
colnames(ann) <- c("Sample_group")#1 factor
colours <- list("Sample_group"=c("Cntl"="#F8766D","AAA1"="#7CAE00","BAA2"="#00BFC4","CAA3"="#C77CFF"))

#2 factors annotation
#y<-data.frame(t(dataHM),Genotype,Drug)
#metadataPlot<-y[,c(ncol(y)-1,ncol(y))]
#ann <- data.frame(metadataPlot$Genotype,metadataPlot$Drug)
#colnames(ann) <- c("Genotype","Drug")
#colours <- list("Genotype"=c("WT"="#C77CFF","KO"="#00BFC4"), "Drug"=c("Untreated"="cyan","AZT"="khaki1"))

#3 factors annotation
#y<-data.frame(t(dataHM),Genotype,Drug,Batch)
#metadataPlot<-y[,c(ncol(y)-2,ncol(y)-1,ncol(y))]
#ann <- data.frame(metadataPlot$Genotype,metadataPlot$Drug,metadataPlot$Batch)
#colnames(ann) <- c("Genotype","Drug","Batch")
#colours <- list("Genotype"=c("WT"="#C77CFF","KO"="#00BFC4"), "Drug"=c("Untreated"="cyan","AZT"="khaki1"), "Batch"=c("A"="pink","B"="orange","C"="brown"))

colAnn <- HeatmapAnnotation(df=ann, which="col", col=colours, annotation_width=unit(c(2, 4), "cm"), gap=unit(1, "mm"))

#[1] "#F8766D"
#[2] "#F8766D" "#00BFC4"
#[3] "#F8766D" "#00BA38" "#619CFF"
#[4] "#F8766D" "#7CAE00" "#00BFC4" "#C77CFF"
#[5] "#F8766D" "#A3A500" "#00BF7D" "#00B0F6" "#E76BF3"
#[6] "#F8766D" "#B79F00" "#00BA38" "#00BFC4" "#619CFF" "#F564E3"
#[7] "#F8766D" "#C49A00" "#53B400" "#00C094" "#00B6EB" "#A58AFF" "#FB61D7"
#[8] "#F8766D" "#CD9600" "#7CAE00" "#00BE67" "#00BFC4" "#00A9FF" "#C77CFF" "#FF61CC"

```


```{r}
#heatmap annotation for mean values
baseMeans <- sapply( levels(dds$Sample_group), function(lvl) rowMeans( counts(dds,normalized=TRUE)[,dds$Sample_group == lvl,drop = FALSE] ) )
head(baseMeans)
```


```{r}
#heatmap annotation for mean values
Sample_groupMn<-factor(c("Cntl","AAA1","BAA2","CAA3"))#note names
GenotypeMn<-factor(c(rep("WT",2),rep("KO",2)),
             levels=c("WT","KO"))#note names
DrugMn<-factor(c(rep("Untreated",1),rep("AZT",1),rep("Untreated",1),rep("AZT",1)),
             levels=c("Untreated","AZT"))#note names
dataHMm<-baseMeans

#single factor annotation
ym<-data.frame(t(dataHMm),Sample_groupMn)
metadataPlotm<-ym[,c(ncol(ym)-1,ncol(ym))]
annm <- data.frame(metadataPlotm$Sample_groupMn)#single factor
colnames(annm) <- c("Sample_groupMn")#1 factor
coloursm <- list("Sample_groupMn"=c("Cntl"="#F8766D","AAA1"="#7CAE00","BAA2"="#00BFC4","CAA3"="#C77CFF"))

#two factors annotation
#ym<-data.frame(t(dataHMm),GenotypeMn,DrugMn)
#metadataPlotm<-ym[,c(ncol(ym)-1,ncol(ym))]
#annm <- data.frame(metadataPlotm$GenotypeMn,metadataPlotm$DrugMn)#2 factors
#colnames(annm) <- c("GenotypeMn","DrugMn")#2 factors
#coloursm <- list("GenotypeMn"=c("WT"="#C77CFF","KO"="#00BFC4"), "DrugMn"=c("Untreated"="cyan","AZT"="khaki1"))

#three factors annotation
#ym<-data.frame(t(dataHMm),GenotypeMn,DrugMn,BatchMn)
#metadataPlotm<-ym[,c(ncol(ym)-2,ncol(ym)-1,ncol(ym))]
#annm <- data.frame(metadataPlotm$GenotypeMn,metadataPlotm$DrugMn,metadataPlotm$BatchMn)
#colnames(annm) <- c("GenotypeMn","DrugMn","BatchMn")
#coloursm <- list("GenotypeMn"=c("WT"="#C77CFF","KO"="#00BFC4"), "DrugMn"=c("Untreated"="cyan","AZT"="khaki1"), "BatchMn"=c("A"="pink","B"="orange","C"="brown"))

colAnnm <- HeatmapAnnotation(df=annm, which="col", col=coloursm, annotation_width=unit(c(2, 4), "cm"), gap=unit(1, "mm"))

```


```{r,fig.height = 12,fig.width = 20}
#heatmap of top 50 by variance across whole dataset
topVarGenes <- head(order(-rowVars(assay(logData))),50)#find indexes of top 50 Genes

dataHM<-assay(logData)[ topVarGenes, ]
dataHM<- t(as.matrix(dataHM))
dataHM <- t(scale(dataHM))#scale (normalise) across Genes

GeneSymbolHM<-BMannot2[ topVarGenes, ]

hmap_hier_factors<- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topVarGenes, ])$Gene_Symbol),
  column_title = "Hierarchical Clustering of Top 50 Genes by Variance", 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  #top_annotation_height=unit(1.0,"cm"), 
  show_row_names = TRUE,
  top_annotation=colAnn  )
draw(hmap_hier_factors, heatmap_legend_side = "left", annotation_legend_side = "left")

hmap_hier_factors2<- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topVarGenes, ])$Gene_Symbol),
  column_title = "Hierarchical Clustering of Top 50 Genes by Variance", 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(1000, "mm"),
  col = col_funF,
  #top_annotation_height=unit(1.0,"cm"), 
  show_row_names = TRUE,
   cluster_columns = FALSE,
  top_annotation=colAnn  )
#draw(hmap_hier_factors2, heatmap_legend_side = "left", annotation_legend_side = "left")

#means
dataHMm<-baseMeans[ topVarGenes, ]
dataHMm <- log2(dataHMm+1)
dataHMm<- t(as.matrix(dataHMm))
dataHMm <- t(scale(dataHMm))

hmap_hier_factors4 <- Heatmap(
  dataHMm,  name = "Expression means",
  column_title = paste0("means"), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  top_annotation=colAnnm)

draw(hmap_hier_factors4+hmap_hier_factors, heatmap_legend_side = "left", annotation_legend_side = "left")

```

```{r,fig.height = 8,fig.width = 20}
#heatmap of selected genes
moduleName="Chosen"
moduleGenes <- BMannot2$Chosen=="Chosen"#find indexes
dataHM<-assay(logData)[ moduleGenes, ]
dataHM<- t(as.matrix(dataHM))
dataHM <- t(scale(dataHM))#scale (normalise) across Genes
dataHM[is.nan(dataHM)] <- 0#NaN if standard dev is zero

hmap_hier_factors2 <- Heatmap(
  dataHM,  #name = "Expression",
  #row_labels = paste0(BMannot[ moduleGenes, ]$Gene_Symbol," ",BMannot[ moduleGenes, ]$Original_ID),
  column_title = paste0(moduleName, " Genes Heatmap in order"), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(150, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  cluster_columns = FALSE,
  top_annotation=colAnn  )

hmap_hier_factors3 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0(BMannot[ moduleGenes, ]$Gene_Symbol),
  column_title = paste0(moduleName, " Genes Heatmap Samples Clustered"), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(150, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  cluster_columns = TRUE,
  top_annotation=colAnn  )
#hmap_hier_factors2

draw(hmap_hier_factors2+hmap_hier_factors3, heatmap_legend_side = "left", annotation_legend_side = "left")

```



```{r,fig.height = 8,fig.width = 20}
#heatmap of selected genes
moduleName="Chosen"
moduleGenes <- BMannot2$Chosen=="Chosen"#find indexes
dataHM<-assay(logData)[ moduleGenes, ]
dataHM<- t(as.matrix(dataHM))
dataHM <- t(scale(dataHM))#scale (normalise) across Genes
dataHM[is.nan(dataHM)] <- 0#NaN if standard dev is zero

hmap_hier_factors2 <- Heatmap(
  dataHM,  #name = "Expression",
  row_labels = paste0(BMannot[ moduleGenes, ]$Gene_Symbol),
  column_title = paste0(moduleName, " Genes Heatmap in order"), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(150, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  #cluster_columns = FALSE,
  top_annotation=colAnn  )

#means
dataHMm<-baseMeans[ moduleGenes, ]
dataHMm <- log2(dataHMm+1)
dataHMm<- t(as.matrix(dataHMm))
dataHMm <- t(scale(dataHMm))
dataHMm[is.nan(dataHMm)] <- 0#NaN if standard dev is zero

hmap_hier_factors3 <- Heatmap(
  dataHMm,  name = "Expression",
  row_labels = paste0(BMannot[ moduleGenes, ]$Gene_Symbol),
  column_title = paste0(moduleName, " Means"), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  #cluster_columns = FALSE,
  top_annotation=colAnnm  )
#hmap_hier_factors2

draw(hmap_hier_factors3+hmap_hier_factors2, heatmap_legend_side = "left", annotation_legend_side = "left")

```






```{r}
resultsNames(dds)
```

### AAA1_vs_Cntl

```{r}

titleDE1<-'Sample_group_AAA1_vs_Cntl'
res<-results(dds,contrast=list(titleDE1))#,alpha=0.05,cooksCutoff=FALSE,independentFiltering=FALSE,alpha=0.05)
titleDE<-'TestA'
print(paste0("Summary of results for ",titleDE))
print("If outlier number is too high (e.g. many hundreds or thousands) then samples may need removing")
print("If outlier number is high (e.g. many thousands) then consider running cooksCutoff=FALSE and warning user")
#http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#approach-to-count-outliers
summary(res)
##str(res)
write.csv(res,paste0("DESeq_",titleDE,".csv"))
#save logfc, pvalue, padj from all comparisons to one data frame resAll
resS <- res[,c(2,5,6)]
resS$log2FoldChange[is.na(resS$log2FoldChange)] <- 0 #replace logfc NA to zero
head(resS)
colnames(resS) <- paste0(colnames(resS),'_',titleDE)
resAll <- cbind(resAll, resS)
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
DEcounter[titleDE] <- c((sum(results$pvalue<0.05,na.rm=TRUE)), 
                        (sum(results$pvalue<0.05 & results$log2FoldChange<0 , na.rm=TRUE)),
                        (sum(results$pvalue<0.05 & results$log2FoldChange>0 , na.rm=TRUE)),
                        (sum(results$padj<0.1,na.rm=TRUE)), 
                        (sum(results$padj<0.1 & results$log2FoldChange<0 , na.rm=TRUE)),
                        (sum(results$padj<0.1 & results$log2FoldChange>0 , na.rm=TRUE))
                        )

```

```{r}
#str(DEcounter)
DEcounter
```

```{r}
#pvalue and padj histograms
#par(mfrow=c(1,2))
#metadata(res)$alpha # significance level
#metadata(res)$filterTheta
#metadata(res)$filterThreshold

plot(metadata(res)$filterNumRej,main=paste0("Independent Filtering ",titleDE),
     type="b", ylab="number of rejections",
     xlab="quantiles of baseMean")
lines(metadata(res)$lo.fit, col="red")
abline(v=metadata(res)$filterTheta)
```


```{r,fig.height = 6,fig.width = 12}
#pvalue and padj histograms
par(mfrow=c(1,2))
use <- res$baseMean > metadata(res)$filterThreshold
h1 <- hist(res$pvalue[!use], breaks=0:50/50, plot=FALSE)
h2 <- hist(res$pvalue[use], breaks=0:50/50, plot=FALSE)
colori <- c(`do not pass`="khaki", `pass`="powderblue")
barplot(height = rbind(h1$counts, h2$counts), beside = FALSE,
        col = colori, space = 0, main = "Effect of Independent Filtering on pvalues", ylab="frequency")
text(x = c(0, length(h1$counts)), y = 0, label = paste(c(0,1)),
     adj = c(0.5,1.7), xpd=NA)
legend("top", fill=rev(colori), legend=rev(names(colori)))
hist( res$padj, breaks=20, col="powderblue",main=paste0("Histogram of padj ",titleDE),xlab="" )
```

```{r,fig.height = 10,fig.width = 14}
##volcano plots
```

```{r,fig.height = 10,fig.width = 14}
#volcano plots
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
results<-cbind(results,BMannot2)
results <- results[order(-1*results$pvalue),]#so significant dots on top of not significant

#pick significance thresholds
results = mutate(results, Significance=
                   ifelse(results$pvalue<0.05&results$log2FoldChange>1&!is.na(results$pvalue), "pvalue<0.05 & fold change>2", 
                          ifelse(results$pvalue<0.05&results$log2FoldChange<(-1)&!is.na(results$pvalue), "pvalue<0.05 & fold change<2",
                                 "Not Significant")))
results = mutate(results, SignificanceFDR=
                   ifelse(results$padj<0.1&results$log2FoldChange>1&!is.na(results$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(results$padj<0.1&results$log2FoldChange<(-1)&!is.na(results$pvalue), "padj<0.1 & fold change<2",
                                 "Not Significant")))
results = mutate(results, SignificanceFDR2=
                   ifelse(results$padj<0.1&!is.na(results$pvalue), "padj<0.1", 
                           "Not Significant"))
results<-results[complete.cases(results), ]#remove rows with NA
results=mutate(results,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
#volcano plots
xmax <- max(c(abs(min(results$log2FoldChange)),abs(max(results$log2FoldChange))))
p1= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Significance)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+2), y=2, label= "pvalue<0.05",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-280))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#volcano plot highlighting FDR
p2= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=SignificanceFDR)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1", results[,"SignificanceFDR2"]))]))-0.01, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+1), y=-log10(results$pvalue[(match("padj<0.1", results[,"SignificanceFDR2"]))])+1, label= "padj<0.1",color="chartreuse4") +  
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero
#MA plots
p3= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Significance)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))
p4= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=SignificanceFDR)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

multiplot(p1, p3, p2,p4, cols=2)

```

Mouseover for gene names
```{r,fig.height = 10,fig.width = 12}
#xplotlyp1 %>%  ggplotly(tooltip = "text") #remove "xplotly"
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp2 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp3 %>%  ggplotly(tooltip = "text")
```


```{r,fig.height = 10,fig.width = 14}
#volcano plots with bonus names
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
results<-cbind(results,BMannot2)
results <- results[order(-1*results$pvalue),]#so significant dots on top of not significant


results=results %>%  arrange(desc(Chosen))
results<-results[complete.cases(results), ]#remove rows with NA
results=mutate(results,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
results = mutate(results, SignificanceFDR=
                   ifelse(results$padj<0.1&results$log2FoldChange>1&!is.na(results$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(results$padj<0.1&results$log2FoldChange<(-1)&!is.na(results$pvalue), "padj<0.1 fold change<2",
                                 "Not Significant")))
results = mutate(results, Significance=
                   ifelse(results$pvalue<0.05&abs(results$log2FoldChange)>1&!is.na(results$pvalue), "sig", # NB choose cutoff 
                                 "Not Significant"))
xmax <- max(c(abs(min(results$log2FoldChange)),abs(max(results$log2FoldChange))))
p1= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(results$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
#xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#MA plots
p2= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

p3= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(results$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#MA plots
p4= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))


multiplot( p1, p2,p3,p4, cols=2)
```
Only showing gene names where pvalue<0.05 and logfc>1

```{r}
#shrunk LFC
resLFC<-lfcShrink(dds,coef=titleDE1, type="apeglm")#,cooksCutoff=FALSE,independentFiltering=FALSE,alpha=0.05)
write.csv(resLFC,paste0("DESeq_LFC_",titleDE,".csv"))
#save logfc, pvalue, padj from all comparisons to one data frame resAll
resSs <- resLFC[,c(2,4,5)]
resSs$log2FoldChange[is.na(resSs$log2FoldChange)] <- 0 #replace logfc NA to zero
#head(resS)
colnames(resSs) <- paste0(colnames(resSs),'Shrunk_',titleDE)
resAll <- resAll[1:(length(resAll)-3)]
resAll <- cbind(resAll, resSs)
resAll <- resAll[1:(length(resAll)-2)]
resAll <- cbind(resAll, resS)
```

```{r,fig.height = 10,fig.width = 14}
#volcano plots
resultsLFCs<-read.csv(paste0("DESeq_LFC_",titleDE,".csv"),header=TRUE)
resultsLFCs<-cbind(resultsLFCs,BMannot2)
resultsLFCs <- resultsLFCs[order(-1*resultsLFCs$pvalue),]#so significant dots on top of not significant

#pick significance thresholds
resultsLFCs = mutate(resultsLFCs, Significance=
                   ifelse(resultsLFCs$pvalue<0.05&resultsLFCs$log2FoldChange>1&!is.na(resultsLFCs$pvalue), "pvalue<0.05 & fold change>2", 
                          ifelse(resultsLFCs$pvalue<0.05&resultsLFCs$log2FoldChange<(-1)&!is.na(resultsLFCs$pvalue), "pvalue<0.05 & fold change<2",
                                 "Not Significant")))
resultsLFCs = mutate(resultsLFCs, SignificanceFDR=
                   ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange>1&!is.na(resultsLFCs$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange<(-1)&!is.na(resultsLFCs$pvalue), "padj<0.1 & fold change<2",
                                 "Not Significant")))
resultsLFCs = mutate(resultsLFCs, SignificanceFDR2=
                   ifelse(resultsLFCs$padj<0.1&!is.na(resultsLFCs$pvalue), "padj<0.1", 
                           "Not Significant"))
resultsLFCs<-resultsLFCs[complete.cases(resultsLFCs), ]#remove rows with NA
resultsLFCs=mutate(resultsLFCs,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
#volcano plots
xmax <- max(c(abs(min(resultsLFCs$log2FoldChange)),abs(max(resultsLFCs$log2FoldChange))))
p1= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Significance)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+2), y=2, label= "pvalue<0.05",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-280))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero

#volcano plot highlighting FDR
p2= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=SignificanceFDR)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=(-log10(resultsLFCs$pvalue[(match("padj<0.1", resultsLFCs[,"SignificanceFDR2"]))]))-0.01, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+1), y=-log10(resultsLFCs$pvalue[(match("padj<0.1", resultsLFCs[,"SignificanceFDR2"]))])+1, label= "padj<0.1",color="chartreuse4") +  
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero
#MA plots
p3= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Significance)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))
p4= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=SignificanceFDR)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

multiplot(p1, p3, p2,p4, cols=2)

```

Mouseover for gene names
```{r,fig.height = 10,fig.width = 12}
#xplotlyp1 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp2 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp3 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 14}
#volcano plots with bonus names
resultsLFCs<-read.csv(paste0("DESeq_LFC_",titleDE,".csv"),header=TRUE)
resultsLFCs<-cbind(resultsLFCs,BMannot2)
resultsLFCs <- resultsLFCs[order(-1*resultsLFCs$pvalue),]#so significant dots on top of not significant


resultsLFCs=resultsLFCs %>%  arrange(desc(Chosen))
resultsLFCs<-resultsLFCs[complete.cases(resultsLFCs), ]#remove rows with NA
resultsLFCs=mutate(resultsLFCs,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
resultsLFCs = mutate(resultsLFCs, SignificanceFDR=
                   ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange>1&!is.na(resultsLFCs$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange<(-1)&!is.na(resultsLFCs$pvalue), "padj<0.1 fold change<2",
                                 "Not Significant")))
resultsLFCs = mutate(resultsLFCs, Significance=
                   ifelse(resultsLFCs$pvalue<0.05&abs(resultsLFCs$log2FoldChange)>1&!is.na(resultsLFCs$pvalue), "sig", # NB choose cutoff 
                                 "Not Significant"))
xmax <- max(c(abs(min(resultsLFCs$log2FoldChange)),abs(max(resultsLFCs$log2FoldChange))))
p1= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(resultsLFCs$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero

#MA plots
p2= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

p3= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(resultsLFCs$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 5,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero

#MA plots
p4= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))


multiplot( p1, p2,p3,p4, cols=2)
```

```{r,fig.height = 12,fig.width = 20}
#heatmaps of top Genes
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
topDEgenes <- head(order(results$pvalue,results$padj),50)#find indexes of top 50 Genes
dataHM<-assay(logData)[ topDEgenes, ]
dataHM<- t(as.matrix(dataHM))
dataHM <- t(scale(dataHM))#scale (normalise) across Genes

data_pval<-results[ topDEgenes, ]
data_padj<-data_pval[,c(7)]
data_pval<-data_pval[,c(6)]
#data_pval
#head(data_pval)
pvalHm<-Heatmap(
  data_pval,  name = "pvalues",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(10, "mm"),
  col = rev(RColorBrewer::brewer.pal(name = "Greys", n = 5)),
  cluster_rows = FALSE,
  show_row_names = TRUE)

padjHm<-Heatmap(
  data_padj,  name = "padj",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(10, "mm"),
  col = rev(RColorBrewer::brewer.pal(name = "Greens", n = 5)),
  cluster_rows = FALSE,
  show_row_names = TRUE)
  
  
hmap_hier_factors1 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Top 50 DE Genes comparison ",titleDE," (ranked most significant at the top)"), #don't cluster rows for this one
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  cluster_rows = FALSE,
  show_row_names = TRUE,
  top_annotation=colAnn  )

hmap_hier_factors2 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Hierarchical clustering Top 50 DE Genes comparison ",titleDE), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  top_annotation=colAnn  )

#means
dataHMm<-baseMeans[ topDEgenes, ]
dataHMm <- log2(dataHMm+1)
dataHMm<- t(as.matrix(dataHMm))
dataHMm <- t(scale(dataHMm))

hmap_hier_factors3 <- Heatmap(
  dataHMm,  name = "Expression means",
column_title = paste0("means"), #don't cluster rows for this one
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
col = col_funF,
  cluster_rows = FALSE,
  show_row_names = TRUE,
top_annotation=colAnnm)

hmap_hier_factors4 <- Heatmap(
  dataHMm,  name = "Expression means",
  column_title = paste0("means"), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  top_annotation=colAnnm)

draw(hmap_hier_factors3+hmap_hier_factors1+pvalHm+padjHm, heatmap_legend_side = "left", annotation_legend_side = "left")
draw(hmap_hier_factors4+hmap_hier_factors2+pvalHm+padjHm, heatmap_legend_side = "left", annotation_legend_side = "left")

```




View selected Top Gene expression
```{r}
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
topDEgenesU <- head(order(results$pvalue,results$padj),1)#find index of top Gene

d <- plotCounts(dds, gene=as.vector(results$X[ topDEgenesU ]), intgroup="Sample_group", #intgroup is the factor we are using
                returnData=TRUE)

ggplot(d, aes(x=Sample_group, y=count)) + ggtitle(BMannot$Gene_Symbol[ topDEgenesU ]) +
  geom_point(position=position_jitter(w=0.1,h=0)) + 
  ylab("normalised count")+  scale_y_log10(breaks=c(25,100,400))
```


```{r}
#Lists of Gene Names for Venn and Functional Annotation on top 1000 Genes using clusterprofiler and enrichR.
#GO
topDEgenes <- head(order(results$pvalue,results$padj),1000)#find indexes of top 1000 Genes
assign(paste0("GOlist1000_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("Reactomelist1000_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
topDEgenes <- which(results$padj<0.1 & results$log2FoldChange>1 &!is.na(results$padj))#find indexes
assign(paste0("GOlistUp_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("ReactomeUplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
topDEgenes <- which(results$padj<0.1 & results$log2FoldChange<0-1 &!is.na(results$padj))#find indexes 
assign(paste0("GOlistDown_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("ReactomeDownlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
#Venn
topDEgenes <- which(results$pvalue<0.05&!is.na(results$pvalue))#find indexes 
assign(paste0("p0.05list_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$pvalue<0.05 & results$log2FoldChange>0&!is.na(results$pvalue))#find indexes  
assign(paste0("p0.05uplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$pvalue<0.05 & results$log2FoldChange<0&!is.na(results$pvalue))#find indexes 
assign(paste0("p0.05downlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1&!is.na(results$padj))#find indexes  
assign(paste0("padj0.1list_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1 & results$log2FoldChange>0 &!is.na(results$padj))#find indexes
assign(paste0("padj0.1uplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1 & results$log2FoldChange<0 &!is.na(results$padj))#find indexes 
assign(paste0("padj0.1downlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

```

```{r}
resIpa<-res[,c(2,5,6)]
topDEgenes <- head(order(resIpa$pvalue,resIpa$padj),1000)#find indexes of top 1000 Genes
resIpa$log2FoldChange[is.na(resIpa$log2FoldChange)] <- 0 #replace logfc NA to zero
resIpa$padj <-1
resIpa$padj[ topDEgenes ] <-resIpa$padj[ topDEgenes ]* 0
colnames(resIpa) <- paste0(colnames(resIpa),'_',titleDE)
ipa1000<-cbind(ipa1000,resIpa)
```

```{r}
resultsNames(dds)
```

### BAA2_vs_Cntl

```{r}
titleDE1<-'Sample_group_BAA2_vs_Cntl'
res<-results(dds,contrast=list(titleDE1))#,cooksCutoff=FALSE,independentFiltering=FALSE,alpha=0.05)
titleDE<-'TestB'
print(paste0("Summary of results for ",titleDE))
print("If outlier number is too high (e.g. many hundreds or thousands) then samples may need removing")
print("If outlier number is high (e.g. many thousands) then consider running cooksCutoff=FALSE and warning user")
#http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#approach-to-count-outliers
summary(res)
##str(res)
write.csv(res,paste0("DESeq_",titleDE,".csv"))
#save logfc, pvalue, padj from all comparisons to one data frame resAll
resS <- res[,c(2,5,6)]
resS$log2FoldChange[is.na(resS$log2FoldChange)] <- 0 #replace logfc NA to zero
head(resS)
colnames(resS) <- paste0(colnames(resS),'_',titleDE)
resAll <- cbind(resAll, resS)
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
DEcounter[titleDE] <- c((sum(results$pvalue<0.05,na.rm=TRUE)), 
                        (sum(results$pvalue<0.05 & results$log2FoldChange<0 , na.rm=TRUE)),
                        (sum(results$pvalue<0.05 & results$log2FoldChange>0 , na.rm=TRUE)),
                        (sum(results$padj<0.1,na.rm=TRUE)), 
                        (sum(results$padj<0.1 & results$log2FoldChange<0 , na.rm=TRUE)),
                        (sum(results$padj<0.1 & results$log2FoldChange>0 , na.rm=TRUE))
                        )

```


```{r}
#pvalue and padj histograms
#par(mfrow=c(1,2))
#metadata(res)$alpha # significance level
#metadata(res)$filterTheta
#metadata(res)$filterThreshold

plot(metadata(res)$filterNumRej,main=paste0("Independent Filtering ",titleDE),
     type="b", ylab="number of rejections",
     xlab="quantiles of baseMean")
lines(metadata(res)$lo.fit, col="red")
abline(v=metadata(res)$filterTheta)
```


```{r,fig.height = 6,fig.width = 12}
#pvalue and padj histograms
par(mfrow=c(1,2))
use <- res$baseMean > metadata(res)$filterThreshold
h1 <- hist(res$pvalue[!use], breaks=0:50/50, plot=FALSE)
h2 <- hist(res$pvalue[use], breaks=0:50/50, plot=FALSE)
colori <- c(`do not pass`="khaki", `pass`="powderblue")
barplot(height = rbind(h1$counts, h2$counts), beside = FALSE,
        col = colori, space = 0, main = "Effect of Independent Filtering on pvalues", ylab="frequency")
text(x = c(0, length(h1$counts)), y = 0, label = paste(c(0,1)),
     adj = c(0.5,1.7), xpd=NA)
legend("top", fill=rev(colori), legend=rev(names(colori)))
hist( res$padj, breaks=20, col="powderblue",main=paste0("Histogram of padj ",titleDE),xlab="" )
```



```{r,fig.height = 10,fig.width = 14}
#volcano plots
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
results<-cbind(results,BMannot2)
results <- results[order(-1*results$pvalue),]#so significant dots on top of not significant

#pick significance thresholds
results = mutate(results, Significance=
                   ifelse(results$pvalue<0.05&results$log2FoldChange>1&!is.na(results$pvalue), "pvalue<0.05 & fold change>2", 
                          ifelse(results$pvalue<0.05&results$log2FoldChange<(-1)&!is.na(results$pvalue), "pvalue<0.05 & fold change<2",
                                 "Not Significant")))
results = mutate(results, SignificanceFDR=
                   ifelse(results$padj<0.1&results$log2FoldChange>1&!is.na(results$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(results$padj<0.1&results$log2FoldChange<(-1)&!is.na(results$pvalue), "padj<0.1 & fold change<2",
                                 "Not Significant")))
results = mutate(results, SignificanceFDR2=
                   ifelse(results$padj<0.1&!is.na(results$pvalue), "padj<0.1", 
                           "Not Significant"))
results<-results[complete.cases(results), ]#remove rows with NA
results=mutate(results,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
#volcano plots
xmax <- max(c(abs(min(results$log2FoldChange)),abs(max(results$log2FoldChange))))
p1= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Significance)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+2), y=2, label= "pvalue<0.05",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-280))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#volcano plot highlighting FDR
p2= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=SignificanceFDR)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1", results[,"SignificanceFDR2"]))]))-0.01, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+1), y=-log10(results$pvalue[(match("padj<0.1", results[,"SignificanceFDR2"]))])+1, label= "padj<0.1",color="chartreuse4") +  
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero
#MA plots
p3= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Significance)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))
p4= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=SignificanceFDR)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

multiplot(p1, p3, p2,p4, cols=2)

```

Mouseover for gene names
```{r,fig.height = 10,fig.width = 12}
#xplotlyp1 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp2 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp3 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 14}
#volcano plots with bonus names
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
results<-cbind(results,BMannot2)
results <- results[order(-1*results$pvalue),]#so significant dots on top of not significant


results=results %>%  arrange(desc(Chosen))
results<-results[complete.cases(results), ]#remove rows with NA
results=mutate(results,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
results = mutate(results, SignificanceFDR=
                   ifelse(results$padj<0.1&results$log2FoldChange>1&!is.na(results$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(results$padj<0.1&results$log2FoldChange<(-1)&!is.na(results$pvalue), "padj<0.1 fold change<2",
                                 "Not Significant")))
results = mutate(results, Significance=
                   ifelse(results$pvalue<0.05&abs(results$log2FoldChange)>1&!is.na(results$pvalue), "sig", # NB choose cutoff 
                                 "Not Significant"))
xmax <- max(c(abs(min(results$log2FoldChange)),abs(max(results$log2FoldChange))))
p1= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(results$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#MA plots
p2= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

p3= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(results$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#MA plots
p4= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))


multiplot( p1, p2,p3,p4, cols=2)
```
Only showing gene names where pvalue<0.05 and logfc>1

```{r}
#shrunk LFC
resLFC<-lfcShrink(dds,coef=titleDE1, type="apeglm")#,cooksCutoff=FALSE,independentFiltering=FALSE,alpha=0.05)
write.csv(resLFC,paste0("DESeq_LFC_",titleDE,".csv"))
#save logfc, pvalue, padj from all comparisons to one data frame resAll
resSs <- resLFC[,c(2,4,5)]
resSs$log2FoldChange[is.na(resSs$log2FoldChange)] <- 0 #replace logfc NA to zero
#head(resS)
colnames(resSs) <- paste0(colnames(resSs),'Shrunk_',titleDE)
resAll <- resAll[1:(length(resAll)-3)]
resAll <- cbind(resAll, resSs)
resAll <- resAll[1:(length(resAll)-2)]
resAll <- cbind(resAll, resS)
```

```{r,fig.height = 10,fig.width = 14}
#volcano plots
resultsLFCs<-read.csv(paste0("DESeq_LFC_",titleDE,".csv"),header=TRUE)
resultsLFCs<-cbind(resultsLFCs,BMannot2)
resultsLFCs <- resultsLFCs[order(-1*resultsLFCs$pvalue),]#so significant dots on top of not significant

#pick significance thresholds
resultsLFCs = mutate(resultsLFCs, Significance=
                   ifelse(resultsLFCs$pvalue<0.05&resultsLFCs$log2FoldChange>1&!is.na(resultsLFCs$pvalue), "pvalue<0.05 & fold change>2", 
                          ifelse(resultsLFCs$pvalue<0.05&resultsLFCs$log2FoldChange<(-1)&!is.na(resultsLFCs$pvalue), "pvalue<0.05 & fold change<2",
                                 "Not Significant")))
resultsLFCs = mutate(resultsLFCs, SignificanceFDR=
                   ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange>1&!is.na(resultsLFCs$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange<(-1)&!is.na(resultsLFCs$pvalue), "padj<0.1 & fold change<2",
                                 "Not Significant")))
resultsLFCs = mutate(resultsLFCs, SignificanceFDR2=
                   ifelse(resultsLFCs$padj<0.1&!is.na(resultsLFCs$pvalue), "padj<0.1", 
                           "Not Significant"))
resultsLFCs<-resultsLFCs[complete.cases(resultsLFCs), ]#remove rows with NA
resultsLFCs=mutate(resultsLFCs,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
#volcano plots
xmax <- max(c(abs(min(resultsLFCs$log2FoldChange)),abs(max(resultsLFCs$log2FoldChange))))
p1= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Significance)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+2), y=2, label= "pvalue<0.05",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-280))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero

#volcano plot highlighting FDR
p2= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=SignificanceFDR)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=(-log10(resultsLFCs$pvalue[(match("padj<0.1", resultsLFCs[,"SignificanceFDR2"]))]))-0.01, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+1), y=-log10(resultsLFCs$pvalue[(match("padj<0.1", resultsLFCs[,"SignificanceFDR2"]))])+1, label= "padj<0.1",color="chartreuse4") +  
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero
#MA plots
p3= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Significance)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))
p4= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=SignificanceFDR)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

multiplot(p1, p3, p2,p4, cols=2)

```

Mouseover for gene names
```{r,fig.height = 10,fig.width = 12}
#xplotlyp1 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp2 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp3 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 14}
#volcano plots with bonus names
resultsLFCs<-read.csv(paste0("DESeq_LFC_",titleDE,".csv"),header=TRUE)
resultsLFCs<-cbind(resultsLFCs,BMannot2)
resultsLFCs <- resultsLFCs[order(-1*resultsLFCs$pvalue),]#so significant dots on top of not significant


resultsLFCs=resultsLFCs %>%  arrange(desc(Chosen))
resultsLFCs<-resultsLFCs[complete.cases(resultsLFCs), ]#remove rows with NA
resultsLFCs=mutate(resultsLFCs,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
resultsLFCs = mutate(resultsLFCs, SignificanceFDR=
                   ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange>1&!is.na(resultsLFCs$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange<(-1)&!is.na(resultsLFCs$pvalue), "padj<0.1 fold change<2",
                                 "Not Significant")))
resultsLFCs = mutate(resultsLFCs, Significance=
                   ifelse(resultsLFCs$pvalue<0.05&abs(resultsLFCs$log2FoldChange)>1&!is.na(resultsLFCs$pvalue), "sig", # NB choose cutoff 
                                 "Not Significant"))
xmax <- max(c(abs(min(resultsLFCs$log2FoldChange)),abs(max(resultsLFCs$log2FoldChange))))
p1= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(resultsLFCs$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero

#MA plots
p2= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

p3= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(resultsLFCs$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 5,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero

#MA plots
p4= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))


multiplot( p1, p2,p3,p4, cols=2)
```

```{r,fig.height = 12,fig.width = 20}
#heatmaps of top Genes
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
topDEgenes <- head(order(results$pvalue,results$padj),50)#find indexes of top 50 Genes
dataHM<-assay(logData)[ topDEgenes, ]
dataHM<- t(as.matrix(dataHM))
dataHM <- t(scale(dataHM))#scale (normalise) across Genes

data_pval<-results[ topDEgenes, ]
data_padj<-data_pval[,c(7)]
data_pval<-data_pval[,c(6)]
#data_pval
#head(data_pval)
pvalHm<-Heatmap(
  data_pval,  name = "pvalues",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(10, "mm"),
  col = rev(RColorBrewer::brewer.pal(name = "Greys", n = 5)),
  cluster_rows = FALSE,
  show_row_names = TRUE)

padjHm<-Heatmap(
  data_padj,  name = "padj",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(10, "mm"),
  col = rev(RColorBrewer::brewer.pal(name = "Greens", n = 5)),
  cluster_rows = FALSE,
  show_row_names = TRUE)
  
  
hmap_hier_factors1 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Top 50 DE Genes comparison ",titleDE," (ranked most significant at the top)"), #don't cluster rows for this one
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  cluster_rows = FALSE,
  show_row_names = TRUE,
  top_annotation=colAnn  )

hmap_hier_factors2 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Hierarchical clustering Top 50 DE Genes comparison ",titleDE), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  top_annotation=colAnn  )

#means
dataHMm<-baseMeans[ topDEgenes, ]
dataHMm <- log2(dataHMm+1)
dataHMm<- t(as.matrix(dataHMm))
dataHMm <- t(scale(dataHMm))

hmap_hier_factors3 <- Heatmap(
  dataHMm,  name = "Expression means",
column_title = paste0("means"), #don't cluster rows for this one
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
col = col_funF,
  cluster_rows = FALSE,
  show_row_names = TRUE,
top_annotation=colAnnm)

hmap_hier_factors4 <- Heatmap(
  dataHMm,  name = "Expression means",
  column_title = paste0("means"), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  top_annotation=colAnnm)

draw(hmap_hier_factors3+hmap_hier_factors1+pvalHm+padjHm, heatmap_legend_side = "left", annotation_legend_side = "left")
draw(hmap_hier_factors4+hmap_hier_factors2+pvalHm+padjHm, heatmap_legend_side = "left", annotation_legend_side = "left")

```




View selected Top Gene expression
```{r}
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
topDEgenesU <- head(order(results$pvalue,results$padj),1)#find index of top Gene

d <- plotCounts(dds, gene=as.vector(results$X[ topDEgenesU ]), intgroup="Sample_group", #intgroup is the factor we are using
                returnData=TRUE)

ggplot(d, aes(x=Sample_group, y=count)) + ggtitle(BMannot$Gene_Symbol[ topDEgenesU ]) +
  geom_point(position=position_jitter(w=0.1,h=0)) + 
  ylab("normalised count")+  scale_y_log10(breaks=c(25,100,400))
```


```{r}
#Lists of Gene Names for Venn and Functional Annotation on top 1000 Genes using clusterprofiler and enrichR.
#GO
topDEgenes <- head(order(results$pvalue,results$padj),1000)#find indexes of top 1000 Genes
assign(paste0("GOlist1000_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("Reactomelist1000_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
topDEgenes <- which(results$padj<0.1 & results$log2FoldChange>1 &!is.na(results$padj))#find indexes
assign(paste0("GOlistUp_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("ReactomeUplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
topDEgenes <- which(results$padj<0.1 & results$log2FoldChange<0-1 &!is.na(results$padj))#find indexes 
assign(paste0("GOlistDown_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("ReactomeDownlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
#Venn
topDEgenes <- which(results$pvalue<0.05&!is.na(results$pvalue))#find indexes 
assign(paste0("p0.05list_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$pvalue<0.05 & results$log2FoldChange>0&!is.na(results$pvalue))#find indexes  
assign(paste0("p0.05uplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$pvalue<0.05 & results$log2FoldChange<0&!is.na(results$pvalue))#find indexes 
assign(paste0("p0.05downlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1&!is.na(results$padj))#find indexes  
assign(paste0("padj0.1list_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1 & results$log2FoldChange>0 &!is.na(results$padj))#find indexes
assign(paste0("padj0.1uplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1 & results$log2FoldChange<0 &!is.na(results$padj))#find indexes 
assign(paste0("padj0.1downlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

```

```{r}
resIpa<-res[,c(2,5,6)]
topDEgenes <- head(order(resIpa$pvalue,resIpa$padj),1000)#find indexes of top 1000 Genes
resIpa$log2FoldChange[is.na(resIpa$log2FoldChange)] <- 0 #replace logfc NA to zero
resIpa$padj <-1
resIpa$padj[ topDEgenes ] <-resIpa$padj[ topDEgenes ]* 0
colnames(resIpa) <- paste0(colnames(resIpa),'_',titleDE)
ipa1000<-cbind(ipa1000,resIpa)
```

```{r}
resultsNames(dds)
```

`

### CAA3_vs_BAA2

```{r}
#relevel
dds$Sample_group <- relevel( dds$Sample_group, "BAA2" )
dds <- DESeq(dds)
resultsNames(dds)
```

```{r}
titleDE1<-'Sample_group_CAA3_vs_BAA2'
res<-results(dds,contrast=list(titleDE1))#,cooksCutoff=FALSE,independentFiltering=FALSE,alpha=0.05)
titleDE<-'TestC'
print(paste0("Summary of results for ",titleDE))
print("If outlier number is too high (e.g. many hundreds or thousands) then samples may need removing")
print("If outlier number is high (e.g. many thousands) then consider running cooksCutoff=FALSE and warning user")
#http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#approach-to-count-outliers
summary(res)
##str(res)
write.csv(res,paste0("DESeq_",titleDE,".csv"))
#save logfc, pvalue, padj from all comparisons to one data frame resAll
resS <- res[,c(2,5,6)]
resS$log2FoldChange[is.na(resS$log2FoldChange)] <- 0 #replace logfc NA to zero
head(resS)
colnames(resS) <- paste0(colnames(resS),'_',titleDE)
resAll <- cbind(resAll, resS)
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
DEcounter[titleDE] <- c((sum(results$pvalue<0.05,na.rm=TRUE)), 
                        (sum(results$pvalue<0.05 & results$log2FoldChange<0 , na.rm=TRUE)),
                        (sum(results$pvalue<0.05 & results$log2FoldChange>0 , na.rm=TRUE)),
                        (sum(results$padj<0.1,na.rm=TRUE)), 
                        (sum(results$padj<0.1 & results$log2FoldChange<0 , na.rm=TRUE)),
                        (sum(results$padj<0.1 & results$log2FoldChange>0 , na.rm=TRUE))
                        )

```


```{r}
#pvalue and padj histograms
#par(mfrow=c(1,2))
#metadata(res)$alpha # significance level
#metadata(res)$filterTheta
#metadata(res)$filterThreshold

plot(metadata(res)$filterNumRej,main=paste0("Independent Filtering ",titleDE),
     type="b", ylab="number of rejections",
     xlab="quantiles of baseMean")
lines(metadata(res)$lo.fit, col="red")
abline(v=metadata(res)$filterTheta)
```

```{r,fig.height = 6,fig.width = 12}
#pvalue and padj histograms
par(mfrow=c(1,2))
use <- res$baseMean > metadata(res)$filterThreshold
h1 <- hist(res$pvalue[!use], breaks=0:50/50, plot=FALSE)
h2 <- hist(res$pvalue[use], breaks=0:50/50, plot=FALSE)
colori <- c(`do not pass`="khaki", `pass`="powderblue")
barplot(height = rbind(h1$counts, h2$counts), beside = FALSE,
        col = colori, space = 0, main = "Effect of Independent Filtering on pvalues", ylab="frequency")
text(x = c(0, length(h1$counts)), y = 0, label = paste(c(0,1)),
     adj = c(0.5,1.7), xpd=NA)
legend("top", fill=rev(colori), legend=rev(names(colori)))
hist( res$padj, breaks=20, col="powderblue",main=paste0("Histogram of padj ",titleDE),xlab="" )
```


```{r,fig.height = 10,fig.width = 14}
#volcano plots
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
results<-cbind(results,BMannot2)
results <- results[order(-1*results$pvalue),]#so significant dots on top of not significant

#pick significance thresholds
results = mutate(results, Significance=
                   ifelse(results$pvalue<0.05&results$log2FoldChange>1&!is.na(results$pvalue), "pvalue<0.05 & fold change>2", 
                          ifelse(results$pvalue<0.05&results$log2FoldChange<(-1)&!is.na(results$pvalue), "pvalue<0.05 & fold change<2",
                                 "Not Significant")))
results = mutate(results, SignificanceFDR=
                   ifelse(results$padj<0.1&results$log2FoldChange>1&!is.na(results$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(results$padj<0.1&results$log2FoldChange<(-1)&!is.na(results$pvalue), "padj<0.1 & fold change<2",
                                 "Not Significant")))
results = mutate(results, SignificanceFDR2=
                   ifelse(results$padj<0.1&!is.na(results$pvalue), "padj<0.1", 
                           "Not Significant"))
results<-results[complete.cases(results), ]#remove rows with NA
results=mutate(results,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
#volcano plots
xmax <- max(c(abs(min(results$log2FoldChange)),abs(max(results$log2FoldChange))))
p1= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Significance)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+2), y=2, label= "pvalue<0.05",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-280))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#volcano plot highlighting FDR
p2= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=SignificanceFDR)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1", results[,"SignificanceFDR2"]))]))-0.01, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+1), y=-log10(results$pvalue[(match("padj<0.1", results[,"SignificanceFDR2"]))])+1, label= "padj<0.1",color="chartreuse4") +  
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero
#MA plots
p3= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Significance)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))
p4= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=SignificanceFDR)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

multiplot(p1, p3, p2,p4, cols=2)

```

Mouseover for gene names
```{r,fig.height = 10,fig.width = 12}
#xplotlyp1 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp2 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp3 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 14}
#volcano plots with bonus names
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
results<-cbind(results,BMannot2)
results <- results[order(-1*results$pvalue),]#so significant dots on top of not significant


results=results %>%  arrange(desc(Chosen))
results<-results[complete.cases(results), ]#remove rows with NA
results=mutate(results,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
results = mutate(results, SignificanceFDR=
                   ifelse(results$padj<0.1&results$log2FoldChange>1&!is.na(results$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(results$padj<0.1&results$log2FoldChange<(-1)&!is.na(results$pvalue), "padj<0.1 fold change<2",
                                 "Not Significant")))
results = mutate(results, Significance=
                   ifelse(results$pvalue<0.05&abs(results$log2FoldChange)>1&!is.na(results$pvalue), "sig", # NB choose cutoff 
                                 "Not Significant"))
xmax <- max(c(abs(min(results$log2FoldChange)),abs(max(results$log2FoldChange))))
p1= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(results$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#MA plots
p2= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

p3= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(results$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#MA plots
p4= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))


multiplot( p1, p2,p3,p4, cols=2)
```
Only showing gene names where pvalue<0.05 and logfc>1

```{r}
#shrunk LFC
resLFC<-lfcShrink(dds,coef=titleDE1, type="apeglm")#,cooksCutoff=FALSE,independentFiltering=FALSE,alpha=0.05)
write.csv(resLFC,paste0("DESeq_LFC_",titleDE,".csv"))
#save logfc, pvalue, padj from all comparisons to one data frame resAll
resSs <- resLFC[,c(2,4,5)]
resSs$log2FoldChange[is.na(resSs$log2FoldChange)] <- 0 #replace logfc NA to zero
#head(resS)
colnames(resSs) <- paste0(colnames(resSs),'Shrunk_',titleDE)
resAll <- resAll[1:(length(resAll)-3)]
resAll <- cbind(resAll, resSs)
resAll <- resAll[1:(length(resAll)-2)]
resAll <- cbind(resAll, resS)
```

```{r,fig.height = 10,fig.width = 14}
#volcano plots
resultsLFCs<-read.csv(paste0("DESeq_LFC_",titleDE,".csv"),header=TRUE)
resultsLFCs<-cbind(resultsLFCs,BMannot2)
resultsLFCs <- resultsLFCs[order(-1*resultsLFCs$pvalue),]#so significant dots on top of not significant

#pick significance thresholds
resultsLFCs = mutate(resultsLFCs, Significance=
                   ifelse(resultsLFCs$pvalue<0.05&resultsLFCs$log2FoldChange>1&!is.na(resultsLFCs$pvalue), "pvalue<0.05 & fold change>2", 
                          ifelse(resultsLFCs$pvalue<0.05&resultsLFCs$log2FoldChange<(-1)&!is.na(resultsLFCs$pvalue), "pvalue<0.05 & fold change<2",
                                 "Not Significant")))
resultsLFCs = mutate(resultsLFCs, SignificanceFDR=
                   ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange>1&!is.na(resultsLFCs$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange<(-1)&!is.na(resultsLFCs$pvalue), "padj<0.1 & fold change<2",
                                 "Not Significant")))
resultsLFCs = mutate(resultsLFCs, SignificanceFDR2=
                   ifelse(resultsLFCs$padj<0.1&!is.na(resultsLFCs$pvalue), "padj<0.1", 
                           "Not Significant"))
resultsLFCs<-resultsLFCs[complete.cases(resultsLFCs), ]#remove rows with NA
resultsLFCs=mutate(resultsLFCs,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
#volcano plots
xmax <- max(c(abs(min(resultsLFCs$log2FoldChange)),abs(max(resultsLFCs$log2FoldChange))))
p1= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Significance)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+2), y=2, label= "pvalue<0.05",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-280))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero

#volcano plot highlighting FDR
p2= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=SignificanceFDR)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=(-log10(resultsLFCs$pvalue[(match("padj<0.1", resultsLFCs[,"SignificanceFDR2"]))]))-0.01, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+1), y=-log10(resultsLFCs$pvalue[(match("padj<0.1", resultsLFCs[,"SignificanceFDR2"]))])+1, label= "padj<0.1",color="chartreuse4") +  
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero
#MA plots
p3= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Significance)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))
p4= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=SignificanceFDR)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

multiplot(p1, p3, p2,p4, cols=2)

```

Mouseover for gene names
```{r,fig.height = 10,fig.width = 12}
#xplotlyp1 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp2 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp3 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 14}
#volcano plots with bonus names
resultsLFCs<-read.csv(paste0("DESeq_LFC_",titleDE,".csv"),header=TRUE)
resultsLFCs<-cbind(resultsLFCs,BMannot2)
resultsLFCs <- resultsLFCs[order(-1*resultsLFCs$pvalue),]#so significant dots on top of not significant


resultsLFCs=resultsLFCs %>%  arrange(desc(Chosen))
resultsLFCs<-resultsLFCs[complete.cases(resultsLFCs), ]#remove rows with NA
resultsLFCs=mutate(resultsLFCs,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
resultsLFCs = mutate(resultsLFCs, SignificanceFDR=
                   ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange>1&!is.na(resultsLFCs$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange<(-1)&!is.na(resultsLFCs$pvalue), "padj<0.1 fold change<2",
                                 "Not Significant")))
resultsLFCs = mutate(resultsLFCs, Significance=
                   ifelse(resultsLFCs$pvalue<0.05&abs(resultsLFCs$log2FoldChange)>1&!is.na(resultsLFCs$pvalue), "sig", # NB choose cutoff 
                                 "Not Significant"))
xmax <- max(c(abs(min(resultsLFCs$log2FoldChange)),abs(max(resultsLFCs$log2FoldChange))))
p1= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(resultsLFCs$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero

#MA plots
p2= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

p3= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(resultsLFCs$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 5,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero

#MA plots
p4= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))


multiplot( p1, p2,p3,p4, cols=2)
```

```{r,fig.height = 12,fig.width = 20}
#heatmaps of top Genes
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
topDEgenes <- head(order(results$pvalue,results$padj),50)#find indexes of top 50 Genes
dataHM<-assay(logData)[ topDEgenes, ]
dataHM<- t(as.matrix(dataHM))
dataHM <- t(scale(dataHM))#scale (normalise) across Genes

data_pval<-results[ topDEgenes, ]
data_padj<-data_pval[,c(7)]
data_pval<-data_pval[,c(6)]
#data_pval
#head(data_pval)
pvalHm<-Heatmap(
  data_pval,  name = "pvalues",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(10, "mm"),
  col = rev(RColorBrewer::brewer.pal(name = "Greys", n = 5)),
  cluster_rows = FALSE,
  show_row_names = TRUE)

padjHm<-Heatmap(
  data_padj,  name = "padj",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(10, "mm"),
  col = rev(RColorBrewer::brewer.pal(name = "Greens", n = 5)),
  cluster_rows = FALSE,
  show_row_names = TRUE)
  
  
hmap_hier_factors1 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Top 50 DE Genes comparison ",titleDE," (ranked most significant at the top)"), #don't cluster rows for this one
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  cluster_rows = FALSE,
  show_row_names = TRUE,
  top_annotation=colAnn  )

hmap_hier_factors2 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Hierarchical clustering Top 50 DE Genes comparison ",titleDE), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  top_annotation=colAnn  )

#means
dataHMm<-baseMeans[ topDEgenes, ]
dataHMm <- log2(dataHMm+1)
dataHMm<- t(as.matrix(dataHMm))
dataHMm <- t(scale(dataHMm))

hmap_hier_factors3 <- Heatmap(
  dataHMm,  name = "Expression means",
column_title = paste0("means"), #don't cluster rows for this one
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
col = col_funF,
  cluster_rows = FALSE,
  show_row_names = TRUE,
top_annotation=colAnnm)

hmap_hier_factors4 <- Heatmap(
  dataHMm,  name = "Expression means",
  column_title = paste0("means"), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  top_annotation=colAnnm)

draw(hmap_hier_factors3+hmap_hier_factors1+pvalHm+padjHm, heatmap_legend_side = "left", annotation_legend_side = "left")
draw(hmap_hier_factors4+hmap_hier_factors2+pvalHm+padjHm, heatmap_legend_side = "left", annotation_legend_side = "left")

```




View selected Top Gene expression
```{r}
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
topDEgenesU <- head(order(results$pvalue,results$padj),1)#find index of top Gene

d <- plotCounts(dds, gene=as.vector(results$X[ topDEgenesU ]), intgroup="Sample_group", #intgroup is the factor we are using
                returnData=TRUE)

ggplot(d, aes(x=Sample_group, y=count)) + ggtitle(BMannot$Gene_Symbol[ topDEgenesU ]) +
  geom_point(position=position_jitter(w=0.1,h=0)) + 
  ylab("normalised count")+  scale_y_log10(breaks=c(25,100,400))
```


```{r}
#Lists of Gene Names for Venn and Functional Annotation on top 1000 Genes using clusterprofiler and enrichR.
#GO
topDEgenes <- head(order(results$pvalue,results$padj),1000)#find indexes of top 1000 Genes
assign(paste0("GOlist1000_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("Reactomelist1000_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
topDEgenes <- which(results$padj<0.1 & results$log2FoldChange>1 &!is.na(results$padj))#find indexes
assign(paste0("GOlistUp_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("ReactomeUplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
topDEgenes <- which(results$padj<0.1 & results$log2FoldChange<0-1 &!is.na(results$padj))#find indexes 
assign(paste0("GOlistDown_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("ReactomeDownlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
#Venn
topDEgenes <- which(results$pvalue<0.05&!is.na(results$pvalue))#find indexes 
assign(paste0("p0.05list_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$pvalue<0.05 & results$log2FoldChange>0&!is.na(results$pvalue))#find indexes  
assign(paste0("p0.05uplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$pvalue<0.05 & results$log2FoldChange<0&!is.na(results$pvalue))#find indexes 
assign(paste0("p0.05downlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1&!is.na(results$padj))#find indexes  
assign(paste0("padj0.1list_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1 & results$log2FoldChange>0 &!is.na(results$padj))#find indexes
assign(paste0("padj0.1uplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1 & results$log2FoldChange<0 &!is.na(results$padj))#find indexes 
assign(paste0("padj0.1downlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

```

```{r}
resIpa<-res[,c(2,5,6)]
topDEgenes <- head(order(resIpa$pvalue,resIpa$padj),1000)#find indexes of top 1000 Genes
resIpa$log2FoldChange[is.na(resIpa$log2FoldChange)] <- 0 #replace logfc NA to zero
resIpa$padj <-1
resIpa$padj[ topDEgenes ] <-resIpa$padj[ topDEgenes ]* 0
colnames(resIpa) <- paste0(colnames(resIpa),'_',titleDE)
ipa1000<-cbind(ipa1000,resIpa)
```

### CAA3_vs_AAA1
```{r}
resultsNames(dds)
```

```{r}
#relevel

dds$Sample_group <- relevel( dds$Sample_group, "AAA1" )
dds <- DESeq(dds)
resultsNames(dds)
```

```{r}
titleDE1<-'Sample_group_CAA3_vs_AAA1'
res<-results(dds,contrast=list(titleDE1))#,cooksCutoff=FALSE,independentFiltering=FALSE,alpha=0.05)
titleDE<-'TestD'
print(paste0("Summary of results for ",titleDE))
print("If outlier number is too high (e.g. many hundreds or thousands) then samples may need removing")
print("If outlier number is high (e.g. many thousands) then consider running cooksCutoff=FALSE and warning user")
#http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#approach-to-count-outliers
summary(res)
##str(res)
write.csv(res,paste0("DESeq_",titleDE,".csv"))
#save logfc, pvalue, padj from all comparisons to one data frame resAll
resS <- res[,c(2,5,6)]
resS$log2FoldChange[is.na(resS$log2FoldChange)] <- 0 #replace logfc NA to zero
head(resS)
colnames(resS) <- paste0(colnames(resS),'_',titleDE)
resAll <- cbind(resAll, resS)
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
DEcounter[titleDE] <- c((sum(results$pvalue<0.05,na.rm=TRUE)), 
                        (sum(results$pvalue<0.05 & results$log2FoldChange<0 , na.rm=TRUE)),
                        (sum(results$pvalue<0.05 & results$log2FoldChange>0 , na.rm=TRUE)),
                        (sum(results$padj<0.1,na.rm=TRUE)), 
                        (sum(results$padj<0.1 & results$log2FoldChange<0 , na.rm=TRUE)),
                        (sum(results$padj<0.1 & results$log2FoldChange>0 , na.rm=TRUE))
                        )

```

```{r}
#pvalue and padj histograms
#par(mfrow=c(1,2))
#metadata(res)$alpha # significance level
#metadata(res)$filterTheta
#metadata(res)$filterThreshold

plot(metadata(res)$filterNumRej,main=paste0("Independent Filtering ",titleDE),
     type="b", ylab="number of rejections",
     xlab="quantiles of baseMean")
lines(metadata(res)$lo.fit, col="red")
abline(v=metadata(res)$filterTheta)
```

```{r,fig.height = 6,fig.width = 12}
#pvalue and padj histograms
par(mfrow=c(1,2))
use <- res$baseMean > metadata(res)$filterThreshold
h1 <- hist(res$pvalue[!use], breaks=0:50/50, plot=FALSE)
h2 <- hist(res$pvalue[use], breaks=0:50/50, plot=FALSE)
colori <- c(`do not pass`="khaki", `pass`="powderblue")
barplot(height = rbind(h1$counts, h2$counts), beside = FALSE,
        col = colori, space = 0, main = "Effect of Independent Filtering on pvalues", ylab="frequency")
text(x = c(0, length(h1$counts)), y = 0, label = paste(c(0,1)),
     adj = c(0.5,1.7), xpd=NA)
legend("top", fill=rev(colori), legend=rev(names(colori)))
hist( res$padj, breaks=20, col="powderblue",main=paste0("Histogram of padj ",titleDE),xlab="" )
```


```{r,fig.height = 10,fig.width = 14}
#volcano plots
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
results<-cbind(results,BMannot2)
results <- results[order(-1*results$pvalue),]#so significant dots on top of not significant

#pick significance thresholds
results = mutate(results, Significance=
                   ifelse(results$pvalue<0.05&results$log2FoldChange>1&!is.na(results$pvalue), "pvalue<0.05 & fold change>2", 
                          ifelse(results$pvalue<0.05&results$log2FoldChange<(-1)&!is.na(results$pvalue), "pvalue<0.05 & fold change<2",
                                 "Not Significant")))
results = mutate(results, SignificanceFDR=
                   ifelse(results$padj<0.1&results$log2FoldChange>1&!is.na(results$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(results$padj<0.1&results$log2FoldChange<(-1)&!is.na(results$pvalue), "padj<0.1 & fold change<2",
                                 "Not Significant")))
results = mutate(results, SignificanceFDR2=
                   ifelse(results$padj<0.1&!is.na(results$pvalue), "padj<0.1", 
                           "Not Significant"))
results<-results[complete.cases(results), ]#remove rows with NA
results=mutate(results,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
#volcano plots
xmax <- max(c(abs(min(results$log2FoldChange)),abs(max(results$log2FoldChange))))
p1= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Significance)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+2), y=2, label= "pvalue<0.05",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-280))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#volcano plot highlighting FDR
p2= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=SignificanceFDR)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1", results[,"SignificanceFDR2"]))]))-0.01, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+1), y=-log10(results$pvalue[(match("padj<0.1", results[,"SignificanceFDR2"]))])+1, label= "padj<0.1",color="chartreuse4") +  
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero
#MA plots
p3= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Significance)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))
p4= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=SignificanceFDR)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

multiplot(p1, p3, p2,p4, cols=2)

```

Mouseover for gene names
```{r,fig.height = 10,fig.width = 12}
#xplotlyp1 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp2 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp3 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 14}
#volcano plots with bonus names
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
results<-cbind(results,BMannot2)
results <- results[order(-1*results$pvalue),]#so significant dots on top of not significant


results=results %>%  arrange(desc(Chosen))
results<-results[complete.cases(results), ]#remove rows with NA
results=mutate(results,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
results = mutate(results, SignificanceFDR=
                   ifelse(results$padj<0.1&results$log2FoldChange>1&!is.na(results$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(results$padj<0.1&results$log2FoldChange<(-1)&!is.na(results$pvalue), "padj<0.1 fold change<2",
                                 "Not Significant")))
results = mutate(results, Significance=
                   ifelse(results$pvalue<0.05&abs(results$log2FoldChange)>1&!is.na(results$pvalue), "sig", # NB choose cutoff 
                                 "Not Significant"))
xmax <- max(c(abs(min(results$log2FoldChange)),abs(max(results$log2FoldChange))))
p1= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(results$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#MA plots
p2= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

p3= ggplot(results, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(results$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#MA plots
p4= ggplot(results, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))


multiplot( p1, p2,p3,p4, cols=2)
```
Only showing gene names where pvalue<0.05 and logfc>1

```{r}
#shrunk LFC
resLFC<-lfcShrink(dds,coef=titleDE1, type="apeglm")#,cooksCutoff=FALSE,independentFiltering=FALSE,alpha=0.05)
write.csv(resLFC,paste0("DESeq_LFC_",titleDE,".csv"))
#save logfc, pvalue, padj from all comparisons to one data frame resAll
resSs <- resLFC[,c(2,4,5)]
resSs$log2FoldChange[is.na(resSs$log2FoldChange)] <- 0 #replace logfc NA to zero
#head(resS)
colnames(resSs) <- paste0(colnames(resSs),'Shrunk_',titleDE)
resAll <- resAll[1:(length(resAll)-3)]
resAll <- cbind(resAll, resSs)
resAll <- resAll[1:(length(resAll)-2)]
resAll <- cbind(resAll, resS)
```

```{r,fig.height = 10,fig.width = 14}
#volcano plots
resultsLFCs<-read.csv(paste0("DESeq_LFC_",titleDE,".csv"),header=TRUE)
resultsLFCs<-cbind(resultsLFCs,BMannot2)
resultsLFCs <- resultsLFCs[order(-1*resultsLFCs$pvalue),]#so significant dots on top of not significant

#pick significance thresholds
resultsLFCs = mutate(resultsLFCs, Significance=
                   ifelse(resultsLFCs$pvalue<0.05&resultsLFCs$log2FoldChange>1&!is.na(resultsLFCs$pvalue), "pvalue<0.05 & fold change>2", 
                          ifelse(resultsLFCs$pvalue<0.05&resultsLFCs$log2FoldChange<(-1)&!is.na(resultsLFCs$pvalue), "pvalue<0.05 & fold change<2",
                                 "Not Significant")))
resultsLFCs = mutate(resultsLFCs, SignificanceFDR=
                   ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange>1&!is.na(resultsLFCs$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange<(-1)&!is.na(resultsLFCs$pvalue), "padj<0.1 & fold change<2",
                                 "Not Significant")))
resultsLFCs = mutate(resultsLFCs, SignificanceFDR2=
                   ifelse(resultsLFCs$padj<0.1&!is.na(resultsLFCs$pvalue), "padj<0.1", 
                           "Not Significant"))
resultsLFCs<-resultsLFCs[complete.cases(resultsLFCs), ]#remove rows with NA
resultsLFCs=mutate(resultsLFCs,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
#volcano plots
xmax <- max(c(abs(min(resultsLFCs$log2FoldChange)),abs(max(resultsLFCs$log2FoldChange))))
p1= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Significance)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+2), y=2, label= "pvalue<0.05",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-280))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero

#volcano plot highlighting FDR
p2= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=SignificanceFDR)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=(-log10(resultsLFCs$pvalue[(match("padj<0.1", resultsLFCs[,"SignificanceFDR2"]))]))-0.01, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+1), y=-log10(resultsLFCs$pvalue[(match("padj<0.1", resultsLFCs[,"SignificanceFDR2"]))])+1, label= "padj<0.1",color="chartreuse4") +  
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero
#MA plots
p3= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Significance)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))
p4= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=SignificanceFDR)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","blue2","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

multiplot(p1, p3, p2,p4, cols=2)

```

Mouseover for gene names
```{r,fig.height = 10,fig.width = 12}
#xplotlyp1 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp2 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 12}
#xplotlyp3 %>%  ggplotly(tooltip = "text")
```

```{r,fig.height = 10,fig.width = 14}
#volcano plots with bonus names
resultsLFCs<-read.csv(paste0("DESeq_LFC_",titleDE,".csv"),header=TRUE)
resultsLFCs<-cbind(resultsLFCs,BMannot2)
resultsLFCs <- resultsLFCs[order(-1*resultsLFCs$pvalue),]#so significant dots on top of not significant


resultsLFCs=resultsLFCs %>%  arrange(desc(Chosen))
resultsLFCs<-resultsLFCs[complete.cases(resultsLFCs), ]#remove rows with NA
resultsLFCs=mutate(resultsLFCs,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
resultsLFCs = mutate(resultsLFCs, SignificanceFDR=
                   ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange>1&!is.na(resultsLFCs$pvalue), "padj<0.1 & fold change>2", 
                          ifelse(resultsLFCs$padj<0.1&resultsLFCs$log2FoldChange<(-1)&!is.na(resultsLFCs$pvalue), "padj<0.1 fold change<2",
                                 "Not Significant")))
resultsLFCs = mutate(resultsLFCs, Significance=
                   ifelse(resultsLFCs$pvalue<0.05&abs(resultsLFCs$log2FoldChange)>1&!is.na(resultsLFCs$pvalue), "sig", # NB choose cutoff 
                                 "Not Significant"))
xmax <- max(c(abs(min(resultsLFCs$log2FoldChange)),abs(max(resultsLFCs$log2FoldChange))))
p1= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(resultsLFCs$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero

#MA plots
p2= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

p3= ggplot(resultsLFCs, aes(log2FoldChange, -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(resultsLFCs$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(resultsLFCs$pvalue[(match("padj<0.1 & fold change>2", resultsLFCs[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 5,force = 5, max.overlaps = Inf)+
  xlim(-xmax,xmax)+ylim (0,-log10(min(resultsLFCs$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_LFCshrunk_",titleDE))##if a pvalue is zero

#MA plots
p4= ggplot(resultsLFCs, aes(log10(baseMean),log2FoldChange,text=Gene_Symbol) )+
  geom_point(aes(col=Chosen)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(resultsLFCs$log2FoldChange),max(resultsLFCs$log2FoldChange))+xlim (0,log10(max(resultsLFCs$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))


multiplot( p1, p2,p3,p4, cols=2)
```

```{r,fig.height = 12,fig.width = 20}
#heatmaps of top Genes
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
topDEgenes <- head(order(results$pvalue,results$padj),50)#find indexes of top 50 Genes
dataHM<-assay(logData)[ topDEgenes, ]
dataHM<- t(as.matrix(dataHM))
dataHM <- t(scale(dataHM))#scale (normalise) across Genes

data_pval<-results[ topDEgenes, ]
data_padj<-data_pval[,c(7)]
data_pval<-data_pval[,c(6)]
#data_pval
#head(data_pval)
pvalHm<-Heatmap(
  data_pval,  name = "pvalues",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(10, "mm"),
  col = rev(RColorBrewer::brewer.pal(name = "Greys", n = 5)),
  cluster_rows = FALSE,
  show_row_names = TRUE)

padjHm<-Heatmap(
  data_padj,  name = "padj",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(10, "mm"),
  col = rev(RColorBrewer::brewer.pal(name = "Greens", n = 5)),
  cluster_rows = FALSE,
  show_row_names = TRUE)
  
  
hmap_hier_factors1 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Top 50 DE Genes comparison ",titleDE," (ranked most significant at the top)"), #don't cluster rows for this one
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  cluster_rows = FALSE,
  show_row_names = TRUE,
  top_annotation=colAnn  )

hmap_hier_factors2 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Hierarchical clustering Top 50 DE Genes comparison ",titleDE), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  top_annotation=colAnn  )

#means
dataHMm<-baseMeans[ topDEgenes, ]
dataHMm <- log2(dataHMm+1)
dataHMm<- t(as.matrix(dataHMm))
dataHMm <- t(scale(dataHMm))

hmap_hier_factors3 <- Heatmap(
  dataHMm,  name = "Expression means",
column_title = paste0("means"), #don't cluster rows for this one
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
col = col_funF,
  cluster_rows = FALSE,
  show_row_names = TRUE,
top_annotation=colAnnm)

hmap_hier_factors4 <- Heatmap(
  dataHMm,  name = "Expression means",
  column_title = paste0("means"), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  top_annotation=colAnnm)

draw(hmap_hier_factors3+hmap_hier_factors1+pvalHm+padjHm, heatmap_legend_side = "left", annotation_legend_side = "left")
draw(hmap_hier_factors4+hmap_hier_factors2+pvalHm+padjHm, heatmap_legend_side = "left", annotation_legend_side = "left")

```




View selected Top Gene expression
```{r}
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
topDEgenesU <- head(order(results$pvalue,results$padj),1)#find index of top Gene

d <- plotCounts(dds, gene=as.vector(results$X[ topDEgenesU ]), intgroup="Sample_group", #intgroup is the factor we are using
                returnData=TRUE)

ggplot(d, aes(x=Sample_group, y=count)) + ggtitle(BMannot$Gene_Symbol[ topDEgenesU ]) +
  geom_point(position=position_jitter(w=0.1,h=0)) + 
  ylab("normalised count")+  scale_y_log10(breaks=c(25,100,400))
```


```{r}
#Lists of Gene Names for Venn and Functional Annotation on top 1000 Genes using clusterprofiler and enrichR.
#GO
topDEgenes <- head(order(results$pvalue,results$padj),1000)#find indexes of top 1000 Genes
assign(paste0("GOlist1000_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("Reactomelist1000_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
topDEgenes <- which(results$padj<0.1 & results$log2FoldChange>1 &!is.na(results$padj))#find indexes
assign(paste0("GOlistUp_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("ReactomeUplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
topDEgenes <- which(results$padj<0.1 & results$log2FoldChange<0-1 &!is.na(results$padj))#find indexes 
assign(paste0("GOlistDown_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("ReactomeDownlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
#Venn
topDEgenes <- which(results$pvalue<0.05&!is.na(results$pvalue))#find indexes 
assign(paste0("p0.05list_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$pvalue<0.05 & results$log2FoldChange>0&!is.na(results$pvalue))#find indexes  
assign(paste0("p0.05uplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$pvalue<0.05 & results$log2FoldChange<0&!is.na(results$pvalue))#find indexes 
assign(paste0("p0.05downlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1&!is.na(results$padj))#find indexes  
assign(paste0("padj0.1list_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1 & results$log2FoldChange>0 &!is.na(results$padj))#find indexes
assign(paste0("padj0.1uplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1 & results$log2FoldChange<0 &!is.na(results$padj))#find indexes 
assign(paste0("padj0.1downlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

```

```{r}
resIpa<-res[,c(2,5,6)]
topDEgenes <- head(order(resIpa$pvalue,resIpa$padj),1000)#find indexes of top 1000 Genes
resIpa$log2FoldChange[is.na(resIpa$log2FoldChange)] <- 0 #replace logfc NA to zero
resIpa$padj <-1
resIpa$padj[ topDEgenes ] <-resIpa$padj[ topDEgenes ]* 0
colnames(resIpa) <- paste0(colnames(resIpa),'_',titleDE)
ipa1000<-cbind(ipa1000,resIpa)
```

```{r}
resultsNames(dds)
```


### Interaction



```{r}
#Relevel

Sample_group<-factor(c(rep("Cntl",4),rep("AAA1",4),rep("BAA2",4),rep("CAA3",4)),
             levels=c("Cntl","BAA2","AAA1","CAA3"))#note names
Genotype<-factor(c(rep("WT",8),rep("KO",8)),
             levels=c("WT","KO"))#note names
Drug<-factor(c(rep("Untreated",4),rep("AZT",4),rep("Untreated",4),rep("AZT",4)),
             levels=c("Untreated","AZT"))#note names
des<-formula(~Genotype*Drug)
myNames<-colnames(countsTable)[1:16]#note length
colDataNames<-data.frame(row.names=myNames, Genotype=Genotype,Drug=Drug)
colDataNames
write.csv(colDataNames,"colDataNames.csv")
write.csv(colDataNames,"design.csv")
ddsHTSeq<-DESeqDataSetFromMatrix(countsTable, colData=colDataNames, design=des, ignoreRank = FALSE)


```


Run DESeq2 analysis
```{r}
dds<-DESeq(ddsHTSeq,betaPrior=FALSE)
```
```{r}
resultsNames(dds)
```

```{r}
titleDE1<-'GenotypeKO.DrugAZT'
res<-results(dds,contrast=list(titleDE1))#,cooksCutoff=FALSE,independentFiltering=FALSE,alpha=0.05)
titleDE<-'TestE'
print(paste0("Summary of results for ",titleDE))
print("If outlier number is too high (e.g. many hundreds or thousands) then samples may need removing")
print("If outlier number is high (e.g. many thousands) then consider running cooksCutoff=FALSE and warning user")
#http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#approach-to-count-outliers
summary(res)
##str(res)
write.csv(res,paste0("DESeq_",titleDE,".csv"))
#save logfc, pvalue, padj from all comparisons to one data frame resAll
resS <- res[,c(2,5,6)]
resS$log2FoldChange[is.na(resS$log2FoldChange)] <- 0 #replace logfc NA to zero
head(resS)
colnames(resS) <- paste0(colnames(resS),'_',titleDE)
resAll <- cbind(resAll, resS)
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
DEcounter[titleDE] <- c((sum(results$pvalue<0.05,na.rm=TRUE)), 
                        (sum(results$pvalue<0.05 & results$log2FoldChange<0 , na.rm=TRUE)),
                        (sum(results$pvalue<0.05 & results$log2FoldChange>0 , na.rm=TRUE)),
                        (sum(results$padj<0.1,na.rm=TRUE)), 
                        (sum(results$padj<0.1 & results$log2FoldChange<0 , na.rm=TRUE)),
                        (sum(results$padj<0.1 & results$log2FoldChange>0 , na.rm=TRUE))
                        )

```
Filter for low expressed genes in padj calculation i.e. Independent Filtering
```{r}
#par(mfrow=c(1,2))
#metadata(res)$alpha # significance level
#metadata(res)$filterTheta
#metadata(res)$filterThreshold

plot(metadata(res)$filterNumRej,main=paste0("Independent Filtering ",titleDE),
     type="b", ylab="number of rejections",
     xlab="quantiles of baseMean")
lines(metadata(res)$lo.fit, col="red")
abline(v=metadata(res)$filterTheta)
```

```{r,fig.height = 6,fig.width = 12}
#pvalue and padj histograms
par(mfrow=c(1,2))
use <- res$baseMean > metadata(res)$filterThreshold
h1 <- hist(res$pvalue[!use], breaks=0:50/50, plot=FALSE)
h2 <- hist(res$pvalue[use], breaks=0:50/50, plot=FALSE)
colori <- c(`do not pass`="khaki", `pass`="powderblue")
barplot(height = rbind(h1$counts, h2$counts), beside = FALSE,
        col = colori, space = 0, main = "Effect of Independent Filtering on pvalues", ylab="frequency")
text(x = c(0, length(h1$counts)), y = 0, label = paste(c(0,1)),
     adj = c(0.5,1.7), xpd=NA)
legend("top", fill=rev(colori), legend=rev(names(colori)))
hist( res$padj, breaks=20, col="powderblue",main=paste0("Histogram of padj ",titleDE),xlab="" )
```



```{r,fig.height = 10,fig.width = 14}
#volcano plots
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
results<-cbind(results,BMannot2)
results <- results[order(-1*results$pvalue),]#so significant dots on top of not significant

#pick significance thresholds
results = mutate(results, SignificanceFDR=
                   ifelse(results$padj<0.1&abs(results$log2FoldChange)>1&!is.na(results$pvalue), "padj<0.1 & fold change>+-2", 
                          "Not Significant"))
results = mutate(results, Significance=
                   ifelse(results$pvalue<0.05&abs(results$log2FoldChange)>1&!is.na(results$pvalue), "sig", # NB choose cutoff 
                                 "Not Significant"))

results = mutate(results, SignificanceFDR2=
                   ifelse(results$padj<0.1&!is.na(results$pvalue), "padj<0.1", 
                           "Not Significant"))
results<-results[complete.cases(results), ]#remove rows with NA
results=mutate(results,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
#volcano plots
xmax <- max(c(abs(min(results$log2FoldChange)),abs(max(results$log2FoldChange))))
p1= ggplot(results, aes(abs(log2FoldChange), -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Significance)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","red","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+2), y=2, label= "pvalue<0.05",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-280))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#volcano plot highlighting FDR
p2= ggplot(results, aes(abs(log2FoldChange), -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=SignificanceFDR)) +
  xlab("Fold change (log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("darkgrey","red"))+ 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1", results[,"SignificanceFDR2"]))]))-0.01, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+1), y=-log10(results$pvalue[(match("padj<0.1", results[,"SignificanceFDR2"]))])+1, label= "padj<0.1",color="chartreuse4") +  
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  xlim(-xmax,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero
#MA plots
p3= ggplot(results, aes(log10(baseMean),abs(log2FoldChange))) +
  geom_point(aes(col=Significance)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))
p4= ggplot(results, aes(log10(baseMean),abs(log2FoldChange))) +
  geom_point(aes(col=SignificanceFDR)) +
  ylab("Fold change (log2)")+
  xlab("mean of normalized counts (log2)")+
  scale_color_manual(values=c("darkgrey","red"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  ylim(min(results$log2FoldChange),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))

multiplot(p1, p3, p2,p4, cols=2)

```
Note that logfc here is ambiguous because an interaction uses 4 conditions


```{r,fig.height = 10,fig.width = 14}
#volcano plots with bonus names
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
results<-cbind(results,BMannot2)
results <- results[order(-1*results$pvalue),]#so significant dots on top of not significant


results=results %>%  arrange(desc(Chosen))
results<-results[complete.cases(results), ]#remove rows with NA
results=mutate(results,pvalue = ifelse(pvalue == 0, 1e-306, pvalue))#replace p=0 with a small value
results = mutate(results, SignificanceFDR=
                   ifelse(results$padj<0.1&abs(results$log2FoldChange)>1&!is.na(results$pvalue), "padj<0.1 & fold change>+-2", 
                          "Not Significant"))
results = mutate(results, Significance=
                   ifelse(results$pvalue<0.05&abs(results$log2FoldChange)>1&!is.na(results$pvalue), "sig", # NB choose cutoff 
                                 "Not Significant"))

xmax <- max(c(abs(min(results$log2FoldChange)),abs(max(results$log2FoldChange))))
p1= ggplot(results, aes(abs(log2FoldChange), -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change abs(log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(results$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  #geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(0,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

p3= ggplot(results, aes(abs(log2FoldChange), -log10(pvalue),text=Gene_Symbol)) +
  geom_point(aes(col=Chosen)) +
  xlab("Fold change abs(log2)")+
  ylab("P-value (-log10)")+
  scale_color_manual(values=c("red","darkgrey"))+#,"blue2","magenta2"))+ 
  geom_hline(yintercept=1.3, linetype="dashed", color = "grey35")+
  ##annotate("text", x=(min(results$log2FoldChange)+1), y=1, label= "pvalue=0.05",color="grey35") + 
  geom_hline(yintercept=(-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))]))-0, linetype="dashed", color = "chartreuse4")+
  #annotate("text", x=(-xmax+0.4), y=-log10(results$pvalue[(match("padj<0.1 & fold change>2", results[,"SignificanceFDR"]))])+5, label= "padj<0.1",color="chartreuse4") + 
  geom_vline(xintercept=1, linetype="dashed", color = "grey35")+
  #geom_vline(xintercept=-1, linetype="dashed", color = "grey35")+
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0,force = 5, max.overlaps = Inf)+
  xlim(0,xmax)+ylim (0,-log10(min(results$pvalue, na.rm = TRUE)))+ggtitle(paste0("Volcano Plot_",titleDE))
  #xlim(-xmax,xmax)+ylim (0,-log10(1e-290))+ggtitle(paste0("Volcano Plot_",titleDE))##if a pvalue is zero

#MA plots
p2= ggplot(results, aes(log10(baseMean),abs(log2FoldChange))) +
  geom_point(aes(col=Chosen)) +
  ylab("Fold change abs(log2)")+
  xlab("mean of normalized counts")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  #geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(0),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))


#MA plots
p4= ggplot(results, aes(log10(baseMean),abs(log2FoldChange))) +
  geom_point(aes(col=Chosen)) +
  ylab("Fold change abs(log2)")+
  xlab("mean of normalized counts")+
  scale_color_manual(values=c("red","darkgrey"))+ 
  geom_hline(yintercept=0, linetype="dashed", color = "grey35")+
  geom_hline(yintercept=1, linetype="dashed", color = "blue")+
  geom_hline(yintercept=-1, linetype="dashed", color = "blue")+
  #annotate("text", x=1, y=1.3, label= "logfc=1",color="blue") +
  theme(axis.title = element_text(colour="black", size = 12)) + 
  theme(axis.text = element_text(colour="black", size = 10)) + 
  theme(legend.title = element_text(colour="black", size = 12)) +
  theme(legend.text = element_text(colour="black", size = 10)) +
  #geom_text_repel(aes(label=ifelse(Chosen=="Chosen"&Significance=="sig",as.character(Gene_Symbol),'')),nudge_x = 0.5,nudge_y = 0.5,force = 5, max.overlaps = Inf)+
  ylim(min(0),max(results$log2FoldChange))+xlim (0,log10(max(results$baseMean, na.rm = TRUE)))+ggtitle(paste0("MA Plot_",titleDE))


multiplot( p1, p2,p3,p4, cols=2)
```
Only showing gene names where pvalue<0.05 and logfc>1

```{r,fig.height = 12,fig.width = 20}
#heatmaps of top Genes
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
topDEgenes <- head(order(results$pvalue,results$padj),50)#find indexes of top 50 Genes
dataHM<-assay(logData)[ topDEgenes, ]
dataHM<- t(as.matrix(dataHM))
dataHM <- t(scale(dataHM))#scale (normalise) across Genes

data_pval<-results[ topDEgenes, ]
data_padj<-data_pval[,c(7)]
data_pval<-data_pval[,c(6)]
#data_pval
#head(data_pval)
pvalHm<-Heatmap(
  data_pval,  name = "pvalues",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(10, "mm"),
  col = rev(RColorBrewer::brewer.pal(name = "Greys", n = 5)),
  cluster_rows = FALSE,
  show_row_names = TRUE)

padjHm<-Heatmap(
  data_padj,  name = "padj",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(10, "mm"),
  col = rev(RColorBrewer::brewer.pal(name = "Greens", n = 5)),
  cluster_rows = FALSE,
  show_row_names = TRUE)
  
  
hmap_hier_factors1 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Top 50 DE Genes comparison ",titleDE," (ranked most significant at the top)"), #don't cluster rows for this one
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  cluster_rows = FALSE,
  show_row_names = TRUE,
  top_annotation=colAnn  )

hmap_hier_factors2 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Hierarchical clustering Top 50 DE Genes comparison ",titleDE), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  top_annotation=colAnn  )

#means
dataHMm<-baseMeans[ topDEgenes, ]
dataHMm <- log2(dataHMm+1)
dataHMm<- t(as.matrix(dataHMm))
dataHMm <- t(scale(dataHMm))

hmap_hier_factors3 <- Heatmap(
  dataHMm,  name = "Expression means",
column_title = paste0("means"), #don't cluster rows for this one
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
col = col_funF,
  cluster_rows = FALSE,
  show_row_names = TRUE,
top_annotation=colAnnm)

hmap_hier_factors4 <- Heatmap(
  dataHMm,  name = "Expression means",
  column_title = paste0("means"), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  top_annotation=colAnnm)

draw(hmap_hier_factors3+hmap_hier_factors1+pvalHm+padjHm, heatmap_legend_side = "left", annotation_legend_side = "left")
draw(hmap_hier_factors4+hmap_hier_factors2+pvalHm+padjHm, heatmap_legend_side = "left", annotation_legend_side = "left")

```

```{r,fig.height = 12,fig.width = 20}
#heatmaps of top Genes
results<-read.csv(paste0("DESeq_",titleDE,".csv"),header=TRUE)
topDEgenes <- head(order(results$pvalue,results$padj),50)#find indexes of top 50 Genes
dataHM<-assay(logData)[ topDEgenes, ]
dataHM<- t(as.matrix(dataHM))
dataHM <- t(scale(dataHM))#scale (normalise) across Genes

data_pval<-results[ topDEgenes, ]
data_padj<-data_pval[,c(7)]
data_pval<-data_pval[,c(6)]
#data_pval
#head(data_pval)
pvalHm<-Heatmap(
  data_pval,  name = "pvalues",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(10, "mm"),
  col = rev(RColorBrewer::brewer.pal(name = "Greys", n = 5)),
  cluster_rows = FALSE,
  show_row_names = TRUE)

padjHm<-Heatmap(
  data_padj,  name = "padj",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(10, "mm"),
  col = rev(RColorBrewer::brewer.pal(name = "Greens", n = 5)),
  cluster_rows = FALSE,
  show_row_names = TRUE)
  
  
hmap_hier_factors1 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Top 50 DE Genes comparison ",titleDE," (ranked most significant at the top)"), #don't cluster rows for this one
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  cluster_rows = FALSE,
  cluster_columns = FALSE,
  show_row_names = TRUE,
  top_annotation=colAnn  )

hmap_hier_factors2 <- Heatmap(
  dataHM,  name = "Expression",
  row_labels = paste0((GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Hierarchical clustering Top 50 DE Genes comparison ",titleDE), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(300, "mm"),
  col = col_funF,
  cluster_columns = FALSE,
  show_row_names = TRUE,
  top_annotation=colAnn  )

#means
dataHMm<-baseMeans[ topDEgenes, ]
dataHMm <- log2(dataHMm+1)
dataHMm<- t(as.matrix(dataHMm))
dataHMm <- t(scale(dataHMm))

hmap_hier_factors3 <- Heatmap(
  dataHMm,  name = "Expression means",
column_title = paste0("means"), #don't cluster rows for this one
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
col = col_funF,
  cluster_rows = FALSE,
cluster_columns = FALSE,
  show_row_names = TRUE,
top_annotation=colAnnm)

hmap_hier_factors4 <- Heatmap(
  dataHMm,  name = "Expression means",
  column_title = paste0("means"), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
  col = col_funF,
  show_row_names = TRUE,
  cluster_columns = FALSE,
  top_annotation=colAnnm)

draw(hmap_hier_factors3+hmap_hier_factors1+pvalHm+padjHm, heatmap_legend_side = "left", annotation_legend_side = "left")
draw(hmap_hier_factors4+hmap_hier_factors2+pvalHm+padjHm, heatmap_legend_side = "left", annotation_legend_side = "left")

```

```{r}
#Lists of Gene Names for Venn and Functional Annotation on top 1000 Genes using clusterprofiler and enrichR.
#GO
#etc
topDEgenes <- head(order(results$pvalue,results$padj),1000)#find indexes of top 1000 Genes
assign(paste0("GOlist1000_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("Reactomelist1000_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
topDEgenes <- which(results$padj<0.1 & results$log2FoldChange>1 &!is.na(results$padj))#find indexes
assign(paste0("GOlistUp_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("ReactomeUplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
topDEgenes <- which(results$padj<0.1 & results$log2FoldChange<0-1 &!is.na(results$padj))#find indexes 
assign(paste0("GOlistDown_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol)
assign(paste0("ReactomeDownlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$entrez)
#Venn
topDEgenes <- which(results$pvalue<0.05&!is.na(results$pvalue))#find indexes 
assign(paste0("p0.05list_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$pvalue<0.05 & results$log2FoldChange>0&!is.na(results$pvalue))#find indexes  
assign(paste0("p0.05uplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$pvalue<0.05 & results$log2FoldChange<0&!is.na(results$pvalue))#find indexes 
assign(paste0("p0.05downlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1&!is.na(results$padj))#find indexes  
assign(paste0("padj0.1list_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1 & results$log2FoldChange>0 &!is.na(results$padj))#find indexes
assign(paste0("padj0.1uplist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

topDEgenes <- which(results$padj<0.1 & results$log2FoldChange<0 &!is.na(results$padj))#find indexes 
assign(paste0("padj0.1downlist_",titleDE),(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Ensembl1)

```

```{r}
resIpa<-res[,c(2,5,6)]
topDEgenes <- head(order(resIpa$pvalue,resIpa$padj),1000)#find indexes of top 1000 Genes
resIpa$log2FoldChange[is.na(resIpa$log2FoldChange)] <- 0 #replace logfc NA to zero
resIpa$padj <-1
resIpa$padj[ topDEgenes ] <-resIpa$padj[ topDEgenes ]* 0
colnames(resIpa) <- paste0(colnames(resIpa),'_',titleDE)
ipa1000<-cbind(ipa1000,resIpa)
```

```{r}
#etc

#other designs

#Interaction Paired
#Sample_group<-factor(c(rep("Cntl",3),rep("AAA1",3),rep("BAA2",3),rep("CAA3",3)),
#             levels=c("Cntl","BAA2","AAA1","CAA3"))#note names
#Genotype<-factor(c(rep("WT",6),rep("KO",6)),
#             levels=c("WT","KO"))#note names
#Drug<-factor(c(rep("Untreated",3),rep("AZT",3),rep("Untreated",3),rep("AZT",3)),
#             levels=c("Untreated","AZT"))#note names
#Batch<-factor(c("A","B","C","A","B","C","A","B","C","A","B","C"), levels=c("A","B","C"))
#des<-formula(~Batch+Genotype*Drug)
#myNames<-colnames(countsTable)[1:12]#note length
#colDataNames<-data.frame(row.names=myNames, Genotype=Genotype,Drug=Drug,Batch=Batch)
#colDataNames
#write.csv(colDataNames,"colDataNames.csv")
#write.csv(colDataNames,"design.csv")
#ddsHTSeq<-DESeqDataSetFromMatrix(countsTable, colData=colDataNames, design=des, ignoreRank = FALSE)

#LRT
#NumericFactor<-(c(14.6,21.9,17.8,9.9,9.3,8.7,8.4,8.4,6.6,3.5,4,4.4))
#myNames<-colnames(countsTable)[1:12]####note length
#colDataNames<-data.frame(row.names=myNames, NumericFactor=NumericFactor)
#colDataNames
#des<-formula(~NumericFactor)
#ddsHTSeq<-DESeqDataSetFromMatrix(countsTable, colData=colDataNames, design=des, ignoreRank = FALSE)
#dds<-DESeq(ddsHTSeq,test="LRT", reduced=~1)#,betaPrior=FALSE)
```



#Final Steps in series of contrasts

### Venn_diagrams

```{r,fig.height = 6,fig.width = 12}
DEcounter
write.csv(DEcounter,paste0("DE_stats.csv"))
#DEcounter <- data.frame(DEcounts)#this resets the dataframe

```

DE stats 
```{r,fig.height = 18,fig.width = 12}
###
DEcounterp<-DEcounter[-c(4:6),]
DEcounterq<-DEcounter[-c(1:3),]
DEcounterp <- melt(DEcounterp)  #library(reshape2) the function melt reshapes it from wide to long 
DEcounterq <- melt(DEcounterq)  
DEcounterp$rowid <- 1:3  #add a rowid identifying variable
DEcounterq$rowid <- 1:3  #add a rowid identifying variable

#line
ppl<-ggplot(DEcounterp, aes(variable, value, group=factor(rowid))) + 
  geom_line(aes(color=DEcounts))+
  scale_color_manual(values=c("blue","red",  "darkgreen"))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
qpl<-ggplot(DEcounterq, aes(variable, value, group=factor(rowid))) + 
  geom_line(aes(color=DEcounts))+ 
  scale_color_manual(values=c("darkgreen","blue","red"))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
#bars
ppb<-ggplot(DEcounterp, aes(variable, value, group=factor(rowid), fill=DEcounts)) + 
    geom_bar(position="dodge", stat="identity")+
  scale_fill_manual(values=c("blue","red",  "green"))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
qpb<-ggplot(DEcounterq, aes(variable, value, group=factor(rowid), fill=DEcounts)) + 
    geom_bar(position="dodge", stat="identity")+
  scale_fill_manual(values=c("green","blue","red"))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

#stacked bars (remove sum of up and down)
DEcounterps<-subset(DEcounterp, DEcounts!="pvalue<0.05")
DEcounterqs<-subset(DEcounterq, DEcounts!="padj<0.1")

ppsb<-ggplot(DEcounterps, aes(variable, value, group=factor(rowid), fill=DEcounts)) + 
    geom_bar(position="stack", stat="identity")+
  scale_fill_manual(values=c("blue","red"))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
qpsb<-ggplot(DEcounterqs, aes(variable, value, group=factor(rowid), fill=DEcounts)) + 
    geom_bar(position="stack", stat="identity")+
  scale_fill_manual(values=c("blue","red"))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

multiplot(ppl, ppb, ppsb,qpl,qpb, qpsb, cols=2)
```

```{r}
DEcounter
```


```{r}
path<-getwd()
dir.create(paste0(path,"/Venn"))
write.csv((DEcounter),paste0(path,"/Venn/","DEcounts.csv"))
```
Venn 2 sets
```{r,fig.height = 6,fig.width =18}
#pvalue
# STEP 1 Choose colour set based on number of sets in Venn Diagram


# STEP 2 likewise number of categoryNames
categoryNames<-c("V1","V2") 

# STEP 3 specify sets plot 1
vennList <- list(p0.05list_TestA,p0.05list_TestB)
vennTitle<-"p<0.05"
vennSamples<-"V1=TestA;V2=TestB"

# STEP 4 run
vennPlot1<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV2,  cat.col = colorsV2,cat.pos=0,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn2_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn2_",vennTitle,".csv"))

# STEP 3 specify sets plot 2
vennList <- list(p0.05uplist_TestA,p0.05uplist_TestB)
vennTitle<-"p<0.05up"

# STEP 4 run
vennPlot2<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV2,  cat.col = colorsV2,cat.pos=0,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn2_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn2_",vennTitle,".csv"))

# STEP 3 specify sets plot 3
vennList <- list(p0.05downlist_TestA,p0.05downlist_TestB)
vennTitle<-"p<0.05down"

# STEP 4 run
vennPlot3<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV2,  cat.col = colorsV2,cat.pos=0,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn2_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn2_",vennTitle,".csv"))

```


```{r,fig.height = 6,fig.width =18}
grid.arrange(gTree(children=vennPlot1), gTree(children=vennPlot2), gTree(children=vennPlot3), ncol=3, top="", bottom=vennSamples)
```


```{r,fig.height = 6,fig.width =18}
#Venn padj
# STEP 1 Choose colour set based on number of sets in Venn Diagram

# STEP 2 likewise number of categoryNames
categoryNames<-c("V1","V2") 

# STEP 3 specify sets plot 1
vennList <- list(padj0.1list_TestA,padj0.1list_TestB)
vennTitle<-"padj<0.1"
vennSamples<-"V1=TestA; V2=TestB"

# STEP 4 run
vennPlot1<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV2,  cat.col = colorsV2,cat.pos=0,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn2_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn2_",vennTitle,".csv"))

# STEP 3 specify sets plot 2
vennList <- list(padj0.1uplist_TestA,padj0.1uplist_TestB)
vennTitle<-"padj<0.1up"

# STEP 4 run
vennPlot2<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV2,  cat.col = colorsV2,cat.pos=0,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn2_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn2_",vennTitle,".csv"))

# STEP 3 specify sets plot 3
vennList <- list(padj0.1downlist_TestA,padj0.1downlist_TestB)
vennTitle<-"padj<0.1down"

# STEP 4 run
vennPlot3<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV2,  cat.col = colorsV2,cat.pos=0,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn2_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn2_",vennTitle,".csv"))

```


```{r,fig.height = 6,fig.width =18}
grid.arrange(gTree(children=vennPlot1), gTree(children=vennPlot2), gTree(children=vennPlot3), ncol=3, top="", bottom=vennSamples)
```



Venn 3 sets
```{r,fig.height = 6,fig.width =18}
#pvalue

# STEP 2 likewise number of categoryNames
categoryNames<-c("V1","V2","V3") 

# STEP 3 specify sets plot 1
vennList <- list(p0.05list_TestA,p0.05list_TestB,p0.05list_TestC)
vennTitle<-"p<0.05"
vennSamples<-"V1=TestA;V2=TestB;V3=TestC"

# STEP 4 run
vennPlot1<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV3,  cat.col = colorsV3,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn3_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn3_",vennTitle,".csv"))

# STEP 3 specify sets plot 2
vennList <- list(p0.05uplist_TestA,p0.05uplist_TestB,p0.05uplist_TestC)
vennTitle<-"p<0.05up"

# STEP 4 run
vennPlot2<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV3,  cat.col = colorsV3,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn3_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn3_",vennTitle,".csv"))

# STEP 3 specify sets plot 3
vennList <- list(p0.05downlist_TestA,p0.05downlist_TestB,p0.05downlist_TestC)
vennTitle<-"p<0.05down"

# STEP 4 run
vennPlot3<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV3,  cat.col = colorsV3,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn3_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn3_",vennTitle,".csv"))

```


```{r,fig.height = 6,fig.width =18}
grid.arrange(gTree(children=vennPlot1), gTree(children=vennPlot2), gTree(children=vennPlot3), ncol=3, top="", bottom=vennSamples)
```


```{r,fig.height = 6,fig.width =18}

# STEP 2 likewise number of categoryNames
categoryNames<-c("V1","V2","V3") 

# STEP 3 specify sets plot 1
vennList <- list(padj0.1list_TestA,padj0.1list_TestB,padj0.1list_TestC)
vennTitle<-"padj<0.1"
vennSamples<-"V1=TestA; V2=TestB; V3=TestC"

# STEP 4 run
vennPlot1<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV3,  cat.col = colorsV3,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn3_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn3_",vennTitle,".csv"))

# STEP 3 specify sets plot 2
vennList <- list(padj0.1uplist_TestA,padj0.1uplist_TestB,padj0.1uplist_TestC)
vennTitle<-"padj<0.1up"

# STEP 4 run
vennPlot2<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV3,  cat.col = colorsV3,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn3_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn3_",vennTitle,".csv"))

# STEP 3 specify sets plot 3
vennList <- list(padj0.1downlist_TestA,padj0.1downlist_TestB,padj0.1downlist_TestC)
vennTitle<-"padj<0.1down"

# STEP 4 run
vennPlot3<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV3,  cat.col = colorsV3,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn3_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn3_",vennTitle,".csv"))

```


```{r,fig.height = 6,fig.width =18}
grid.arrange(gTree(children=vennPlot1), gTree(children=vennPlot2), gTree(children=vennPlot3), ncol=3, top="", bottom=vennSamples)
```

Venn 4 sets

Venn pvalue
```{r,fig.height = 6,fig.width =18}

# STEP 2 likewise number of categoryNames
categoryNames<-c("V1","V2","V3","V4") 

# STEP 3 specify sets plot 1
vennList <- list(p0.05list_TestA,p0.05list_TestB,p0.05list_TestC,p0.05list_TestD)
vennTitle<-"p<0.05"
vennSamples<-"V1=TestA; V2=TestB; V3=TestC; V4=TestD"

# STEP 4 run
vennPlot1<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV4,  cat.col = colorsV4,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn4_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn4_",vennTitle,".csv"))

# STEP 3 specify sets plot 2
vennList <- list(p0.05uplist_TestA,p0.05uplist_TestB,p0.05uplist_TestC,p0.05uplist_TestD)
vennTitle<-"p<0.05up"

# STEP 4 run
vennPlot2<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV4,  cat.col = colorsV4,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn4_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn4_",vennTitle,".csv"))

# STEP 3 specify sets plot 3
vennList <- list(p0.05downlist_TestA,p0.05downlist_TestB,p0.05downlist_TestC,p0.05downlist_TestD)
vennTitle<-"p<0.05down"

# STEP 4 run
vennPlot3<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV4,  cat.col = colorsV4,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn4_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn4_",vennTitle,".csv"))

```


```{r,fig.height = 6,fig.width =18}
grid.arrange(gTree(children=vennPlot1), gTree(children=vennPlot2), gTree(children=vennPlot3), ncol=3, top="", bottom=vennSamples)
```


Venn padj
```{r,fig.height = 6,fig.width =18}
# STEP 1 Choose colour set based on number of sets in Venn Diagram

# STEP 2 likewise number of categoryNames
categoryNames<-c("V1","V2","V3","V4") 

# STEP 3 specify sets plot 1
vennList <- list(padj0.1list_TestA,padj0.1list_TestB,padj0.1list_TestC,padj0.1list_TestD)
vennTitle<-"padj<0.1"

# STEP 4 run
vennPlot1<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV4,  cat.col = colorsV4,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn4_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn4_",vennTitle,".csv"))

# STEP 3 specify sets plot 2
vennList <- list(padj0.1uplist_TestA,padj0.1uplist_TestB,padj0.1uplist_TestC,padj0.1uplist_TestD)
vennTitle<-"padj<0.1up"

# STEP 4 run
vennPlot2<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV4,  cat.col = colorsV4,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn4_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn4_",vennTitle,".csv"))

# STEP 3 specify sets plot 3
vennList <- list(padj0.1downlist_TestA,padj0.1downlist_TestB,padj0.1downlist_TestC,padj0.1downlist_TestD)
vennTitle<-"padj<0.1down"

# STEP 4 run
vennPlot3<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV4,  cat.col = colorsV4,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn4_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn4_",vennTitle,".csv"))

```


```{r,fig.height = 6,fig.width =18}
grid.arrange(gTree(children=vennPlot1), gTree(children=vennPlot2), gTree(children=vennPlot3), ncol=3, top="", bottom=vennSamples)
```


Venn 5 sets

Venn pvalue
```{r,fig.height = 6,fig.width =18}
# STEP 1 Choose colour set based on number of sets in Venn Diagram

# STEP 2 likewise number of categoryNames
categoryNames<-c("V1","V2","V3","V4","V5") 

# STEP 3 specify sets plot 1
vennList <- list(p0.05list_TestA,p0.05list_TestB,p0.05list_TestC,p0.05list_TestD,p0.05list_TestE)
vennTitle<-"p<0.05"
vennSamples<-"V1=TestA; V2=TestB; V3=TestC; V4=TestD; V5=TestE"

# STEP 4 run
vennPlot1<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV5,  cat.col = colorsV5,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn5_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn5_",vennTitle,".csv"))

# STEP 3 specify sets plot 2
vennList <- list(p0.05uplist_TestA,p0.05uplist_TestB,p0.05uplist_TestC,p0.05uplist_TestD,p0.05uplist_TestE)
vennTitle<-"p<0.05up"

# STEP 4 run
vennPlot2<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV5,  cat.col = colorsV5,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn5_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn5_",vennTitle,".csv"))

# STEP 3 specify sets plot 3
vennList <- list(p0.05downlist_TestA,p0.05downlist_TestB,p0.05downlist_TestC,p0.05downlist_TestD,p0.05downlist_TestE)
vennTitle<-"p<0.05down"

# STEP 4 run
vennPlot3<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV5,  cat.col = colorsV5,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn5_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn5_",vennTitle,".csv"))

```


```{r,fig.height = 6,fig.width =18}
grid.arrange(gTree(children=vennPlot1), gTree(children=vennPlot2), gTree(children=vennPlot3), ncol=3, top="", bottom=vennSamples)
```


Venn padj
```{r,fig.height = 6,fig.width =18}
# STEP 1 Choose colour set based on number of sets in Venn Diagram

# STEP 2 likewise number of categoryNames
categoryNames<-c("V1","V2","V3","V4","V5") 

# STEP 3 specify sets plot 1
vennList <- list(padj0.1list_TestA,padj0.1list_TestB,padj0.1list_TestC,padj0.1list_TestD,padj0.1list_TestE)
vennTitle<-"padj<0.1"
vennSamples<-"V1=TestA; V2=TestB; V3=TestC; V4=TestD; V5=TestE"

# STEP 4 run
vennPlot1<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV5,  cat.col = colorsV5,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn5_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn5_",vennTitle,".csv"))

# STEP 3 specify sets plot 2
vennList <- list(padj0.1uplist_TestA,padj0.1uplist_TestB,padj0.1uplist_TestC,padj0.1uplist_TestD,padj0.1uplist_TestE)
vennTitle<-"padj<0.1up"

# STEP 4 run
vennPlot2<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV5,  cat.col = colorsV5,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn5_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn5_",vennTitle,".csv"))

# STEP 3 specify sets plot 3
vennList <- list(padj0.1downlist_TestA,padj0.1downlist_TestB,padj0.1downlist_TestC,padj0.1downlist_TestD,padj0.1downlist_TestE)
vennTitle<-"padj<0.1down"

# STEP 4 run
vennPlot3<-venn.diagram(vennList , category.names = categoryNames, filename = NULL,output=FALSE, imagetype="png",main=vennTitle,  
                     scaled = FALSE, fill = colorsV5,  cat.col = colorsV5,
                     cat.cex = 1, cat.dist=0.19, margin = 0.15 #get positioning right
                    )
png(paste0(path,"/Venn/","Venn5_",vennTitle,".png"), width = 300, height = 300)
v.table <- venn(vennList)
dev.off()
venn.df<-(attributes(v.table)$intersections)
n.obs <- sapply(venn.df, length)
seq.max <- seq_len(max(n.obs))
mat <- (sapply(venn.df, "[", i = seq.max))
write.csv((mat),paste0(path,"/Venn/","venn5_",vennTitle,".csv"))

```


```{r,fig.height = 6,fig.width =18}
grid.arrange(gTree(children=vennPlot1), gTree(children=vennPlot2), gTree(children=vennPlot3), ncol=3, top="", bottom=vennSamples)
```




```{r}
#up and down
#listAll<-list(p0.05list_TestA,p0.05list_TestB,p0.05list_TestC,p0.05list_TestD,p0.05list_TestE)
#need to name the vectors in the list, example here is for 5 groups
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE")

#listAll<-list(p0.05list_TestA,p0.05list_TestB,p0.05list_TestC,p0.05list_TestD,p0.05list_TestE,p0.05list_TestF,p0.05list_TestG,p0.05list_TestH,p0.05list_TestI,p0.05list_TestJ,p0.05list_TestK,p0.05list_TestL,p0.05list_TestM,p0.05list_TestN,p0.05list_TestO,p0.05list_TestP,p0.05list_TestQ,p0.05list_TestR,p0.05list_TestS,p0.05list_TestT,p0.05list_TestU,p0.05list_TestV,p0.05list_TestW,p0.05list_TestX,p0.05list_TestY,p0.05list_TestZ)
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE","TestF","TestG", "TestH","TestI","TestJ", "TestK","TestL","TestM", "TestN","TestO","TestP", "TestQ","TestR","TestS","TestT","TestU","TestV","TestW","TestX","TestY","TestZ")

#overlaps<-fromList(listAll)
#write.csv(overlaps,paste0(path,"/Venn/","All_overlaps_p<0.05_up&down.csv"))

```

```{r}
#up
#listAll<-list(p0.05uplist_TestA,p0.05uplist_TestB,p0.05uplist_TestC,p0.05uplist_TestD,p0.05uplist_TestE)
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE")

#listAll<-list(p0.05uplist_TestA,p0.05uplist_TestB,p0.05uplist_TestC,p0.05uplist_TestD,p0.05uplist_TestE,p0.05uplist_TestF,p0.05uplist_TestG,p0.05uplist_TestH,p0.05uplist_TestI,p0.05uplist_TestJ,p0.05uplist_TestK,p0.05uplist_TestL,p0.05uplist_TestM,p0.05uplist_TestN,p0.05uplist_TestO,p0.05uplist_TestP,p0.05uplist_TestQ,p0.05uplist_TestR,p0.05uplist_TestS,p0.05uplist_TestT,p0.05uplist_TestU,p0.05uplist_TestV,p0.05uplist_TestW,p0.05uplist_TestX,p0.05uplist_TestY,p0.05uplist_TestZ)
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE","TestF","TestG", "TestH","TestI","TestJ", "TestK","TestL","TestM", "TestN","TestO","TestP", "TestQ","TestR","TestS","TestT","TestU","TestV","TestW","TestX","TestY","TestZ")

#overlaps<-fromList(listAll)
#write.csv(overlaps,paste0(path,"/Venn/","All_overlaps_p<0.05_up.csv"))
```

```{r}
#down
#listAll<-list(p0.05downlist_TestA,p0.05downlist_TestB,p0.05downlist_TestC,p0.05downlist_TestD,p0.05downlist_TestE)
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE")

#listAll<-list(p0.05downlist_TestA,p0.05downlist_TestB,p0.05downlist_TestC,p0.05downlist_TestD,p0.05downlist_TestE,p0.05downlist_TestF,p0.05downlist_TestG,p0.05downlist_TestH,p0.05downlist_TestI,p0.05downlist_TestJ,p0.05downlist_TestK,p0.05downlist_TestL,p0.05downlist_TestM,p0.05downlist_TestN,p0.05downlist_TestO,p0.05downlist_TestP,p0.05downlist_TestQ,p0.05downlist_TestR,p0.05downlist_TestS,p0.05downlist_TestT,p0.05downlist_TestU,p0.05downlist_TestV,p0.05downlist_TestW,p0.05downlist_TestX,p0.05downlist_TestY,p0.05downlist_TestZ)
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE","TestF","TestG", "TestH","TestI","TestJ", "TestK","TestL","TestM", "TestN","TestO","TestP", "TestQ","TestR","TestS","TestT","TestU","TestV","TestW","TestX","TestY","TestZ")

#overlaps<-fromList(listAll)
#write.csv(overlaps,paste0(path,"/Venn/","All_overlaps_p<0.05_down.csv"))
```



```{r}
#up and down
#listAll<-list(padj0.1list_TestA,padj0.1list_TestB,padj0.1list_TestC,padj0.1list_TestD,padj0.1list_TestE)
#need to name the vectors in the list, example here is for 5 groups
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE")

#listAll<-list(padj0.1list_TestA,padj0.1list_TestB,padj0.1list_TestC,padj0.1list_TestD,padj0.1list_TestE,padj0.1list_TestF,padj0.1list_TestG,padj0.1list_TestH,padj0.1list_TestI,padj0.1list_TestJ,padj0.1list_TestK,padj0.1list_TestL,padj0.1list_TestM,padj0.1list_TestN,padj0.1list_TestO,padj0.1list_TestP,padj0.1list_TestQ,padj0.1list_TestR,padj0.1list_TestS,padj0.1list_TestT,padj0.1list_TestU,padj0.1list_TestV,padj0.1list_TestW,padj0.1list_TestX,padj0.1list_TestY,padj0.1list_TestZ)
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE","TestF","TestG", "TestH","TestI","TestJ", "TestK","TestL","TestM", "TestN","TestO","TestP", "TestQ","TestR","TestS","TestT","TestU","TestV","TestW","TestX","TestY","TestZ")

#overlaps<-fromList(listAll)
#write.csv(overlaps,paste0(path,"/Venn/","All_overlaps_padj<0.1_up&down.csv"))

```

```{r}
#up
#listAll<-list(padj0.1uplist_TestA,padj0.1uplist_TestB,padj0.1uplist_TestC,padj0.1uplist_TestD,padj0.1uplist_TestE)
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE")

#listAll<-list(padj0.1uplist_TestA,padj0.1uplist_TestB,padj0.1uplist_TestC,padj0.1uplist_TestD,padj0.1uplist_TestE,padj0.1uplist_TestF,padj0.1uplist_TestG,padj0.1uplist_TestH,padj0.1uplist_TestI,padj0.1uplist_TestJ,padj0.1uplist_TestK,padj0.1uplist_TestL,padj0.1uplist_TestM,padj0.1uplist_TestN,padj0.1uplist_TestO,padj0.1uplist_TestP,padj0.1uplist_TestQ,padj0.1uplist_TestR,padj0.1uplist_TestS,padj0.1uplist_TestT,padj0.1uplist_TestU,padj0.1uplist_TestV,padj0.1uplist_TestW,padj0.1uplist_TestX,padj0.1uplist_TestY,padj0.1uplist_TestZ)
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE","TestF","TestG", "TestH","TestI","TestJ", "TestK","TestL","TestM", "TestN","TestO","TestP", "TestQ","TestR","TestS","TestT","TestU","TestV","TestW","TestX","TestY","TestZ")

#overlaps<-fromList(listAll)
#write.csv(overlaps,paste0(path,"/Venn/","All_overlaps_padj<0.1_up.csv"))
```

```{r}
#down
#listAll<-list(padj0.1downlist_TestA,padj0.1downlist_TestB,padj0.1downlist_TestC,padj0.1downlist_TestD,padj0.1downlist_TestE)
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE")

#listAll<-list(padj0.1downlist_TestA,padj0.1downlist_TestB,padj0.1downlist_TestC,padj0.1downlist_TestD,padj0.1downlist_TestE,padj0.1downlist_TestF,padj0.1downlist_TestG,padj0.1downlist_TestH,padj0.1downlist_TestI,padj0.1downlist_TestJ,padj0.1downlist_TestK,padj0.1downlist_TestL,padj0.1downlist_TestM,padj0.1downlist_TestN,padj0.1downlist_TestO,padj0.1downlist_TestP,padj0.1downlist_TestQ,padj0.1downlist_TestR,padj0.1downlist_TestS,padj0.1downlist_TestT,padj0.1downlist_TestU,padj0.1downlist_TestV,padj0.1downlist_TestW,padj0.1downlist_TestX,padj0.1downlist_TestY,padj0.1downlist_TestZ)
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE","TestF","TestG", "TestH","TestI","TestJ", "TestK","TestL","TestM", "TestN","TestO","TestP", "TestQ","TestR","TestS","TestT","TestU","TestV","TestW","TestX","TestY","TestZ")

#overlaps<-fromList(listAll)
#write.csv(overlaps,paste0(path,"/Venn/","All_overlaps_padj<0.1_down.csv"))
```

```{r,fig.height = 5,fig.width = 12}
#upset plots
#listInput <- list(TestA=padj0.1list_TestA,TestB=padj0.1list_TestB,TestC=padj0.1list_TestC,TestD=padj0.1list_TestD,TestE=padj0.1list_TestE)
#upset1<-Reduce(intersect, list(padj0.1list_TestA,padj0.1list_TestB,padj0.1list_TestC,padj0.1list_TestD,padj0.1list_TestE))
#write.csv(upset1,paste0(path,"/Venn/","padj0.1_intersect",groupsName,".csv"))
#upset(fromList(listInput), order.by = "freq",nsets = 5, sets.x.label = "padj<0.1", empty.intersections = "on" )
```

### GO_enrichments_top1000

Functional Annotation
```{r}
#If various groupings for functional annotation
groupsName<-"_clusterProfiler"
```


```{r}
path<-getwd()
dir.create(paste0(path,"/GO1000"))
write.csv((DEcounter),paste0(path,"/GO1000/","DEcounts.csv"))
```

Functional Annotation on top 1000 Genes using clusterprofiler. GO Analysis 





```{r}
#Vector containing all Sample_group names for GO analysis
#AllGeneNames<-row.names(read.delim("RNAseqJune2020_reanalysis_dupRemovedISGs_2Sym.txt", header = TRUE, sep = "\t",check.names=FALSE,row.names=1))
AllGeneNames<-BMannot$Gene_Symbol
```


```{r}

listAll<-list(GOlist1000_TestA,GOlist1000_TestB,GOlist1000_TestC,GOlist1000_TestD,GOlist1000_TestE)
#need to name the vectors in the list, example here is for 5 groups
names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE")#make sure order the same as above!!!

#listAll<-list(GOlist1000_TestA,GOlist1000_TestB,GOlist1000_TestC,GOlist1000_TestD,GOlist1000_TestE,GOlist1000_TestF,GOlist1000_TestG,GOlist1000_TestH,GOlist1000_TestI,GOlist1000_TestJ,GOlist1000_TestK,GOlist1000_TestL,GOlist1000_TestM,GOlist1000_TestN,GOlist1000_TestO,GOlist1000_TestP,GOlist1000_TestQ,GOlist1000_TestR,GOlist1000_TestS,GOlist1000_TestT,GOlist1000_TestU,GOlist1000_TestV,GOlist1000_TestW,GOlist1000_TestX,GOlist1000_TestY,GOlist1000_TestZ)
#names(listAll)<-c("TestA", "TestB","TestC","TestD", "TestE","TestF","TestG", "TestH","TestI","TestJ", "TestK","TestL","TestM", "TestN","TestO","TestP", "TestQ","TestR","TestS","TestT","TestU","TestV","TestW","TestX","TestY","TestZ")#make sure order the same as above!!!

#if you want to rearrange the order
#listAll<-listAll[c("x3", "x7", "x8", "x2", "x6", "x5", "x4", "x1")]

lapply(listAll, head)

```

```{r}

listAllR<-list(Reactomelist1000_TestA,Reactomelist1000_TestB,Reactomelist1000_TestC,Reactomelist1000_TestD,Reactomelist1000_TestE)
#need to name the vectors in the list, example here is for 5 groups
names(listAllR)<-c("TestA", "TestB","TestC","TestD", "TestE")#make sure order the same as above!!!

#listAllR<-list(Reactomelist1000_TestA,Reactomelist1000_TestB,Reactomelist1000_TestC,Reactomelist1000_TestD,Reactomelist1000_TestE,Reactomelist1000_TestF,Reactomelist1000_TestG,Reactomelist1000_TestH,Reactomelist1000_TestI,Reactomelist1000_TestJ,Reactomelist1000_TestK,Reactomelist1000_TestL,Reactomelist1000_TestM,Reactomelist1000_TestN,Reactomelist1000_TestO,Reactomelist1000_TestP,Reactomelist1000_TestQ,Reactomelist1000_TestR,Reactomelist1000_TestS,Reactomelist1000_TestT,Reactomelist1000_TestU,Reactomelist1000_TestV,Reactomelist1000_TestW,Reactomelist1000_TestX,Reactomelist1000_TestY,Reactomelist1000_TestZ)
#names(listAllR)<-c("TestA", "TestB","TestC","TestD", "TestE","TestF","TestG", "TestH","TestI","TestJ", "TestK","TestL","TestM", "TestN","TestO","TestP", "TestQ","TestR","TestS","TestT","TestU","TestV","TestW","TestX","TestY","TestZ")#make sure order the same as above!!!

lapply(listAllR, head)

```

```{r}
res <- compareCluster(listAllR, fun="enrichPathway",universe = BMannot$entrez,
                   organism="human",
                   pvalueCutoff=0.05, 
                   qvalueCutoff = 0.10,
                   readable=T)

head(res)
write.csv(as.data.frame(res),paste0(path,"/GO1000/","Reactome",groupsName,".csv"))
```

```{r,fig.height = 8,fig.width = 10}
plotR <-dotplot(res,showCategory = 8
               ,
        title = paste0("Reactome Enrichment ",groupsName))+
   scale_y_discrete(labels=function(x) str_wrap(x, width=80))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
plotR
```

```{r}
pdf(paste0(path,"/GO1000/","Reactome1.pdf"), width = 10, height = 10)
plotR
dev.off()
```

- CC cellular compartment
- BP biological process
- MF molecular function

```{r}
#Could lump everything together by not specifying "ont".
#The simplify function has been used to cut down on GO redundancy
#CC
cgoCC <- compareCluster(geneCluster = listAll, 
                      universe = AllGeneNames,
                      fun = "enrichGO",
                      OrgDb=org.Hs.eg.db, 
                      #OrgDb=org.Mm.eg.db,
                      keyType="SYMBOL",
                      ont = "CC", 
                      pvalueCutoff=0.05,
                      qvalueCutoff = 0.10)
cgoCC2 <- clusterProfiler::simplify(cgoCC, cutoff=0.7, by="p.adjust", select_fun=min)
#write as spreadsheet
write.csv(as.data.frame(cgoCC2),paste0(path,"/GO1000/","GO_CC",groupsName,".csv"))



```

```{r}
#CC
png("dummy")
plotCC=dotplot(cgoCC2,showCategory = 30,
        title = paste0("GO Cellular Compartment ",groupsName))+
   scale_y_discrete(labels=function(x) str_wrap(x, width=80))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
dev.off()
```
```{r,fig.height = 24,fig.width = 12}
plotCC
```

```{r}
png(paste0(path,"/GO1000/","GO_CC",groupsName,".png"), width = 1224, height = 2024)
plotCC
dev.off()
```

GO BP
```{r}
#BP
cgoBP <- compareCluster(geneCluster = listAll, 
                      universe = AllGeneNames,
                      fun = "enrichGO",
                      OrgDb=org.Hs.eg.db, 
                      keyType="SYMBOL",
                      ont = "BP", 
                      pvalueCutoff=0.05,
                      qvalueCutoff = 0.10)
cgoBP2 <- clusterProfiler::simplify(cgoBP, cutoff=0.7, by="p.adjust", select_fun=min)
#write as spreadsheet
write.csv(as.data.frame(cgoBP2),paste0(path,"/GO1000/","GO_BP",groupsName,".csv"))



```

```{r}
#CC
png("dummy")
plotBP=dotplot(cgoBP2,showCategory = 30,
        title = paste0("GO Biological Process ",groupsName))+
   scale_y_discrete(labels=function(x) str_wrap(x, width=80))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
dev.off()
```

```{r,fig.height = 42,fig.width = 12}
plotBP

```

```{r}
png(paste0(path,"/GO1000/","GO_BP",groupsName,".png"), width = 1024, height = 4024)
plotBP
dev.off()
```


GO MF

```{r}
#MF
cgoMF <- compareCluster(geneCluster = listAll, 
                      universe = AllGeneNames,
                      fun = "enrichGO",
                      OrgDb=org.Hs.eg.db, 
                      keyType="SYMBOL",
                      ont = "MF", 
                      pvalueCutoff=0.05,
                      qvalueCutoff = 0.10)
cgoMF2 <- clusterProfiler::simplify(cgoMF, cutoff=0.7, by="p.adjust", select_fun=min)
#write as spreadsheet
write.csv(as.data.frame(cgoMF2),paste0(path,"/GO1000/","GO_MF",groupsName,".csv"))


```

```{r}
#MF
png("dummy")
plotMF=dotplot(cgoMF2,showCategory = 30,
        title = paste0("GO Molecular Function  ",groupsName))+
   scale_y_discrete(labels=function(x) str_wrap(x, width=80))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
dev.off()
```

```{r,fig.height = 30,fig.width = 12}
plotMF

```


```{r}
#print plot to file
png(paste0(path,"/GO1000/","GO_MF",groupsName,".png"), width = 1424, height = 2024)
plotMF
dev.off()
```

```{r}
#If various groupings for functional annotation
groupsName<-"_Enrichr"
```

```{r}
#Enrichr
setEnrichrSite("Enrichr") # Human genes
```
```{r}
websiteLive <- TRUE
dbs <- listEnrichrDbs()
if (is.null(dbs)) websiteLive <- FALSE
if (websiteLive) head(dbs)
```

```{r,fig.height = 7,fig.width = 12}
#dbs <- c("GO_Molecular_Function_2018", "GO_Cellular_Component_2018", "GO_Biological_Process_2018")


dbs <- c("Reactome_Pathways_2024","Jensen_DISEASES_Curated_2025", "GO_Biological_Process_2025")
#dbs <- c("Reactome_Pathways_2024","KEGG_2019_Mouse", "GO_Biological_Process_2025")

if (websiteLive) {    enrichedA <- enrichr(GOlist1000_TestA, dbs)}
if (websiteLive) plotEnrich(enrichedA[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestA")
write.csv(head(enrichedA[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestA",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestA")
write.csv(head(enrichedA[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestA",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestA")
write.csv(head(enrichedA[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestA",groupsName,".csv"))

if (websiteLive) {    enrichedB <- enrichr(GOlist1000_TestB, dbs)}
if (websiteLive) plotEnrich(enrichedB[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestB")
write.csv(head(enrichedB[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestB",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestB")
write.csv(head(enrichedB[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestB",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestB")
write.csv(head(enrichedB[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestB",groupsName,".csv"))

if (websiteLive) {    enrichedC <- enrichr(GOlist1000_TestC, dbs)}
if (websiteLive) plotEnrich(enrichedC[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestC")
write.csv(head(enrichedC[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestC",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestC")
write.csv(head(enrichedC[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestC",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestC")
write.csv(head(enrichedC[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestC",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlist1000_TestD, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestD")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestD",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestD")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestD",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestD")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestD",groupsName,".csv"))

if (websiteLive) {    enrichedE <- enrichr(GOlist1000_TestE, dbs)}
if (websiteLive) plotEnrich(enrichedE[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestE")
write.csv(head(enrichedE[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestE",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedE[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestE")
write.csv(head(enrichedE[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestE",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedE[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestE")
write.csv(head(enrichedE[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestE",groupsName,".csv"))

```

```{r,fig.height = 7,fig.width = 12}
dbs <- c("CellMarker_2024","Descartes_Cell_Types_and_Tissue_2021", "RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO")

if (websiteLive) {    enrichedA <- enrichr(GOlist1000_TestA, dbs)}
if (websiteLive) plotEnrich(enrichedA[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis TestA")
write.csv(head(enrichedA[[1]], 50),paste0(path,"/GO1000/","CellMarker_2024_Enrichment_Analysis_TestA",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis TestA")
write.csv(head(enrichedA[[2]], 50),paste0(path,"/GO1000/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestA",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis TestA")
write.csv(head(enrichedA[[3]], 50),paste0(path,"/GO1000/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestA",groupsName,".csv"))

if (websiteLive) {    enrichedB <- enrichr(GOlist1000_TestB, dbs)}
if (websiteLive) plotEnrich(enrichedB[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis TestB")
write.csv(head(enrichedB[[1]], 50),paste0(path,"/GO1000/","CellMarker_2024_Enrichment_Analysis_TestB",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis TestB")
write.csv(head(enrichedB[[2]], 50),paste0(path,"/GO1000/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestB",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis TestB")
write.csv(head(enrichedB[[3]], 50),paste0(path,"/GO1000/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestB",groupsName,".csv"))

if (websiteLive) {    enrichedC <- enrichr(GOlist1000_TestC, dbs)}
if (websiteLive) plotEnrich(enrichedC[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis TestC")
write.csv(head(enrichedC[[1]], 50),paste0(path,"/GO1000/","CellMarker_2024_Enrichment_Analysis_TestC",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis TestC")
write.csv(head(enrichedC[[2]], 50),paste0(path,"/GO1000/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestC",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis TestC")
write.csv(head(enrichedC[[3]], 50),paste0(path,"/GO1000/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestC",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlist1000_TestD, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis TestD")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GO1000/","CellMarker_2024_Enrichment_Analysis_TestD",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis TestD")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GO1000/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestD",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis TestD")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GO1000/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestD",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlist1000_TestE, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis TestE")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GO1000/","CellMarker_2024_Enrichment_Analysis_TestE",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis TestE")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GO1000/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestE",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis TestE")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GO1000/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestE",groupsName,".csv"))
```

```{r,fig.height = 7,fig.width = 12}
#Rare Diseases
dbs <- c("Rare_Diseases_GeneRIF_ARCHS4_Predictions","Rare_Diseases_GeneRIF_Gene_Lists", "Rare_Diseases_AutoRIF_ARCHS4_Predictions", "Rare_Diseases_AutoRIF_Gene_Lists")


if (websiteLive) {    enrichedA <- enrichr(GOlist1000_TestA, dbs)}
if (websiteLive) plotEnrich(enrichedA[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis TestA")
write.csv(head(enrichedA[[1]], 50),paste0(path,"/GO1000/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestA",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis TestA")
write.csv(head(enrichedA[[2]], 50),paste0(path,"/GO1000/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestA",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis TestA")
write.csv(head(enrichedA[[3]], 50),paste0(path,"/GO1000/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestA",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis TestA")
write.csv(head(enrichedA[[4]], 50),paste0(path,"/GO1000/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestA",groupsName,".csv"))

if (websiteLive) {    enrichedB <- enrichr(GOlist1000_TestB, dbs)}
if (websiteLive) plotEnrich(enrichedB[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis TestB")
write.csv(head(enrichedB[[1]], 50),paste0(path,"/GO1000/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestB",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis TestB")
write.csv(head(enrichedB[[2]], 50),paste0(path,"/GO1000/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestB",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis TestB")
write.csv(head(enrichedB[[3]], 50),paste0(path,"/GO1000/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestB",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis TestB")
write.csv(head(enrichedB[[4]], 50),paste0(path,"/GO1000/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestB",groupsName,".csv"))

if (websiteLive) {    enrichedC <- enrichr(GOlist1000_TestC, dbs)}
if (websiteLive) plotEnrich(enrichedC[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis TestC")
write.csv(head(enrichedC[[1]], 50),paste0(path,"/GO1000/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestC",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis TestC")
write.csv(head(enrichedC[[2]], 50),paste0(path,"/GO1000/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestC",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis TestC")
write.csv(head(enrichedC[[3]], 50),paste0(path,"/GO1000/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestC",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis TestC")
write.csv(head(enrichedC[[4]], 50),paste0(path,"/GO1000/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestC",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlist1000_TestD, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis TestD")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GO1000/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestD",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis TestD")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GO1000/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestD",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis TestD")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GO1000/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestD",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis TestD")
write.csv(head(enrichedD[[4]], 50),paste0(path,"/GO1000/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestD",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlist1000_TestE, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis TestE")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GO1000/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestE",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis TestE")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GO1000/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestE",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis TestE")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GO1000/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestE",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis TestE")
write.csv(head(enrichedD[[4]], 50),paste0(path,"/GO1000/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestE",groupsName,".csv"))

```

```{r,fig.height = 7,fig.width = 12}

#if (websiteLive) {    enrichedF <- enrichr(GOlist1000_TestF, dbs)}
#if (websiteLive) plotEnrich(enrichedF[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestF")
#write.csv(head(enrichedF[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestF",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedF[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestF")
#write.csv(head(enrichedF[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestF",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedF[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestF")
#write.csv(head(enrichedF[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestF",groupsName,".csv"))

#if (websiteLive) {    enrichedG <- enrichr(GOlist1000_TestG, dbs)}
#if (websiteLive) plotEnrich(enrichedG[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestG")
#write.csv(head(enrichedG[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestG",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedG[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestG")
#write.csv(head(enrichedG[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestG",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedG[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestG")
#write.csv(head(enrichedG[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestG",groupsName,".csv"))

#if (websiteLive) {    enrichedH <- enrichr(GOlist1000_TestH, dbs)}
#if (websiteLive) plotEnrich(enrichedH[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestH")
#write.csv(head(enrichedH[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestH",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedH[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestH")
#write.csv(head(enrichedH[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestH",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedH[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestH")
#write.csv(head(enrichedH[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestH",groupsName,".csv"))

#if (websiteLive) {    enrichedI <- enrichr(GOlist1000_TestI, dbs)}
#if (websiteLive) plotEnrich(enrichedI[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestI")
#write.csv(head(enrichedI[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestI",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedI[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestI")
#write.csv(head(enrichedI[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestI",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedI[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestI")
#write.csv(head(enrichedI[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestI",groupsName,".csv"))

#if (websiteLive) {    enrichedJ <- enrichr(GOlist1000_TestJ, dbs)}
#if (websiteLive) plotEnrich(enrichedJ[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestJ")
#write.csv(head(enrichedJ[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestJ",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedJ[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestJ")
#write.csv(head(enrichedJ[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestJ",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedJ[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestJ")
#write.csv(head(enrichedJ[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestJ",groupsName,".csv"))

#if (websiteLive) {    enrichedK <- enrichr(GOlist1000_TestK, dbs)}
#if (websiteLive) plotEnrich(enrichedK[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestK")
#write.csv(head(enrichedK[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestK",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedK[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestK")
#write.csv(head(enrichedK[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestK",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedK[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestK")
#write.csv(head(enrichedK[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestK",groupsName,".csv"))

#if (websiteLive) {    enrichedL <- enrichr(GOlist1000_TestL, dbs)}
#if (websiteLive) plotEnrich(enrichedL[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestL")
#write.csv(head(enrichedL[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestL",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedL[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestL")
#write.csv(head(enrichedL[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestL",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedL[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestL")
#write.csv(head(enrichedL[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestL",groupsName,".csv"))

#if (websiteLive) {    enrichedM <- enrichr(GOlist1000_TestM, dbs)}
#if (websiteLive) plotEnrich(enrichedM[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestM")
#write.csv(head(enrichedM[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestM",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedM[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestM")
#write.csv(head(enrichedM[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestM",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedM[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestM")
#write.csv(head(enrichedM[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestM",groupsName,".csv"))

#if (websiteLive) {    enrichedN <- enrichr(GOlist1000_TestN, dbs)}
#if (websiteLive) plotEnrich(enrichedN[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestN")
#write.csv(head(enrichedN[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestN",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedN[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestN")
#write.csv(head(enrichedN[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestN",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedN[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestN")
#write.csv(head(enrichedN[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestN",groupsName,".csv"))

#if (websiteLive) {    enrichedO <- enrichr(GOlist1000_TestO, dbs)}
#if (websiteLive) plotEnrich(enrichedO[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestO")
#write.csv(head(enrichedO[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestO",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedO[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestO")
#write.csv(head(enrichedO[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestO",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedO[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestO")
#write.csv(head(enrichedO[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestO",groupsName,".csv"))

#if (websiteLive) {    enrichedP <- enrichr(GOlist1000_TestP, dbs)}
#if (websiteLive) plotEnrich(enrichedP[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestP")
#write.csv(head(enrichedP[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestP",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedP[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestP")
#write.csv(head(enrichedP[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestP",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedP[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestP")
#write.csv(head(enrichedP[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestP",groupsName,".csv"))

#if (websiteLive) {    enrichedQ <- enrichr(GOlist1000_TestQ, dbs)}
#if (websiteLive) plotEnrich(enrichedQ[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestQ")
#write.csv(head(enrichedQ[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestQ",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedQ[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestQ")
#write.csv(head(enrichedQ[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestQ",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedQ[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestQ")
#write.csv(head(enrichedQ[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestQ",groupsName,".csv"))

#if (websiteLive) {    enrichedR <- enrichr(GOlist1000_TestR, dbs)}
#if (websiteLive) plotEnrich(enrichedR[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestR")
#write.csv(head(enrichedR[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestR",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedR[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestR")
#write.csv(head(enrichedR[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestR",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedR[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestR")
#write.csv(head(enrichedR[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestR",groupsName,".csv"))

#if (websiteLive) {    enrichedS <- enrichr(GOlist1000_TestS, dbs)}
#if (websiteLive) plotEnrich(enrichedS[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestS")
#write.csv(head(enrichedS[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestS",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedS[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestS")
#write.csv(head(enrichedS[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestS",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedS[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestS")
#write.csv(head(enrichedS[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestS",groupsName,".csv"))

#if (websiteLive) {    enrichedT <- enrichr(GOlist1000_TestT, dbs)}
#if (websiteLive) plotEnrich(enrichedT[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestT")
#write.csv(head(enrichedT[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestT",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedT[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestT")
#write.csv(head(enrichedT[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestT",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedT[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestT")
#write.csv(head(enrichedT[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestT",groupsName,".csv"))

#if (websiteLive) {    enrichedU <- enrichr(GOlist1000_TestU, dbs)}
#if (websiteLive) plotEnrich(enrichedU[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestU")
#write.csv(head(enrichedU[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestU",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedU[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestU")
#write.csv(head(enrichedU[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestU",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedU[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestU")
#write.csv(head(enrichedU[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestU",groupsName,".csv"))

#if (websiteLive) {    enrichedV <- enrichr(GOlist1000_TestV, dbs)}
#if (websiteLive) plotEnrich(enrichedV[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestV")
#write.csv(head(enrichedV[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestV",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedV[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestV")
#write.csv(head(enrichedV[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestV",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedV[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestV")
#write.csv(head(enrichedV[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestV",groupsName,".csv"))

#if (websiteLive) {    enrichedV <- enrichr(GOlist1000_TestW, dbs)}
#if (websiteLive) plotEnrich(enrichedV[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestW")
#write.csv(head(enrichedV[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestW",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedV[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestW")
#write.csv(head(enrichedV[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestW",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedV[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestW")
#write.csv(head(enrichedV[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestW",groupsName,".csv"))

#if (websiteLive) {    enrichedX <- enrichr(GOlist1000_TestX, dbs)}
#if (websiteLive) plotEnrich(enrichedX[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestX")
#write.csv(head(enrichedX[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestX",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedX[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestX")
#write.csv(head(enrichedX[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestX",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedX[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestX")
#write.csv(head(enrichedX[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestX",groupsName,".csv"))

#if (websiteLive) {    enrichedY <- enrichr(GOlist1000_TestY, dbs)}
#if (websiteLive) plotEnrich(enrichedY[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestY")
#write.csv(head(enrichedY[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestY",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedY[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestY")
#write.csv(head(enrichedY[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestY",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedY[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestY")
#write.csv(head(enrichedY[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestY",groupsName,".csv"))

#if (websiteLive) {    enrichedZ <- enrichr(GOlist1000_TestZ, dbs)}
#if (websiteLive) plotEnrich(enrichedZ[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis TestZ")
#write.csv(head(enrichedZ[[1]], 50),paste0(path,"/GO1000/","Reactome_Enrichment_Analysis_TestZ",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedZ[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis TestZ")
#write.csv(head(enrichedZ[[2]], 50),paste0(path,"/GO1000/","Jensen_DISEASES_Enrichment_Analysis_TestZ",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedZ[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis TestZ")
#write.csv(head(enrichedZ[[3]], 50),paste0(path,"/GO1000/","GO_BP_Enrichment_Analysis_TestZ",groupsName,".csv"))

```

### GO_enrichments_padjLOGFC

Functional Annotation
```{r}
#If various groupings for functional annotation
groupsName<-"_clusterProfiler"
```


```{r}
path<-getwd()
dir.create(paste0(path,"/GOpadjLOGFC"))
write.csv((DEcounter),paste0(path,"/GOpadjLOGFC/","DEcounts.csv"))
```

Functional Annotation on top 1000 Genes using clusterprofiler. GO Analysis 





```{r}
#Vector containing all Sample_group names for GO analysis
#AllGeneNames<-row.names(read.delim("RNAseqJune2020_reanalysis_dupRemovedISGs_2Sym.txt", header = TRUE, sep = "\t",check.names=FALSE,row.names=1))
AllGeneNames<-BMannot$Gene_Symbol
```


```{r}

listAll<-list(GOlistUp_TestA,GOlistDown_TestA,GOlistUp_TestB,GOlistDown_TestB,GOlistUp_TestC,GOlistDown_TestC,GOlistUp_TestD,GOlistDown_TestD,GOlistUp_TestE,GOlistDown_TestE)
#need to name the vectors in the list, example here is for 5 groups
names(listAll)<-c("TestA_UP", "TestA_Down","TestB_UP","TestB_Down", "TestC_UP","TestC_Down", "TestD_UP","TestD_Down", "TestE_UP","TestE_Down")#make sure order the same as above!!!

#listAll<-list(GOlistUp_TestA,GOlistDown_TestA,GOlistUp_TestB,GOlistDown_TestB,GOlistUp_TestC,GOlistDown_TestC,GOlistUp_TestD,GOlistDown_TestD,GOlistUp_TestE,GOlistDown_TestE,GOlistUp_TestF,GOlistDown_TestF,GOlistUp_TestG,GOlistDown_TestG,GOlistUp_TestH,GOlistDown_TestH,GOlistUp_TestI,GOlistDown_TestI,GOlistUp_TestJ,GOlistDown_TestJ,GOlistUp_TestK,GOlistDown_TestK,GOlistUp_TestL,GOlistDown_TestL,GOlistUp_TestM,GOlistDown_TestM,GOlistUp_TestN,GOlistDown_TestN,GOlistUp_TestO,GOlistDown_TestO,GOlistUp_TestP,GOlistDown_TestP,GOlistUp_TestQ,GOlistDown_TestQ,GOlistUp_TestR,GOlistDown_TestR,GOlistUp_TestS,GOlistDown_TestS,GOlistUp_TestT,GOlistDown_TestT,GOlistUp_TestU,GOlistDown_TestU,GOlistUp_TestV,GOlistDown_TestV,GOlistUp_TestW,GOlistDown_TestW,GOlistUp_TestX,GOlistDown_TestX,GOlistUp_TestY,GOlistDown_TestY,GOlistUp_TestZ,GOlistDown_TestZ)
#need to name the vectors in the list, example here is for 5 groups
#names(listAll)<-c("TestA_UP", "TestA_Down","TestB_UP","TestB_Down", "TestC_UP","TestC_Down", "TestD_UP","TestD_Down", "TestE_UP","TestE_Down", "TestF_UP","TestF_Down", "TestG_UP","TestG_Down", "TestH_UP","TestH_Down", "TestI_UP","TestI_Down", "TestJ_UP","TestJ_Down", "TestK_UP","TestK_Down", "TestL_UP","TestL_Down", "TestM_UP","TestM_Down", "TestN_UP","TestN_Down", "TestO_UP","TestO_Down", "TestP_UP","TestP_Down", "TestQ_UP","TestQ_Down", "TestR_UP","TestR_Down", "TestS_UP","TestS_Down", "TestT_UP","TestT_Down", "TestU_UP","TestU_Down", "TestV_UP","TestV_Down", "TestW_UP","TestW_Down", "TestX_UP","TestX_Down", "TestY_UP","TestY_Down", "TestZ_UP","TestZ_Down")#make sure order the same as above!!!

#if you want to rearrange the order
#listAll<-listAll[c("x3", "x7", "x8", "x2", "x6", "x5", "x4", "x1")]

lapply(listAll, head)

```

```{r}

listAllR<-list(ReactomeUplist_TestA,ReactomeDownlist_TestA,ReactomeUplist_TestB,ReactomeDownlist_TestB,ReactomeUplist_TestC,ReactomeDownlist_TestC,ReactomeUplist_TestD,ReactomeDownlist_TestD,ReactomeUplist_TestE,ReactomeDownlist_TestE)
#need to name the vectors in the list, example here is for 5 groups
names(listAllR)<-c("TestA_UP", "TestB_UP", "TestC_UP", "TestD_UP", "TestE_UP")#make sure order the same as above!!!

#listAllR<-list(ReactomeUplist_TestA,ReactomeDownlist_TestA,ReactomeUplist_TestB,ReactomeDownlist_TestB,ReactomeUplist_TestC,ReactomeDownlist_TestC,ReactomeUplist_TestD,ReactomeDownlist_TestD,ReactomeUplist_TestE,ReactomeDownlist_TestE,ReactomeUplist_TestF,ReactomeDownlist_TestF,ReactomeUplist_TestG,ReactomeDownlist_TestG,ReactomeUplist_TestH,ReactomeDownlist_TestH,ReactomeUplist_TestI,ReactomeDownlist_TestI,ReactomeUplist_TestJ,ReactomeDownlist_TestJ,ReactomeUplist_TestK,ReactomeDownlist_TestK,ReactomeUplist_TestL,ReactomeDownlist_TestL,ReactomeUplist_TestM,ReactomeDownlist_TestM,ReactomeUplist_TestN,ReactomeDownlist_TestN,ReactomeUplist_TestO,ReactomeDownlist_TestO,ReactomeUplist_TestP,ReactomeDownlist_TestP,ReactomeUplist_TestQ,ReactomeDownlist_TestQ,ReactomeUplist_TestR,ReactomeDownlist_TestR,ReactomeUplist_TestS,ReactomeDownlist_TestS,ReactomeUplist_TestT,ReactomeDownlist_TestT,ReactomeUplist_TestU,ReactomeDownlist_TestU,ReactomeUplist_TestV,ReactomeDownlist_TestV,ReactomeUplist_TestW,ReactomeDownlist_TestW,ReactomeUplist_TestX,ReactomeDownlist_TestX,ReactomeUplist_TestY,ReactomeDownlist_TestY,ReactomeUplist_TestZ,ReactomeDownlist_TestZ)
#names(listAll)<-c("TestA_UP", "TestA_Down","TestB_UP","TestB_Down", "TestC_UP","TestC_Down", "TestD_UP","TestD_Down", "TestE_UP","TestE_Down", "TestF_UP","TestF_Down", "TestG_UP","TestG_Down", "TestH_UP","TestH_Down", "TestI_UP","TestI_Down", "TestJ_UP","TestJ_Down", "TestK_UP","TestK_Down", "TestL_UP","TestL_Down", "TestM_UP","TestM_Down", "TestN_UP","TestN_Down", "TestO_UP","TestO_Down", "TestP_UP","TestP_Down", "TestQ_UP","TestQ_Down", "TestR_UP","TestR_Down", "TestS_UP","TestS_Down", "TestT_UP","TestT_Down", "TestU_UP","TestU_Down", "TestV_UP","TestV_Down", "TestW_UP","TestW_Down", "TestX_UP","TestX_Down", "TestY_UP","TestY_Down", "TestZ_UP","TestZ_Down")#make sure order the same as above!!!


lapply(listAllR, head)

```

```{r}
res <- compareCluster(listAllR, fun="enrichPathway",universe = BMannot$entrez,
                   organism="human",
                   pvalueCutoff=0.05, 
                   qvalueCutoff = 0.10,
                   readable=T)

head(res)
write.csv(as.data.frame(res),paste0(path,"/GOpadjLOGFC/","Reactome",groupsName,".csv"))
```

```{r,fig.height = 8,fig.width = 10}
plotR <-dotplot(res,showCategory = 8
               ,
        title = paste0("Reactome Enrichment ",groupsName))+
   scale_y_discrete(labels=function(x) str_wrap(x, width=80))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
plotR
```

```{r}
pdf(paste0(path,"/GOpadjLOGFC/","Reactome1.pdf"), width = 10, height = 10)
plotR
dev.off()
```



- CC cellular compartment
- BP biological process
- MF molecular function


```{r}
#Could lump everything together by not specifying "ont".
#The simplify function has been used to cut down on GO redundancy
#CC
cgoCC <- compareCluster(geneCluster = listAll, 
                      universe = AllGeneNames,
                      fun = "enrichGO",
                      OrgDb=org.Hs.eg.db, 
                      #OrgDb=org.Mm.eg.db,
                      keyType="SYMBOL",
                      ont = "CC", 
                      pvalueCutoff=0.05,
                      qvalueCutoff = 0.10)
cgoCC2 <- clusterProfiler::simplify(cgoCC, cutoff=0.7, by="p.adjust", select_fun=min)
#write as spreadsheet
write.csv(as.data.frame(cgoCC2),paste0(path,"/GOpadjLOGFC/","GO_CC",groupsName,".csv"))



```

```{r}
#CC
png("dummy")
plotCC=dotplot(cgoCC2,showCategory = 30,
        title = paste0("GO Cellular Compartment ",groupsName))+
   scale_y_discrete(labels=function(x) str_wrap(x, width=80))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
dev.off()
```
```{r,fig.height = 24,fig.width = 12}
plotCC
```

```{r}
png(paste0(path,"/GOpadjLOGFC/","GO_CC",groupsName,".png"), width = 1224, height = 2024)
plotCC
dev.off()
```

GO BP
```{r}
#BP
cgoBP <- compareCluster(geneCluster = listAll, 
                      universe = AllGeneNames,
                      fun = "enrichGO",
                      OrgDb=org.Hs.eg.db, 
                      keyType="SYMBOL",
                      ont = "BP", 
                      pvalueCutoff=0.05,
                      qvalueCutoff = 0.10)
cgoBP2 <- clusterProfiler::simplify(cgoBP, cutoff=0.7, by="p.adjust", select_fun=min)
#write as spreadsheet
write.csv(as.data.frame(cgoBP2),paste0(path,"/GOpadjLOGFC/","GO_BP",groupsName,".csv"))



```

```{r}
#CC
png("dummy")
plotBP=dotplot(cgoBP2,showCategory = 30,
        title = paste0("GO Biological Process ",groupsName))+
   scale_y_discrete(labels=function(x) str_wrap(x, width=80))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
dev.off()
```

```{r,fig.height = 42,fig.width = 12}
plotBP

```

```{r}
png(paste0(path,"/GOpadjLOGFC/","GO_BP",groupsName,".png"), width = 1024, height = 4024)
plotBP
dev.off()
```


GO MF

```{r}
#MF
cgoMF <- compareCluster(geneCluster = listAll, 
                      universe = AllGeneNames,
                      fun = "enrichGO",
                      OrgDb=org.Hs.eg.db, 
                      keyType="SYMBOL",
                      ont = "MF", 
                      pvalueCutoff=0.05,
                      qvalueCutoff = 0.10)
cgoMF2 <- clusterProfiler::simplify(cgoMF, cutoff=0.7, by="p.adjust", select_fun=min)
#write as spreadsheet
write.csv(as.data.frame(cgoMF2),paste0(path,"/GOpadjLOGFC/","GO_MF",groupsName,".csv"))


```

```{r}
#MF
png("dummy")
plotMF=dotplot(cgoMF2,showCategory = 30,
        title = paste0("GO Molecular Function  ",groupsName))+
   scale_y_discrete(labels=function(x) str_wrap(x, width=80))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
dev.off()
```

```{r,fig.height = 30,fig.width = 12}
plotMF

```


```{r}
#print plot to file
png(paste0(path,"/GOpadjLOGFC/","GO_MF",groupsName,".png"), width = 1424, height = 2024)
plotMF
dev.off()
```


```{r}
#If various groupings for functional annotation
groupsName<-"_Enrichr"
```

```{r}
#Enrichr
setEnrichrSite("Enrichr") # Human genes
```
```{r}
websiteLive <- TRUE
dbs <- listEnrichrDbs()
if (is.null(dbs)) websiteLive <- FALSE
if (websiteLive) head(dbs)
```

```{r,fig.height = 7,fig.width = 12}
#dbs <- c("GO_Molecular_Function_2018", "GO_Cellular_Component_2018", "GO_Biological_Process_2018")


dbs <- c("Reactome_Pathways_2024","Jensen_DISEASES_Curated_2025", "GO_Biological_Process_2025")
#dbs <- c("Reactome_Pathways_2024","KEGG_2019_Mouse", "GO_Biological_Process_2025")

if (websiteLive) {    enrichedA <- enrichr(GOlistUp_TestA, dbs)}
if (websiteLive) plotEnrich(enrichedA[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_A_Up")
write.csv(head(enrichedA[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_A_Up")
write.csv(head(enrichedA[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_A_Up")
write.csv(head(enrichedA[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_A",groupsName,".csv"))

if (websiteLive) {    enrichedB <- enrichr(GOlistUp_TestB, dbs)}
if (websiteLive) plotEnrich(enrichedB[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_B_Up")
write.csv(head(enrichedB[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_B_Up")
write.csv(head(enrichedB[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_B_Up")
write.csv(head(enrichedB[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_B",groupsName,".csv"))

if (websiteLive) {    enrichedC <- enrichr(GOlistUp_TestC, dbs)}
if (websiteLive) plotEnrich(enrichedC[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_C_Up")
write.csv(head(enrichedC[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_C_Up")
write.csv(head(enrichedC[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_C_Up")
write.csv(head(enrichedC[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_C",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlistUp_TestD, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_D_Up")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_D_Up")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_D_Up")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_D",groupsName,".csv"))

if (websiteLive) {    enrichedE <- enrichr(GOlistUp_TestE, dbs)}
if (websiteLive) plotEnrich(enrichedE[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_E_Up")
write.csv(head(enrichedE[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedE[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_E_Up")
write.csv(head(enrichedE[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedE[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_E_Up")
write.csv(head(enrichedE[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_E",groupsName,".csv"))

```

```{r,fig.height = 7,fig.width = 12}
dbs <- c("CellMarker_2024","Descartes_Cell_Types_and_Tissue_2021", "RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO")

if (websiteLive) {    enrichedA <- enrichr(GOlistUp_TestA, dbs)}
if (websiteLive) plotEnrich(enrichedA[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis Test_A_Up")
write.csv(head(enrichedA[[1]], 50),paste0(path,"/GOpadjLOGFC/","CellMarker_2024_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis Test_A_Up")
write.csv(head(enrichedA[[2]], 50),paste0(path,"/GOpadjLOGFC/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis Test_A_Up")
write.csv(head(enrichedA[[3]], 50),paste0(path,"/GOpadjLOGFC/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestUp_A",groupsName,".csv"))

if (websiteLive) {    enrichedB <- enrichr(GOlistUp_TestB, dbs)}
if (websiteLive) plotEnrich(enrichedB[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis Test_B_Up")
write.csv(head(enrichedB[[1]], 50),paste0(path,"/GOpadjLOGFC/","CellMarker_2024_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis Test_B_Up")
write.csv(head(enrichedB[[2]], 50),paste0(path,"/GOpadjLOGFC/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis Test_B_Up")
write.csv(head(enrichedB[[3]], 50),paste0(path,"/GOpadjLOGFC/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestUp_B",groupsName,".csv"))

if (websiteLive) {    enrichedC <- enrichr(GOlistUp_TestC, dbs)}
if (websiteLive) plotEnrich(enrichedC[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis Test_C_Up")
write.csv(head(enrichedC[[1]], 50),paste0(path,"/GOpadjLOGFC/","CellMarker_2024_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis Test_C_Up")
write.csv(head(enrichedC[[2]], 50),paste0(path,"/GOpadjLOGFC/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis Test_C_Up")
write.csv(head(enrichedC[[3]], 50),paste0(path,"/GOpadjLOGFC/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestUp_C",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlistUp_TestD, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis Test_D_Up")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GOpadjLOGFC/","CellMarker_2024_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis Test_D_Up")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GOpadjLOGFC/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis Test_D_Up")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GOpadjLOGFC/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestUp_D",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlistUp_TestE, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis Test_E_Up")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GOpadjLOGFC/","CellMarker_2024_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis Test_E_Up")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GOpadjLOGFC/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis Test_E_Up")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GOpadjLOGFC/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
```

```{r,fig.height = 7,fig.width = 12}
#Rare Diseases
dbs <- c("Rare_Diseases_GeneRIF_ARCHS4_Predictions","Rare_Diseases_GeneRIF_Gene_Lists", "Rare_Diseases_AutoRIF_ARCHS4_Predictions", "Rare_Diseases_AutoRIF_Gene_Lists")


if (websiteLive) {    enrichedA <- enrichr(GOlistUp_TestA, dbs)}
if (websiteLive) plotEnrich(enrichedA[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis Test_A_Up")
write.csv(head(enrichedA[[1]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis Test_A_Up")
write.csv(head(enrichedA[[2]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis Test_A_Up")
write.csv(head(enrichedA[[3]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis Test_A_Up")
write.csv(head(enrichedA[[4]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestUp_A",groupsName,".csv"))

if (websiteLive) {    enrichedB <- enrichr(GOlistUp_TestB, dbs)}
if (websiteLive) plotEnrich(enrichedB[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis Test_B_Up")
write.csv(head(enrichedB[[1]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis Test_B_Up")
write.csv(head(enrichedB[[2]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis Test_B_Up")
write.csv(head(enrichedB[[3]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis Test_B_Up")
write.csv(head(enrichedB[[4]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestUp_B",groupsName,".csv"))

if (websiteLive) {    enrichedC <- enrichr(GOlistUp_TestC, dbs)}
if (websiteLive) plotEnrich(enrichedC[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis Test_C_Up")
write.csv(head(enrichedC[[1]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis Test_C_Up")
write.csv(head(enrichedC[[2]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis Test_C_Up")
write.csv(head(enrichedC[[3]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis Test_C_Up")
write.csv(head(enrichedC[[4]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestUp_C",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlistUp_TestD, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis Test_D_Up")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis Test_D_Up")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis Test_D_Up")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis Test_D_Up")
write.csv(head(enrichedD[[4]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestUp_D",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlistUp_TestE, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis Test_E_Up")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis Test_E_Up")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis Test_E_Up")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis Test_E_Up")
write.csv(head(enrichedD[[4]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestUp_E",groupsName,".csv"))

```

```{r,fig.height = 7,fig.width = 12}

#if (websiteLive) {    enrichedF <- enrichr(GOlistUp_TestF, dbs)}
#if (websiteLive) plotEnrich(enrichedF[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_F_Up")
#write.csv(head(enrichedF[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_F",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedF[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_F_Up")
#write.csv(head(enrichedF[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_F",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedF[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_F_Up")
#write.csv(head(enrichedF[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_F",groupsName,".csv"))

#if (websiteLive) {    enrichedG <- enrichr(GOlistUp_TestG, dbs)}
#if (websiteLive) plotEnrich(enrichedG[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_G_Up")
#write.csv(head(enrichedG[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_G",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedG[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_G_Up")
#write.csv(head(enrichedG[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_G",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedG[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_G_Up")
#write.csv(head(enrichedG[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_G",groupsName,".csv"))

#if (websiteLive) {    enrichedH <- enrichr(GOlistUp_TestH, dbs)}
#if (websiteLive) plotEnrich(enrichedH[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_H_Up")
#write.csv(head(enrichedH[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_H",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedH[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_H_Up")
#write.csv(head(enrichedH[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_H",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedH[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_H_Up")
#write.csv(head(enrichedH[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_H",groupsName,".csv"))

#if (websiteLive) {    enrichedI <- enrichr(GOlistUp_TestI, dbs)}
#if (websiteLive) plotEnrich(enrichedI[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_I_Up")
#write.csv(head(enrichedI[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_I",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedI[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_I_Up")
#write.csv(head(enrichedI[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_I",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedI[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_I_Up")
#write.csv(head(enrichedI[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_I",groupsName,".csv"))

#if (websiteLive) {    enrichedJ <- enrichr(GOlistUp_TestJ, dbs)}
#if (websiteLive) plotEnrich(enrichedJ[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_J_Up")
#write.csv(head(enrichedJ[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_J",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedJ[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_J_Up")
#write.csv(head(enrichedJ[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_J",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedJ[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_J_Up")
#write.csv(head(enrichedJ[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_J",groupsName,".csv"))

#if (websiteLive) {    enrichedK <- enrichr(GOlistUp_TestK, dbs)}
#if (websiteLive) plotEnrich(enrichedK[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_K_Up")
#write.csv(head(enrichedK[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_K",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedK[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_K_Up")
#write.csv(head(enrichedK[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_K",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedK[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_K_Up")
#write.csv(head(enrichedK[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_K",groupsName,".csv"))

#if (websiteLive) {    enrichedL <- enrichr(GOlistUp_TestL, dbs)}
#if (websiteLive) plotEnrich(enrichedL[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_L_Up")
#write.csv(head(enrichedL[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_L",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedL[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_L_Up")
#write.csv(head(enrichedL[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_L",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedL[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_L_Up")
#write.csv(head(enrichedL[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_L",groupsName,".csv"))

#if (websiteLive) {    enrichedM <- enrichr(GOlistUp_TestM, dbs)}
#if (websiteLive) plotEnrich(enrichedM[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_M_Up")
#write.csv(head(enrichedM[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_M",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedM[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_M_Up")
#write.csv(head(enrichedM[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_M",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedM[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_M_Up")
#write.csv(head(enrichedM[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_M",groupsName,".csv"))

#if (websiteLive) {    enrichedN <- enrichr(GOlistUp_TestN, dbs)}
#if (websiteLive) plotEnrich(enrichedN[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_N_Up")
#write.csv(head(enrichedN[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_N",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedN[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_N_Up")
#write.csv(head(enrichedN[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_N",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedN[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_N_Up")
#write.csv(head(enrichedN[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_N",groupsName,".csv"))

#if (websiteLive) {    enrichedO <- enrichr(GOlistUp_TestO, dbs)}
#if (websiteLive) plotEnrich(enrichedO[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_O_Up")
#write.csv(head(enrichedO[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_O",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedO[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_O_Up")
#write.csv(head(enrichedO[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_O",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedO[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_O_Up")
#write.csv(head(enrichedO[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_O",groupsName,".csv"))

#if (websiteLive) {    enrichedP <- enrichr(GOlistUp_TestP, dbs)}
#if (websiteLive) plotEnrich(enrichedP[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_P_Up")
#write.csv(head(enrichedP[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_P",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedP[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_P_Up")
#write.csv(head(enrichedP[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_P",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedP[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_P_Up")
#write.csv(head(enrichedP[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_P",groupsName,".csv"))

#if (websiteLive) {    enrichedQ <- enrichr(GOlistUp_TestQ, dbs)}
#if (websiteLive) plotEnrich(enrichedQ[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_Q_Up")
#write.csv(head(enrichedQ[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_Q",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedQ[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_Q_Up")
#write.csv(head(enrichedQ[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_Q",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedQ[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_Q_Up")
#write.csv(head(enrichedQ[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_Q",groupsName,".csv"))

#if (websiteLive) {    enrichedR <- enrichr(GOlistUp_TestR, dbs)}
#if (websiteLive) plotEnrich(enrichedR[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_R_Up")
#write.csv(head(enrichedR[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_R",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedR[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_R_Up")
#write.csv(head(enrichedR[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_R",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedR[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_R_Up")
#write.csv(head(enrichedR[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_R",groupsName,".csv"))

#if (websiteLive) {    enrichedS <- enrichr(GOlistUp_TestS, dbs)}
#if (websiteLive) plotEnrich(enrichedS[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_S_Up")
#write.csv(head(enrichedS[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_S",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedS[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_S_Up")
#write.csv(head(enrichedS[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_S",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedS[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_S_Up")
#write.csv(head(enrichedS[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_S",groupsName,".csv"))

#if (websiteLive) {    enrichedT <- enrichr(GOlistUp_TestT, dbs)}
#if (websiteLive) plotEnrich(enrichedT[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_T_Up")
#write.csv(head(enrichedT[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_T",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedT[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_T_Up")
#write.csv(head(enrichedT[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_T",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedT[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_T_Up")
#write.csv(head(enrichedT[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_T",groupsName,".csv"))

#if (websiteLive) {    enrichedU <- enrichr(GOlistUp_TestU, dbs)}
#if (websiteLive) plotEnrich(enrichedU[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_U_Up")
#write.csv(head(enrichedU[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_U",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedU[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_U_Up")
#write.csv(head(enrichedU[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_U",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedU[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_U_Up")
#write.csv(head(enrichedU[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_U",groupsName,".csv"))

#if (websiteLive) {    enrichedV <- enrichr(GOlistUp_TestV, dbs)}
#if (websiteLive) plotEnrich(enrichedV[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_V_Up")
#write.csv(head(enrichedV[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_V",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedV[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_V_Up")
#write.csv(head(enrichedV[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_V",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedV[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_V_Up")
#write.csv(head(enrichedV[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_V",groupsName,".csv"))

#if (websiteLive) {    enrichedV <- enrichr(GOlistUp_TestW, dbs)}
#if (websiteLive) plotEnrich(enrichedV[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_Up_W")
#write.csv(head(enrichedV[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_W",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedV[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_Up_W")
#write.csv(head(enrichedV[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_W",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedV[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_Up_W")
#write.csv(head(enrichedV[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_W",groupsName,".csv"))

#if (websiteLive) {    enrichedX <- enrichr(GOlistUp_TestX, dbs)}
#if (websiteLive) plotEnrich(enrichedX[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_X_Up")
#write.csv(head(enrichedX[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_X",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedX[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_X_Up")
#write.csv(head(enrichedX[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_X",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedX[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_X_Up")
#write.csv(head(enrichedX[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_X",groupsName,".csv"))

#if (websiteLive) {    enrichedY <- enrichr(GOlistUp_TestY, dbs)}
#if (websiteLive) plotEnrich(enrichedY[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_Y_Up")
#write.csv(head(enrichedY[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_Y",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedY[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_Y_Up")
#write.csv(head(enrichedY[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_Y",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedY[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_Y_Up")
#write.csv(head(enrichedY[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_Y",groupsName,".csv"))

#if (websiteLive) {    enrichedZ <- enrichr(GOlistUp_TestZ, dbs)}
#if (websiteLive) plotEnrich(enrichedZ[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_Z_Up")
#write.csv(head(enrichedZ[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_Z",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedZ[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_Z_Up")
#write.csv(head(enrichedZ[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_Z",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedZ[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_Z_Up")
#write.csv(head(enrichedZ[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_Z",groupsName,".csv"))

```

```{r,fig.height = 7,fig.width = 12}
#dbs <- c("GO_Molecular_Function_2018", "GO_Cellular_Component_2018", "GO_Biological_Process_2018")


dbs <- c("Reactome_Pathways_2024","Jensen_DISEASES_Curated_2025", "GO_Biological_Process_2025")
#dbs <- c("Reactome_Pathways_2024","KEGG_2019_Mouse", "GO_Biological_Process_2025")

if (websiteLive) {    enrichedA <- enrichr(GOlistDown_TestA, dbs)}
if (websiteLive) plotEnrich(enrichedA[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_A_Down")
write.csv(head(enrichedA[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_A_Down")
write.csv(head(enrichedA[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_A_Down")
write.csv(head(enrichedA[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_A",groupsName,".csv"))

if (websiteLive) {    enrichedB <- enrichr(GOlistDown_TestB, dbs)}
if (websiteLive) plotEnrich(enrichedB[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_B_Down")
write.csv(head(enrichedB[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_B_Down")
write.csv(head(enrichedB[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_B_Down")
write.csv(head(enrichedB[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_B",groupsName,".csv"))

if (websiteLive) {    enrichedC <- enrichr(GOlistDown_TestC, dbs)}
if (websiteLive) plotEnrich(enrichedC[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_C_Down")
write.csv(head(enrichedC[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_C_Down")
write.csv(head(enrichedC[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_C_Down")
write.csv(head(enrichedC[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_C",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlistDown_TestD, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_D_Down")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_D_Down")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_D_Down")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_D",groupsName,".csv"))

if (websiteLive) {    enrichedE <- enrichr(GOlistDown_TestE, dbs)}
if (websiteLive) plotEnrich(enrichedE[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_E_Down")
write.csv(head(enrichedE[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedE[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_E_Down")
write.csv(head(enrichedE[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedE[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_E_Down")
write.csv(head(enrichedE[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_E",groupsName,".csv"))

```

```{r,fig.height = 7,fig.width = 12}
dbs <- c("CellMarker_2024","Descartes_Cell_Types_and_Tissue_2021", "RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO")

if (websiteLive) {    enrichedA <- enrichr(GOlistDown_TestA, dbs)}
if (websiteLive) plotEnrich(enrichedA[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis Test_A_Down")
write.csv(head(enrichedA[[1]], 50),paste0(path,"/GOpadjLOGFC/","CellMarker_2024_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis Test_A_Down")
write.csv(head(enrichedA[[2]], 50),paste0(path,"/GOpadjLOGFC/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis Test_A_Down")
write.csv(head(enrichedA[[3]], 50),paste0(path,"/GOpadjLOGFC/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestUp_A",groupsName,".csv"))

if (websiteLive) {    enrichedB <- enrichr(GOlistDown_TestB, dbs)}
if (websiteLive) plotEnrich(enrichedB[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis Test_B_Down")
write.csv(head(enrichedB[[1]], 50),paste0(path,"/GOpadjLOGFC/","CellMarker_2024_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis Test_B_Down")
write.csv(head(enrichedB[[2]], 50),paste0(path,"/GOpadjLOGFC/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis Test_B_Down")
write.csv(head(enrichedB[[3]], 50),paste0(path,"/GOpadjLOGFC/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestUp_B",groupsName,".csv"))

if (websiteLive) {    enrichedC <- enrichr(GOlistDown_TestC, dbs)}
if (websiteLive) plotEnrich(enrichedC[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis Test_C_Down")
write.csv(head(enrichedC[[1]], 50),paste0(path,"/GOpadjLOGFC/","CellMarker_2024_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis Test_C_Down")
write.csv(head(enrichedC[[2]], 50),paste0(path,"/GOpadjLOGFC/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis Test_C_Down")
write.csv(head(enrichedC[[3]], 50),paste0(path,"/GOpadjLOGFC/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestUp_C",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlistDown_TestD, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis Test_D_Down")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GOpadjLOGFC/","CellMarker_2024_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis Test_D_Down")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GOpadjLOGFC/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis Test_D_Down")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GOpadjLOGFC/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestUp_D",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlistDown_TestE, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="CellMarker_2024 Enrichment Analysis Test_E_Down")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GOpadjLOGFC/","CellMarker_2024_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Descartes_Cell_Types_and_Tissue_2021 Enrichment Analysis Test_E_Down")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GOpadjLOGFC/","Descartes_Cell_Types_and_Tissue_2021_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO Enrichment Analysis Test_E_Down")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GOpadjLOGFC/","RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
```

```{r,fig.height = 7,fig.width = 12}
#Rare Diseases
dbs <- c("Rare_Diseases_GeneRIF_ARCHS4_Predictions","Rare_Diseases_GeneRIF_Gene_Lists", "Rare_Diseases_AutoRIF_ARCHS4_Predictions", "Rare_Diseases_AutoRIF_Gene_Lists")


if (websiteLive) {    enrichedA <- enrichr(GOlistDown_TestA, dbs)}
if (websiteLive) plotEnrich(enrichedA[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis Test_A_Down")
write.csv(head(enrichedA[[1]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis Test_A_Down")
write.csv(head(enrichedA[[2]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis Test_A_Down")
write.csv(head(enrichedA[[3]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_A",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedA[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis Test_A_Down")
write.csv(head(enrichedA[[4]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestUp_A",groupsName,".csv"))

if (websiteLive) {    enrichedB <- enrichr(GOlistDown_TestB, dbs)}
if (websiteLive) plotEnrich(enrichedB[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis Test_B_Down")
write.csv(head(enrichedB[[1]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis Test_B_Down")
write.csv(head(enrichedB[[2]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis Test_B_Down")
write.csv(head(enrichedB[[3]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_B",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedB[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis Test_B_Down")
write.csv(head(enrichedB[[4]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestUp_B",groupsName,".csv"))

if (websiteLive) {    enrichedC <- enrichr(GOlistDown_TestC, dbs)}
if (websiteLive) plotEnrich(enrichedC[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis Test_C_Down")
write.csv(head(enrichedC[[1]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis Test_C_Down")
write.csv(head(enrichedC[[2]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis Test_C_Down")
write.csv(head(enrichedC[[3]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_C",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedC[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis Test_C_Down")
write.csv(head(enrichedC[[4]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestUp_C",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlistDown_TestD, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis Test_D_Down")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis Test_D_Down")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis Test_D_Down")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_D",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis Test_D_Down")
write.csv(head(enrichedD[[4]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestUp_D",groupsName,".csv"))

if (websiteLive) {    enrichedD <- enrichr(GOlistDown_TestE, dbs)}
if (websiteLive) plotEnrich(enrichedD[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_ARCHS4_Predictions Enrichment Analysis Test_E_Down")
write.csv(head(enrichedD[[1]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_GeneRIF_Gene_Lists Enrichment Analysis Test_E_Down")
write.csv(head(enrichedD[[2]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_GeneRIF_Gene_Lists_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_ARCHS4_Predictions Enrichment Analysis Test_E_Down")
write.csv(head(enrichedD[[3]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_ARCHS4_Predictions_Enrichment_Analysis_TestUp_E",groupsName,".csv"))
if (websiteLive) plotEnrich(enrichedD[[4]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Rare_Diseases_AutoRIF_Gene_Lists Enrichment Analysis Test_E_Down")
write.csv(head(enrichedD[[4]], 50),paste0(path,"/GOpadjLOGFC/","Rare_Diseases_AutoRIF_Gene_Lists_Enrichment_Analysis_TestUp_E",groupsName,".csv"))

```

```{r,fig.height = 7,fig.width = 12}

#if (websiteLive) {    enrichedF <- enrichr(GOlistDown_TestF, dbs)}
#if (websiteLive) plotEnrich(enrichedF[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_F_Down")
#write.csv(head(enrichedF[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_F",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedF[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_F_Down")
#write.csv(head(enrichedF[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_F",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedF[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_F_Down")
#write.csv(head(enrichedF[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_F",groupsName,".csv"))

#if (websiteLive) {    enrichedG <- enrichr(GOlistDown_TestG, dbs)}
#if (websiteLive) plotEnrich(enrichedG[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_G_Down")
#write.csv(head(enrichedG[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_G",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedG[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_G_Down")
#write.csv(head(enrichedG[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_G",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedG[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_G_Down")
#write.csv(head(enrichedG[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_G",groupsName,".csv"))

#if (websiteLive) {    enrichedH <- enrichr(GOlistDown_TestH, dbs)}
#if (websiteLive) plotEnrich(enrichedH[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_H_Down")
#write.csv(head(enrichedH[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_H",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedH[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_H_Down")
#write.csv(head(enrichedH[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_H",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedH[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_H_Down")
#write.csv(head(enrichedH[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_H",groupsName,".csv"))

#if (websiteLive) {    enrichedI <- enrichr(GOlistDown_TestI, dbs)}
#if (websiteLive) plotEnrich(enrichedI[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_I_Down")
#write.csv(head(enrichedI[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_I",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedI[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_I_Down")
#write.csv(head(enrichedI[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_I",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedI[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_I_Down")
#write.csv(head(enrichedI[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_I",groupsName,".csv"))

#if (websiteLive) {    enrichedJ <- enrichr(GOlistDown_TestJ, dbs)}
#if (websiteLive) plotEnrich(enrichedJ[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_J_Down")
#write.csv(head(enrichedJ[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_J",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedJ[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_J_Down")
#write.csv(head(enrichedJ[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_J",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedJ[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_J_Down")
#write.csv(head(enrichedJ[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_J",groupsName,".csv"))

#if (websiteLive) {    enrichedK <- enrichr(GOlistDown_TestK, dbs)}
#if (websiteLive) plotEnrich(enrichedK[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_K_Down")
#write.csv(head(enrichedK[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_K",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedK[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_K_Down")
#write.csv(head(enrichedK[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_K",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedK[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_K_Down")
#write.csv(head(enrichedK[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_K",groupsName,".csv"))

#if (websiteLive) {    enrichedL <- enrichr(GOlistDown_TestL, dbs)}
#if (websiteLive) plotEnrich(enrichedL[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_L_Down")
#write.csv(head(enrichedL[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_L",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedL[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_L_Down")
#write.csv(head(enrichedL[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_L",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedL[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_L_Down")
#write.csv(head(enrichedL[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_L",groupsName,".csv"))

#if (websiteLive) {    enrichedM <- enrichr(GOlistDown_TestM, dbs)}
#if (websiteLive) plotEnrich(enrichedM[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_M_Down")
#write.csv(head(enrichedM[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_M",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedM[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_M_Down")
#write.csv(head(enrichedM[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_M",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedM[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_M_Down")
#write.csv(head(enrichedM[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_M",groupsName,".csv"))

#if (websiteLive) {    enrichedN <- enrichr(GOlistDown_TestN, dbs)}
#if (websiteLive) plotEnrich(enrichedN[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_N_Down")
#write.csv(head(enrichedN[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_N",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedN[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_N_Down")
#write.csv(head(enrichedN[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_N",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedN[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_N_Down")
#write.csv(head(enrichedN[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_N",groupsName,".csv"))

#if (websiteLive) {    enrichedO <- enrichr(GOlistDown_TestO, dbs)}
#if (websiteLive) plotEnrich(enrichedO[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_O_Down")
#write.csv(head(enrichedO[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_O",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedO[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_O_Down")
#write.csv(head(enrichedO[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_O",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedO[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_O_Down")
#write.csv(head(enrichedO[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_O",groupsName,".csv"))

#if (websiteLive) {    enrichedP <- enrichr(GOlistDown_TestP, dbs)}
#if (websiteLive) plotEnrich(enrichedP[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_P_Down")
#write.csv(head(enrichedP[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_P",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedP[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_P_Down")
#write.csv(head(enrichedP[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_P",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedP[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_P_Down")
#write.csv(head(enrichedP[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_P",groupsName,".csv"))

#if (websiteLive) {    enrichedQ <- enrichr(GOlistDown_TestQ, dbs)}
#if (websiteLive) plotEnrich(enrichedQ[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_Q_Down")
#write.csv(head(enrichedQ[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_Q",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedQ[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_Q_Down")
#write.csv(head(enrichedQ[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_Q",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedQ[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_Q_Down")
#write.csv(head(enrichedQ[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_Q",groupsName,".csv"))

#if (websiteLive) {    enrichedR <- enrichr(GOlistDown_TestR, dbs)}
#if (websiteLive) plotEnrich(enrichedR[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_R_Down")
#write.csv(head(enrichedR[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_R",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedR[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_R_Down")
#write.csv(head(enrichedR[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_R",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedR[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_R_Down")
#write.csv(head(enrichedR[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_R",groupsName,".csv"))

#if (websiteLive) {    enrichedS <- enrichr(GOlistDown_TestS, dbs)}
#if (websiteLive) plotEnrich(enrichedS[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_S_Down")
#write.csv(head(enrichedS[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_S",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedS[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_S_Down")
#write.csv(head(enrichedS[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_S",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedS[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_S_Down")
#write.csv(head(enrichedS[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_S",groupsName,".csv"))

#if (websiteLive) {    enrichedT <- enrichr(GOlistDown_TestT, dbs)}
#if (websiteLive) plotEnrich(enrichedT[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_T_Down")
#write.csv(head(enrichedT[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_T",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedT[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_T_Down")
#write.csv(head(enrichedT[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_T",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedT[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_T_Down")
#write.csv(head(enrichedT[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_T",groupsName,".csv"))

#if (websiteLive) {    enrichedU <- enrichr(GOlistDown_TestU, dbs)}
#if (websiteLive) plotEnrich(enrichedU[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_U_Down")
#write.csv(head(enrichedU[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_U",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedU[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_U_Down")
#write.csv(head(enrichedU[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_U",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedU[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_U_Down")
#write.csv(head(enrichedU[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_U",groupsName,".csv"))

#if (websiteLive) {    enrichedV <- enrichr(GOlistDown_TestV, dbs)}
#if (websiteLive) plotEnrich(enrichedV[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_V_Down")
#write.csv(head(enrichedV[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_V",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedV[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_V_Down")
#write.csv(head(enrichedV[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_V",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedV[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_V_Down")
#write.csv(head(enrichedV[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_V",groupsName,".csv"))

#if (websiteLive) {    enrichedV <- enrichr(GOlistDown_TestW, dbs)}
#if (websiteLive) plotEnrich(enrichedV[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_Down_W")
#write.csv(head(enrichedV[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_W",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedV[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_Down_W")
#write.csv(head(enrichedV[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_W",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedV[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_Down_W")
#write.csv(head(enrichedV[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_W",groupsName,".csv"))

#if (websiteLive) {    enrichedX <- enrichr(GOlistDown_TestX, dbs)}
#if (websiteLive) plotEnrich(enrichedX[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_X_Down")
#write.csv(head(enrichedX[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_X",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedX[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_X_Down")
#write.csv(head(enrichedX[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_X",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedX[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_X_Down")
#write.csv(head(enrichedX[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_X",groupsName,".csv"))

#if (websiteLive) {    enrichedY <- enrichr(GOlistDown_TestY, dbs)}
#if (websiteLive) plotEnrich(enrichedY[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_Y_Down")
#write.csv(head(enrichedY[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_Y",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedY[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_Y_Down")
#write.csv(head(enrichedY[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_Y",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedY[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_Y_Down")
#write.csv(head(enrichedY[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_Y",groupsName,".csv"))

#if (websiteLive) {    enrichedZ <- enrichr(GOlistDown_TestZ, dbs)}
#if (websiteLive) plotEnrich(enrichedZ[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Test_Z_Down")
#write.csv(head(enrichedZ[[1]], 50),paste0(path,"/GOpadjLOGFC/","Reactome_Enrichment_Analysis_TestUp_Z",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedZ[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Test_Z_Down")
#write.csv(head(enrichedZ[[2]], 50),paste0(path,"/GOpadjLOGFC/","Jensen_DISEASES_Enrichment_Analysis_TestUp_Z",groupsName,".csv"))
#if (websiteLive) plotEnrich(enrichedZ[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Test_Z_Down")
#write.csv(head(enrichedZ[[3]], 50),paste0(path,"/GOpadjLOGFC/","GO_BP_Enrichment_Analysis_TestUp_Z",groupsName,".csv"))

```

### k-means clustering
```{r}
groupsName<-"kmeans"
DEcounter
```

```{r}
path<-getwd()
dir.create(paste0(path,"/Kmeans"))
```

```{r}
#decide on genelist, in this case using interaction pvalue
ResAllTr<-resAll[-c(10:30) ]
#rownames(ResAllTr)
rownames(ResAllTr) <- NULL
ResAllTr = mutate(ResAllTr, Include=
                   ifelse(ResAllTr$pvalue_TestE<0.05&!is.na(ResAllTr$pvalue_TestE), "in", 
                                                                       "out"))


#ResAllTr
#library(dplyr)
ResAllTr %>%
     group_by(Include) %>% 
     tally()
```

genes selected for k-means clustering

```{r}
topDEgenes <- which(ResAllTr$Include=="in")#find indexes 
```

```{r}
head(baseMeans)
```
```{r}
#means
dataHMm<-baseMeans[ topDEgenes, ]
dataHMm <- log2(dataHMm+1)
dataHMm<- t(as.matrix(dataHMm))
dataHMm <- t(scale(dataHMm))

colAnnm <- HeatmapAnnotation(df=annm, which="col", col=coloursm, annotation_width=unit(c(2, 4), "cm"), gap=unit(1, "mm"))

hmap_hier_factors4 <- Heatmap(
  dataHMm,  name = "Expression",
  row_labels = paste0(rownames(dataHMm)," ",(GeneSymbolHM<-BMannot2[ topDEgenes, ])$Gene_Symbol),
  column_title = paste0("Top DE genes means"), 
  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
  width = unit(50, "mm"),
  cluster_columns = FALSE,
  show_row_names = FALSE)#,  top_annotation=colAnnm  )
hmap_hier_factors4
```


```{r,fig.height = 5,fig.width =12 }
par(mfrow=c(1,2))
# Silhouette method
fviz_nbclust(dataHMm, kmeans, method = "silhouette",k.max = 16)+
  labs(subtitle = "Silhouette method")

# Elbow method
fviz_nbclust(dataHMm, kmeans, method = "wss",k.max = 16) +
  labs(subtitle = "Elbow method")

```

```{r}
#gap stat slow!!!
#set.seed(123)
#fviz_nbclust(dataHMm, kmeans, nstart = 25,  method = "gap_stat", nboot = 100,k.max = 16)+
#  labs(subtitle = "Gap statistic method")
```


```{r,fig.height = 6}
#decide on number of cluster (8 in this case) and run
kclust1 <- kmeans(dataHMm, 8)
#silhouette plot
distK<-daisy(dataHMm)
plot(silhouette(kclust1$cluster, distK), col=1:8, border=NA)

```

```{r,fig.height = 8,fig.width = 8}
split <- paste0("Cluster\n", kclust1$cluster)

hmap_k <- Heatmap(dataHMm, split=split,# cluster_row_slices = FALSE
                  cluster_columns = FALSE,
                  show_row_names = FALSE,
                  name = "Expression",
                  col = col_funF,
                  column_title = "k-means clustering", 
                  column_title_gp = gpar(fontsize = 16, fontface = "bold"),
                        )#top_annotation=colAnnm)
hmap_k
```

```{r}
#cluster annotation
#Response_Time<-data.frame(kclust1$cluster)

#Response_Time = mutate(Response_Time, Response=
 #                  ifelse(Response_Time$kclust1.cluster==1, "early", 
  #                        ifelse(Response_Time$kclust1.cluster==2, "transient",
   #                              ifelse(Response_Time$kclust1.cluster==3, "late",
    #                                    ifelse(Response_Time$kclust1.cluster==4, "transient",
     #                                          ifelse(Response_Time$kclust1.cluster==5, "early",
      #                                                ifelse(Response_Time$kclust1.cluster==6, "late",
       #                                                      ifelse(Response_Time$kclust1.cluster==7, "late",
        #                                                            ifelse(Response_Time$kclust1.cluster==8, "late",
         #                                                              "out")))))))))
#Response_Time<-Response_Time[c(2)]
#rownames(Response_Time) <- NULL
#ha = HeatmapAnnotation(df = Response_Time, which = "row", width = unit(1, "cm"),col = list(Response = c("early" =  "#C77CFF3", "late" = "brown", "transient" = "violet")))
#
#hmap_k+ha
```

```{r}
head(dataHMm)
```

```{r}
head(BMannot2)
```

```{r,fig.height = 9,fig.width = 12}
#Plot cluster means
#NB orderN vector needs editing and number of plots depends on cluster number
orderN<-c("Cntl","AAA1","BAA2","CAA3")# manual

clustercount<-data.frame(kclust1$cluster)
clustersizes<-table(clustercount$kclust1.cluster)
clusterMeans<-data.frame(kclust1$centers)
clusterMeans1<-data.frame(t(clusterMeans))
clusterMeans1 <- cbind(rownames(clusterMeans1), clusterMeans1)
rownames(clusterMeans1) <- NULL
names(clusterMeans1)[names(clusterMeans1)=="rownames(clusterMeans1)"] <- "Sample"
#clusterMeans1
pX1<-ggplot(data=clusterMeans1, aes(x=Sample, y=X1,group=1)) +
  geom_line()+  geom_point()+ggtitle(paste("Cluster X1 Profile ",clustersizes[1]," genes"))+  scale_x_discrete(limits=orderN)+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank())+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
pX2<-ggplot(data=clusterMeans1, aes(x=Sample, y=X2,group=1)) +
  geom_line()+  geom_point()+ggtitle(paste("Cluster X2 Profile ",clustersizes[2]," genes"))+  scale_x_discrete(limits=orderN)+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank())+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
pX3<-ggplot(data=clusterMeans1, aes(x=Sample, y=X3,group=1)) +
  geom_line()+  geom_point()+ggtitle(paste("Cluster X3 Profile ",clustersizes[3]," genes"))+  scale_x_discrete(limits=orderN)+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank())+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
pX4<-ggplot(data=clusterMeans1, aes(x=Sample, y=X4,group=1)) +
  geom_line()+  geom_point()+ggtitle(paste("Cluster X4 Profile ",clustersizes[4]," genes"))+  scale_x_discrete(limits=orderN)+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank())+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
pX5<-ggplot(data=clusterMeans1, aes(x=Sample, y=X5,group=1)) +
  geom_line()+  geom_point()+ggtitle(paste("Cluster X5 Profile ",clustersizes[5]," genes"))+  scale_x_discrete(limits=orderN)+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank())+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
pX6<-ggplot(data=clusterMeans1, aes(x=Sample, y=X6,group=1)) +
  geom_line()+  geom_point()+ggtitle(paste("Cluster X6 Profile ",clustersizes[6]," genes"))+  scale_x_discrete(limits=orderN)+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank())+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
pX7<-ggplot(data=clusterMeans1, aes(x=Sample, y=X7,group=1)) +
  geom_line()+  geom_point()+ggtitle(paste("Cluster X7 Profile ",clustersizes[7]," genes"))+  scale_x_discrete(limits=orderN)+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank())+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
pX8<-ggplot(data=clusterMeans1, aes(x=Sample, y=X8,group=1)) +
  geom_line()+  geom_point()+ggtitle(paste("Cluster X8 Profile ",clustersizes[8]," genes"))+  scale_x_discrete(limits=orderN)+
  theme(axis.title.x = element_blank(),axis.title.y = element_blank())+   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

#plot
multiplot(pX1, pX2, pX3, pX4,pX5,pX6,pX7,pX8, cols=3)

```

```{r}
topDEgenes <- which(ResAllTr$Include=="in")#find indexes
ResAllTrkm<-ResAllTr[ topDEgenes, ]
SymbolsKm<-dplyr::pull(ResAllTrkm, Gene_Symbol)
# export the gene expression data for the clusters
write.csv(clusterMeans,paste0(path,"/Kmeans/ClusterMeansKm.csv"))
ClusteredGenes<-data.frame(kclust1$cluster,SymbolsKm,dataHMm)
write.csv(ClusteredGenes,paste0(path,"/Kmeans/ScaledDataInClustersKm.csv"))
head(ClusteredGenes)
```

```{r}
#write files for ipa
bottomDEgenes<-which(ResAllTr$Include=="out")#find indexes 
bottomG<-ResAllTr[ bottomDEgenes, ]
bottomG<-dplyr::pull(bottomG,Original_ID, Gene_Symbol)
write.csv(bottomG,paste0(path,"/Kmeans/","ipaBottomKmeans.csv"))#genes that were not included in clustering
topDEgenes <- which(ResAllTr$Include=="in")#find indexes 
ResAllTrkm<-ResAllTr[ topDEgenes, ]
SymbolsKm<-dplyr::pull(ResAllTrkm, Gene_Symbol)
ipaKmeans<-ClusteredGenes
ipaKmeans<-ipaKmeans[,c(1:2)]
ipaKmeans$name2<-rownames(ipaKmeans)
ipaKmeans = mutate(ipaKmeans, x1= ifelse(ipaKmeans$kclust1.cluster==1, "1", "0"))
ipaKmeans = mutate(ipaKmeans, x2= ifelse(ipaKmeans$kclust1.cluster==2, "1", "0"))
ipaKmeans = mutate(ipaKmeans, x3= ifelse(ipaKmeans$kclust1.cluster==3, "1", "0"))
ipaKmeans = mutate(ipaKmeans, x4= ifelse(ipaKmeans$kclust1.cluster==4, "1", "0"))
ipaKmeans = mutate(ipaKmeans, x5= ifelse(ipaKmeans$kclust1.cluster==5, "1", "0"))
ipaKmeans = mutate(ipaKmeans, x6= ifelse(ipaKmeans$kclust1.cluster==6, "1", "0"))
ipaKmeans = mutate(ipaKmeans, x7= ifelse(ipaKmeans$kclust1.cluster==7, "1", "0"))
ipaKmeans = mutate(ipaKmeans, x8= ifelse(ipaKmeans$kclust1.cluster==8, "1", "0"))
write.table(ipaKmeans,paste0(path,"/Kmeans/ipaKmeans.txt"),  sep = "\t")
```


```{r}
#Create list of vectors containing genes in each cluster. 
#NB, this needs editing if there are more or less than 8 clusters
ClusteredGenes2<-ClusteredGenes[c(1)]
#ClusteredGenes2
listAll<-list()
for(i in 1:8) {
  clusterName<-paste0("x",i)
  #clusterName<-row.names(subset(ClusteredGenes,ClusteredGenes==i))
  clusterName<-(subset(ClusteredGenes$SymbolsKm,ClusteredGenes==i))
  listAll[[i]]<-clusterName
}
#need to name the vectors in the list, example here is for 8 clusters
names(listAll)<-c("X1", "X2", "X3", "X4", "X5", "X6", "X7", "X8")
#if you want to rearrange the order
#listAll<-listAll[c("x3", "x7", "x8", "x2", "x6", "x5", "x4", "x1")]
#lapply(listAll, head)
```

```{r}
groupsName<-"_k-means"
path<-getwd()
dir.create(paste0(path,"/Kmeans/GO"))
```

- CC cellular compartment
- BP biological process
- MF molecular function


```{r}
#Could lump everything together by not specifying "ont".
#The simplify function has been used to cut down on GO redundancy
#CC
cgoCC <- compareCluster(geneCluster = listAll, 
                      universe = AllGeneNames,
                      fun = "enrichGO",
                      OrgDb=org.Hs.eg.db, 
                      #OrgDb=org.Mm.eg.db,
                      keyType="SYMBOL",
                      ont = "CC", 
                      pvalueCutoff=0.05,
                      qvalueCutoff = 0.10)
cgoCC2 <- clusterProfiler::simplify(cgoCC, cutoff=0.7, by="p.adjust", select_fun=min)
#write as spreadsheet
write.csv(as.data.frame(cgoCC2),paste0(path,"/Kmeans/GO/","GO_CC",groupsName,".csv"))



```

```{r}
#CC
png("dummy")
plotcCC=dotplot(cgoCC2,showCategory = 30,
        title = paste0("GO Cellular Compartment ",groupsName))+
   scale_y_discrete(labels=function(x) str_wrap(x, width=80))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
dev.off()
```
```{r,fig.height = 12,fig.width = 12}
plotcCC
```

```{r}
png(paste0(path,"/Kmeans/GO/","GO_CC",groupsName,".png"), width = 1224, height = 1024)
plotcCC
dev.off()
```

GO BP
```{r}
#BP
cgoBP <- compareCluster(geneCluster = listAll, 
                      universe = AllGeneNames,
                      fun = "enrichGO",
                      OrgDb=org.Hs.eg.db, 
                      keyType="SYMBOL",
                      ont = "BP", 
                      pvalueCutoff=0.05,
                      qvalueCutoff = 0.10)
cgoBP2 <- clusterProfiler::simplify(cgoBP, cutoff=0.7, by="p.adjust", select_fun=min)
#write as spreadsheet
write.csv(as.data.frame(cgoBP2),paste0(path,"/Kmeans/GO/","GO_BP",groupsName,".csv"))



```

```{r}
#CC
png("dummy")
plotcBP=dotplot(cgoBP2,showCategory = 30,
        title = paste0("GO Biological Process ",groupsName))+
   scale_y_discrete(labels=function(x) str_wrap(x, width=80))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
dev.off()
```

```{r,fig.height = 20,fig.width = 12}
plotcBP

```

```{r}
png(paste0(path,"/Kmeans/GO/","GO_BP",groupsName,".png"), width = 1024, height = 2024)
plotcBP
dev.off()
```


GO MF

```{r}
#MF
cgoMF <- compareCluster(geneCluster = listAll, 
                      universe = AllGeneNames,
                      fun = "enrichGO",
                      OrgDb=org.Hs.eg.db, 
                      keyType="SYMBOL",
                      ont = "MF", 
                      pvalueCutoff=0.05,
                      qvalueCutoff = 0.10)
cgoMF2 <- clusterProfiler::simplify(cgoMF, cutoff=0.7, by="p.adjust", select_fun=min)
#write as spreadsheet
write.csv(as.data.frame(cgoMF2),paste0(path,"/Kmeans/GO/","GO_MF",groupsName,".csv"))


```

```{r}
#MF
png("dummy")
plotcMF=dotplot(cgoMF2,showCategory = 30,
        title = paste0("GO Molecular Function  ",groupsName))+
   scale_y_discrete(labels=function(x) str_wrap(x, width=80))+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
dev.off()
```

```{r,fig.height = 10,fig.width = 12}
plotcMF

```


```{r}
#print plot to file
png(paste0(path,"/Kmeans/GO/","GO_MF",groupsName,".png"), width = 1424, height = 1024)
plotcMF
dev.off()
```


```{r}
#Enrichr
setEnrichrSite("Enrichr") # Human genes
```

```{r}
websiteLive <- TRUE
dbs <- listEnrichrDbs()
if (is.null(dbs)) websiteLive <- FALSE
if (websiteLive) head(dbs)
```

```{r,fig.height = 10,fig.width = 10}
dbs <- c("Reactome_Pathways_2024","Jensen_DISEASES_Curated_2025", "GO_Biological_Process_2025")
#dbs <- c("Reactome_Pathways_2024","KEGG_2019_Mouse", "GO_Biological_Process_2025")

if (websiteLive) {    enriched1 <- enrichr((subset(ClusteredGenes$SymbolsKm,ClusteredGenes==1)), dbs)}
if (websiteLive) plotEnrich(enriched1[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Cluster 1")
if (websiteLive) plotEnrich(enriched1[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Cluster 1")
if (websiteLive) plotEnrich(enriched1[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Cluster 1")
write.csv(head(enriched1[[1]], 50),paste0(path,"/Kmeans/GO/Reactome_Enrichment_Analysis",groupsName,"_cluster1.csv"))
write.csv(head(enriched1[[2]], 50),paste0(path,"/Kmeans/GO/","Jensen_DISEASES_Enrichment_Analysis",groupsName,"_cluster1.csv"))
write.csv(head(enriched1[[3]], 50),paste0(path,"/Kmeans/GO/","GO_BP_Enrichment_Analysis",groupsName,"_cluster1.csv"))

if (websiteLive) {    enriched2 <- enrichr((subset(ClusteredGenes$SymbolsKm,ClusteredGenes==2)), dbs)}
if (websiteLive) plotEnrich(enriched2[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Cluster 2")
if (websiteLive) plotEnrich(enriched2[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Cluster 2")
if (websiteLive) plotEnrich(enriched2[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Cluster 2")
write.csv(head(enriched2[[1]], 50),paste0(path,"/Kmeans/GO/Reactome_Enrichment_Analysis",groupsName,"_cluster2.csv"))
write.csv(head(enriched2[[2]], 50),paste0(path,"/Kmeans/GO/","Jensen_DISEASES_Enrichment_Analysis",groupsName,"_cluster2.csv"))
write.csv(head(enriched2[[3]], 50),paste0(path,"/Kmeans/GO/","GO_BP_Enrichment_Analysis",groupsName,"_cluster2.csv"))

if (websiteLive) {    enriched3 <- enrichr((subset(ClusteredGenes$SymbolsKm,ClusteredGenes==3)), dbs)}
if (websiteLive) plotEnrich(enriched3[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Cluster 3")
if (websiteLive) plotEnrich(enriched3[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Cluster 3")
if (websiteLive) plotEnrich(enriched3[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Cluster 3")
write.csv(head(enriched3[[1]], 50),paste0(path,"/Kmeans/GO/Reactome_Enrichment_Analysis",groupsName,"_cluster3.csv"))
write.csv(head(enriched3[[2]], 50),paste0(path,"/Kmeans/GO/","Jensen_DISEASES_Enrichment_Analysis",groupsName,"_cluster3.csv"))
write.csv(head(enriched3[[3]], 50),paste0(path,"/Kmeans/GO/","GO_BP_Enrichment_Analysis",groupsName,"_cluster3.csv"))

if (websiteLive) {    enriched4 <- enrichr((subset(ClusteredGenes$SymbolsKm,ClusteredGenes==4)), dbs)}
if (websiteLive) plotEnrich(enriched4[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Cluster 4")
if (websiteLive) plotEnrich(enriched4[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Cluster 4")
if (websiteLive) plotEnrich(enriched4[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Cluster 4")
write.csv(head(enriched4[[1]], 50),paste0(path,"/Kmeans/GO/Reactome_Enrichment_Analysis",groupsName,"_cluster4.csv"))
write.csv(head(enriched4[[2]], 50),paste0(path,"/Kmeans/GO/","Jensen_DISEASES_Enrichment_Analysis",groupsName,"_cluster4.csv"))
write.csv(head(enriched4[[3]], 50),paste0(path,"/Kmeans/GO/","GO_BP_Enrichment_Analysis",groupsName,"_cluster4.csv"))

if (websiteLive) {    enriched5 <- enrichr((subset(ClusteredGenes$SymbolsKm,ClusteredGenes==5)), dbs)}
if (websiteLive) plotEnrich(enriched5[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Cluster 5")
if (websiteLive) plotEnrich(enriched5[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Cluster 5")
if (websiteLive) plotEnrich(enriched5[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Cluster 5")
write.csv(head(enriched5[[1]], 50),paste0(path,"/Kmeans/GO/Reactome_Enrichment_Analysis",groupsName,"_cluster5.csv"))
write.csv(head(enriched5[[2]], 50),paste0(path,"/Kmeans/GO/","Jensen_DISEASES_Enrichment_Analysis",groupsName,"_cluster5.csv"))
write.csv(head(enriched5[[3]], 50),paste0(path,"/Kmeans/GO/","GO_BP_Enrichment_Analysis",groupsName,"_cluster5.csv"))

if (websiteLive) {    enriched6 <- enrichr((subset(ClusteredGenes$SymbolsKm,ClusteredGenes==6)), dbs)}
if (websiteLive) plotEnrich(enriched6[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Cluster 6")
if (websiteLive) plotEnrich(enriched6[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Cluster 6")
if (websiteLive) plotEnrich(enriched6[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Cluster 6")
write.csv(head(enriched6[[1]], 50),paste0(path,"/Kmeans/GO/Reactome_Enrichment_Analysis",groupsName,"_cluster6.csv"))
write.csv(head(enriched6[[2]], 50),paste0(path,"/Kmeans/GO/","Jensen_DISEASES_Enrichment_Analysis",groupsName,"_cluster6.csv"))
write.csv(head(enriched6[[3]], 50),paste0(path,"/Kmeans/GO/","GO_BP_Enrichment_Analysis",groupsName,"_cluster6.csv"))

if (websiteLive) {    enriched7 <- enrichr((subset(ClusteredGenes$SymbolsKm,ClusteredGenes==7)), dbs)}
if (websiteLive) plotEnrich(enriched7[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Cluster 7")
if (websiteLive) plotEnrich(enriched7[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Cluster 7")
if (websiteLive) plotEnrich(enriched7[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Cluster 7")
write.csv(head(enriched7[[1]], 50),paste0(path,"/Kmeans/GO/Reactome_Enrichment_Analysis",groupsName,"_cluster7.csv"))
write.csv(head(enriched7[[2]], 50),paste0(path,"/Kmeans/GO/","Jensen_DISEASES_Enrichment_Analysis",groupsName,"_cluster7.csv"))
write.csv(head(enriched7[[3]], 50),paste0(path,"/Kmeans/GO/","GO_BP_Enrichment_Analysis",groupsName,"_cluster7.csv"))

if (websiteLive) {    enriched8 <- enrichr((subset(ClusteredGenes$SymbolsKm,ClusteredGenes==8)), dbs)}
if (websiteLive) plotEnrich(enriched8[[1]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Reactome Enrichment Analysis Cluster 8")
if (websiteLive) plotEnrich(enriched8[[2]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="Jensen_DISEASES Enrichment Analysis Cluster 8")
if (websiteLive) plotEnrich(enriched8[[3]], showTerms = 30, numChar = 80, y = "Count", orderBy = "P.value", title ="GO_BP Enrichment Analysis Cluster 8")
write.csv(head(enriched8[[1]], 50),paste0(path,"/Kmeans/GO/Reactome_Enrichment_Analysis",groupsName,"_cluster8.csv"))
write.csv(head(enriched8[[2]], 50),paste0(path,"/Kmeans/GO/","Jensen_DISEASES_Enrichment_Analysis",groupsName,"_cluster8.csv"))
write.csv(head(enriched8[[3]], 50),paste0(path,"/Kmeans/GO/","GO_BP_Enrichment_Analysis",groupsName,"_cluster8.csv"))
```


```{r}
#add column of k-means results
clusteredG<-as.data.frame(kclust1$cluster)
colnames(clusteredG)<-("kmeans_cluster")

idxb<-match( resAll$Original_ID, rownames(clusteredG))
resAll$kmeans<- clusteredG$kmeans_cluster[ idxb ]
resAll$kmeans[is.na(resAll$kmeans)] = "Not_clustered"
```



```{r}
#Final steps
#write table with all results
#countsTable <-resAll[,-c(1:2)]
write.table(resAll[,-c(1:2)],"resAll.txt",  sep = "\t")
write.table(ipa1000,"ipa1000.txt",  sep = "\t")
```

```{r}
#save: once happy with clustering save workspace so that it can be recalled
save.image(file="DESeq2&Km.RData")
```




